<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.55">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Richard Goodman">
<meta name="dcterms.date" content="2025-06-24">

<title>Noma Metagenomics: Short-read taxonomic based metagenomic analysis of noma swab vs saliva samples in R</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Noma - swab vs saliva_files/libs/clipboard/clipboard.min.js"></script>
<script src="Noma - swab vs saliva_files/libs/quarto-html/quarto.js"></script>
<script src="Noma - swab vs saliva_files/libs/quarto-html/popper.min.js"></script>
<script src="Noma - swab vs saliva_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Noma - swab vs saliva_files/libs/quarto-html/anchor.min.js"></script>
<link href="Noma - swab vs saliva_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Noma - swab vs saliva_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Noma - swab vs saliva_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Noma - swab vs saliva_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Noma - swab vs saliva_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Noma Metagenomics: Short-read taxonomic based metagenomic analysis of noma swab vs saliva samples in R</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Richard Goodman </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 24, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>This analysis was used in the short-read (Illumina) taxonomic based metagenomic analysis of swab and saliva samples taken from the same patients with noma. The analysis was conducted in R and the paper is posted as a preprint on bioRxiv:</p>
<p>Shotgun metagenomic analysis of the oral microbiomes of noma patients reveals a novel disease-associated organism and potential avenues for disease prevention</p>
<p>Olaleye, M., O’Ferrall A.M., Goodman, R.N. et al.&nbsp;</p>
<section id="getting-started-in-r" class="level1">
<h1>1. Getting Started in R</h1>
<section id="installing-and-loading-packages" class="level2">
<h2 class="anchored" data-anchor-id="installing-and-loading-packages">1.1 Installing and Loading Packages</h2>
<p>Install all necessary packages into R.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure these are all installed as packages first </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Load necessary libraries</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TreeSummarizedExperiment)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="import-and-clean-data" class="level1">
<h1>2. Import and Clean Data</h1>
<p>We will be importing MetaPhlan style Bracken data</p>
<p>First we’ll write a function to import MetaPhlan style bracken data</p>
<section id="load-taxonomic-data" class="level2">
<h2 class="anchored" data-anchor-id="load-taxonomic-data">2.2 Load taxonomic data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">=</span>  <span class="st">"../data/noma_HMP_saliva_bracken_MetaPhlan_style_report_bacteria_only_A1_A40_plus_A14.txt"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb <span class="ot">=</span> <span class="fu">loadFromMetaphlan</span>(file_path)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining the TSE for the rest of the script</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: TreeSummarizedExperiment 
dim: 6187 28 
metadata(0):
assays(1): counts
rownames(6187): s__Coprothermobacter_proteolyticus
  s__Caldisericum_exile ... s__Erysipelothrix_phage_SE-1
  s__Streptococcus_phage_PH10
rowData names(8): Kingdom Phylum ... Species clade_name
colnames(28): A10 A11 ... A8 A9
colData names(0):
reducedDimNames(0):
mainExpName: NULL
altExpNames(5): Phylum Class Order Family Genus
rowLinks: NULL
rowTree: NULL
colLinks: NULL
colTree: NULL</code></pre>
</div>
</div>
</section>
<section id="add-metadata" class="level2">
<h2 class="anchored" data-anchor-id="add-metadata">2.2 Add metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>patient_metadata <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"../data/micro_study_metadata.xlsx"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>sample_to_patient <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"../data/sample_to_patient_A1_A40.xlsx"</span>)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(patient_metadata, sample_to_patient, <span class="at">by =</span> <span class="st">"respondent_id"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>metadata_2 <span class="ot">=</span> metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample_name  <span class="sc">%in%</span> <span class="fu">colnames</span>(tse_metaphlan_sal_swb))</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(metadata_2)</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">sample_name =</span> <span class="fu">colnames</span>(tse_metaphlan_sal_swb))</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(coldata, metadata_2, <span class="at">by =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with this information</span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">=</span> <span class="fu">DataFrame</span>(metadata_3)</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata_df) <span class="ot">=</span> metadata_3<span class="sc">$</span>sample_name</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>t_metadata_df <span class="ot">=</span> <span class="fu">t</span>(metadata_df)</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(t_metadata_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 28</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_sal_swb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 28 rows and 0 columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "sample_name"             "respondent_id"          
[3] "sex"                     "age"                    
[5] "noma_stage_on_admission" "sample_type"            </code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Coun specific metadata</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># count saliva and swab samples</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_type) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 2
  sample_type Count
  &lt;chr&gt;       &lt;int&gt;
1 saliva         17
2 swab           11</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 2 × 2</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co">#sample_type Count</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;chr&gt;       &lt;int&gt;</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  1 saliva         17</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  2 swab           11</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># count participants with both swab and saliva</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(respondent_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 19 × 2
   respondent_id Count
   &lt;chr&gt;         &lt;int&gt;
 1 N1                1
 2 N10               2
 3 N11               1
 4 N12               1
 5 N13               2
 6 N14               1
 7 N15               1
 8 N16               1
 9 N17               2
10 N18               2
11 N19               1
12 N2                1
13 N3                1
14 N4                2
15 N5                2
16 N6                2
17 N7                2
18 N8                1
19 N9                2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(respondent_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Count <span class="sc">==</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 9 × 2
  respondent_id Count
  &lt;chr&gt;         &lt;int&gt;
1 N10               2
2 N13               2
3 N17               2
4 N18               2
5 N4                2
6 N5                2
7 N6                2
8 N7                2
9 N9                2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(respondent_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Count <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 1
  Count
  &lt;int&gt;
1     9</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># next  pull out respondants with both saliva and swab data</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(respondent_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Count <span class="sc">==</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 9 × 2
  respondent_id Count
  &lt;chr&gt;         &lt;int&gt;
1 N10               2
2 N13               2
3 N17               2
4 N18               2
5 N4                2
6 N5                2
7 N6                2
8 N7                2
9 N9                2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>swb_and_sal <span class="ot">=</span> metadata_3 <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(respondent_id) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Count <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>ids_to_remove <span class="ot">=</span> swb_and_sal <span class="sc">%&gt;%</span> <span class="fu">pull</span>(respondent_id)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>swb_and_sal_filtered <span class="ot">=</span> metadata_3  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span>respondent_id <span class="sc">%in%</span> ids_to_remove)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># count saliva and swab samples without particpants above</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>swb_and_sal_filtered <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample_type) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">Count =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 2
  sample_type Count
  &lt;chr&gt;       &lt;int&gt;
1 saliva          8
2 swab            2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Print results</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(ids_to_remove)   <span class="co"># Output as a character vector</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "N10" "N13" "N17" "N18" "N4"  "N5"  "N6"  "N7"  "N9" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add this DataFrame as colData to your TreeSummarizedExperiment object</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_sal_swb) <span class="ot">=</span> metadata_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="inspecting-the-data" class="level2">
<h2 class="anchored" data-anchor-id="inspecting-the-data">2.3 Inspecting the Data</h2>
</section>
<section id="converting-tse-to-other-common-data-formats-e.g.-phyloseq" class="level2">
<h2 class="anchored" data-anchor-id="converting-tse-to-other-common-data-formats-e.g.-phyloseq">2.4 Converting TSE to other common data formats e.g.&nbsp;Phyloseq</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_sal_swb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="non-parametric-statistical-tests" class="level1">
<h1>3. Non-parametric statistical tests</h1>
<section id="preparing-the-data" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data">3.1 Preparing the data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># See above "Converting TSE to other common data formats e.g. Phyloseq"</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># make an assay for abundance</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_metaphlan_sal_swb, <span class="at">assay.type=</span><span class="st">"counts"</span>, <span class="at">method=</span><span class="st">"relabundance"</span>)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyRanks</span>(tse_metaphlan_sal_swb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make an altExp and matrix for order</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse_metaphlan_sal_swb,<span class="st">"Genus"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: TreeSummarizedExperiment 
dim: 1687 28 
metadata(0):
assays(1): counts
rownames(1687): g__Coprothermobacter g__Caldisericum ... g__Phicbkvirus
  g__Casadabanvirus
rowData names(8): Kingdom Phylum ... Species clade_name
colnames(28): A10 A11 ... A8 A9
colData names(0):
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):
rowLinks: NULL
rowTree: NULL
colLinks: NULL
colTree: NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb_genus <span class="ot">=</span> <span class="fu">altExp</span>(tse_metaphlan_sal_swb, <span class="st">"Genus"</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that colData was added successfully</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_sal_swb_genus) <span class="ot">=</span> metadata_df</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>metadata_noma_genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan_sal_swb_genus))</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="co"># species</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>phyloseq_noma <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_sal_swb)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="co"># genus</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>phyloseq_noma <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_sal_swb_genus)</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>phyloseq_noma_esd <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(phyloseq_noma, <span class="cf">function</span>(x) <span class="fl">1E6</span> <span class="sc">*</span> x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ntaxa</span>(phyloseq_noma_esd) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1687</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nsamples</span>(phyloseq_noma_esd) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 28</code></pre>
</div>
</div>
</section>
<section id="permanova-across-entire-dataset" class="level2">
<h2 class="anchored" data-anchor-id="permanova-across-entire-dataset">3.2 Permanova across entire dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate bray curtis distance matrix on main variables </span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>noma.bray <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_noma_esd, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>sample.noma.df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phyloseq_noma_esd))</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>permanova_all <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(noma.bray <span class="sc">~</span> sex , <span class="at">data =</span> sample.noma.df)</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>permanova_all</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = noma.bray ~ sex, data = sample.noma.df)
         Df SumOfSqs      R2      F Pr(&gt;F)
Model     1   0.1930 0.04626 1.2612  0.294
Residual 26   3.9798 0.95374              
Total    27   4.1729 1.00000              </code></pre>
</div>
</div>
<p>Next we will test the beta dispersion</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># All together now</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">adonis2</span>(noma.bray <span class="sc">~</span> sex, <span class="at">data =</span> sample.noma.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = noma.bray ~ sex, data = sample.noma.df)
         Df SumOfSqs      R2      F Pr(&gt;F)
Model     1   0.1930 0.04626 1.2612  0.263
Residual 26   3.9798 0.95374              
Total    27   4.1729 1.00000              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">=</span> <span class="fu">betadisper</span>(noma.bray, sample.noma.df<span class="sc">$</span>sex)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">permutest</span>(beta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df  Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)
Groups     1 0.06942 0.069419 2.3454    999  0.153
Residuals 26 0.76954 0.029598                     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we don't want this to be significant </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="anosim-across-entire-dataset" class="level2">
<h2 class="anchored" data-anchor-id="anosim-across-entire-dataset">3.3 Anosim across entire dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>condition_group <span class="ot">=</span> <span class="fu">get_variable</span>(phyloseq_noma_esd, <span class="st">"sex"</span>)</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span> (<span class="dv">123456</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">anosim</span>(<span class="fu">distance</span>(phyloseq_noma_esd, <span class="st">"bray"</span>), condition_group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
anosim(x = distance(phyloseq_noma_esd, "bray"), grouping = condition_group) 
Dissimilarity: bray 

ANOSIM statistic R: -0.02907 
      Significance: 0.669 

Permutation: free
Number of permutations: 999</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>condition_ano <span class="ot">=</span> <span class="fu">anosim</span>(<span class="fu">distance</span>(phyloseq_noma_esd, <span class="st">"bray"</span>), condition_group)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>condition_ano</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
anosim(x = distance(phyloseq_noma_esd, "bray"), grouping = condition_group) 
Dissimilarity: bray 

ANOSIM statistic R: -0.02907 
      Significance: 0.644 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</section>
<section id="mrpp-across-entire-dataset" class="level2">
<h2 class="anchored" data-anchor-id="mrpp-across-entire-dataset">3.4 MRPP across entire dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co">#condition</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>noma.bray <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_noma_esd, <span class="at">method =</span> <span class="st">"bray"</span>) <span class="co"># Calculate bray curtis distance matrix</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>condition_group <span class="ot">=</span> <span class="fu">get_variable</span>(phyloseq_noma_esd, <span class="st">"sex"</span>) <span class="co"># Make condition Grouping</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Run MRPP</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">mrpp</span>(noma.bray, condition_group, <span class="at">permutations =</span> <span class="dv">999</span>, </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">weight.type =</span> <span class="dv">1</span>, <span class="at">strata =</span> <span class="cn">NULL</span>, <span class="at">parallel =</span> <span class="fu">getOption</span>(<span class="st">"mc.cores"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
vegan::mrpp(dat = noma.bray, grouping = condition_group, permutations = 999,      weight.type = 1, strata = NULL, parallel = getOption("mc.cores")) 

Dissimilarity index: bray 
Weights for groups:  n 

Class means and counts:

      Female Male 
delta 0.4277 0.567
n     12     16   

Chance corrected within-group agreement A: 0.007183 
Based on observed delta 0.5073 and expected delta 0.511 

Significance of delta: 0.266 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</section>
<section id="running-non-parametric-tests-across-several-variables" class="level2">
<h2 class="anchored" data-anchor-id="running-non-parametric-tests-across-several-variables">3.4 Running non-parametric tests across several variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>((<span class="fu">colData</span>(tse_metaphlan_sal_swb_genus)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "sample_name"             "respondent_id"          
[3] "sex"                     "age"                    
[5] "noma_stage_on_admission" "sample_type"            </code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the list of metadata variables you want to test</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>variables_to_test <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"sex"</span>, <span class="st">"age"</span>, <span class="st">"noma_stage_on_admission"</span>, <span class="st">"respondent_id"</span>, <span class="st">"sample_type"</span>)</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set a seed for reproducibility of permutation-based tests</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>bray_dist <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_noma_esd, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the sample data into a data frame for use with adonis2</span></span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a>sample_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phyloseq_noma_esd))</span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an empty list to store the results from each iteration</span></span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>results_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each variable name in the 'variables_to_test' vector</span></span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (variable <span class="cf">in</span> variables_to_test) {</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">"Running tests for variable:"</span>, variable))</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># PERMANOVA (adonis2) </span></span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the statistical formula dynamically for the current variable</span></span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>  formula <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"bray_dist ~"</span>, variable))</span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run the PERMANOVA test using the adonis2 function</span></span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>  permanova_res <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(formula, <span class="at">data =</span> sample_df, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value from the results. It's in the 'Pr(&gt;F)' column.</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>  p_permanova <span class="ot">=</span> permanova_res<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ANOSIM </span></span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the grouping factor (the actual variable data) from the phyloseq object</span></span>
<span id="cb49-32"><a href="#cb49-32" aria-hidden="true" tabindex="-1"></a>  grouping_factor <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">get_variable</span>(phyloseq_noma_esd, variable)</span>
<span id="cb49-33"><a href="#cb49-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-34"><a href="#cb49-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run the ANOSIM test</span></span>
<span id="cb49-35"><a href="#cb49-35" aria-hidden="true" tabindex="-1"></a>  anosim_res <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">anosim</span>(bray_dist, grouping_factor, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb49-36"><a href="#cb49-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-37"><a href="#cb49-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value (significance) from the ANOSIM result</span></span>
<span id="cb49-38"><a href="#cb49-38" aria-hidden="true" tabindex="-1"></a>  p_anosim <span class="ot">=</span> anosim_res<span class="sc">$</span>signif</span>
<span id="cb49-39"><a href="#cb49-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-40"><a href="#cb49-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># MRPP </span></span>
<span id="cb49-41"><a href="#cb49-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The grouping factor is the same as for ANOSIM</span></span>
<span id="cb49-42"><a href="#cb49-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run the MRPP test</span></span>
<span id="cb49-43"><a href="#cb49-43" aria-hidden="true" tabindex="-1"></a>  mrpp_res <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">mrpp</span>(bray_dist, grouping_factor, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb49-44"><a href="#cb49-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-45"><a href="#cb49-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value from the MRPP result</span></span>
<span id="cb49-46"><a href="#cb49-46" aria-hidden="true" tabindex="-1"></a>  p_mrpp <span class="ot">=</span> mrpp_res<span class="sc">$</span>Pvalue</span>
<span id="cb49-47"><a href="#cb49-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb49-48"><a href="#cb49-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store Results </span></span>
<span id="cb49-49"><a href="#cb49-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values for the current variable in our results list.</span></span>
<span id="cb49-50"><a href="#cb49-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># We create a small data frame for this iteration's results.</span></span>
<span id="cb49-51"><a href="#cb49-51" aria-hidden="true" tabindex="-1"></a>  results_list[[variable]] <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb49-52"><a href="#cb49-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">Variable =</span> <span class="fu">paste0</span>(variable, <span class="st">"."</span>),</span>
<span id="cb49-53"><a href="#cb49-53" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">permanova.</span><span class="st">`</span> <span class="ot">=</span> p_permanova,</span>
<span id="cb49-54"><a href="#cb49-54" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">anosim.</span><span class="st">`</span> <span class="ot">=</span> p_anosim,</span>
<span id="cb49-55"><a href="#cb49-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">mrpp.</span><span class="st">`</span> <span class="ot">=</span> p_mrpp,</span>
<span id="cb49-56"><a href="#cb49-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'check.names = FALSE' prevents R from changing our column names</span></span>
<span id="cb49-57"><a href="#cb49-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span> </span>
<span id="cb49-58"><a href="#cb49-58" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb49-59"><a href="#cb49-59" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Running tests for variable: sex</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running tests for variable: age</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running tests for variable: noma_stage_on_admission</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running tests for variable: respondent_id</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Running tests for variable: sample_type</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the list of individual data frames into one final table</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>final_results_table <span class="ot">=</span> <span class="fu">do.call</span>(rbind, results_list)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up the row names of the final table</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(final_results_table) <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the final, consolidated table to the console</span></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(final_results_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  Variable permanova. anosim. mrpp.
1                     sex.      0.294   0.644 0.217
2                     age.      0.001   0.014 0.001
3 noma_stage_on_admission.      0.024   0.029 0.007
4           respondent_id.      0.001   0.001 0.001
5             sample_type.      0.932   0.438 0.928</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(final_results_table, <span class="at">file =</span><span class="st">"../tbls/Table_1B.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permanova-function-for-specific-taxa" class="level2">
<h2 class="anchored" data-anchor-id="permanova-function-for-specific-taxa">3.5 Permanova function for specific taxa</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the counts and taxonomic table</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">=</span> <span class="fu">assay</span>(tse_metaphlan_sal_swb_genus, <span class="st">"counts"</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>tax_table <span class="ot">=</span> <span class="fu">rowData</span>(tse_metaphlan_sal_swb_genus)<span class="sc">$</span>Genus  <span class="co"># Replace "Genus" with your taxonomic level of interest</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">=</span> <span class="fu">colData</span>(tse_metaphlan_sal_swb_genus)</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">=</span> <span class="fu">as.data.frame</span>(sample_data)</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Aggregate counts by Genus</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>aggregated_counts <span class="ot">=</span> <span class="fu">rowsum</span>(counts, tax_table)</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new aggregated TreeSummarizedExperiment object</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>tse_aggregated <span class="ot">=</span> <span class="fu">TreeSummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> aggregated_counts), </span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">colData =</span> sample_data)</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate relative abundances</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>relative_abundances <span class="ot">=</span> <span class="fu">sweep</span>(<span class="fu">assay</span>(tse_aggregated, <span class="st">"counts"</span>), <span class="dv">2</span>, <span class="fu">colSums</span>(<span class="fu">assay</span>(tse_aggregated, <span class="st">"counts"</span>)), <span class="at">FUN =</span> <span class="st">"/"</span>) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to a data frame and group by Treatment</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>relative_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(relative_abundances))</span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span> (<span class="dv">123456</span>)</span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the vector of genera names (without the "g__" prefix)</span></span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>genera <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Prevotella"</span>, <span class="st">"Treponema"</span>, <span class="st">"Neisseria"</span>, <span class="st">"Bacteroides"</span>, </span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Filifactor"</span>, <span class="st">"Porphyromonas"</span>, <span class="st">"Fusobacterium"</span>, <span class="st">"Escherichia"</span>,</span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>            <span class="st">"Selenomonas"</span>, <span class="st">"Aggregatibacter"</span>, <span class="st">"Capnocytophaga"</span>)</span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize an empty data frame to store the results</span></span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>permanova_taxa_results <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Genus =</span> <span class="fu">character</span>(), <span class="at">pvalue =</span> <span class="fu">numeric</span>(), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over each genus</span></span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (genus <span class="cf">in</span> genera) {</span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span> (<span class="dv">123456</span>)</span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the data for the genus; adjust column selection as needed</span></span>
<span id="cb58-35"><a href="#cb58-35" aria-hidden="true" tabindex="-1"></a>  subset_data <span class="ot">=</span> relative_df <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">paste0</span>(<span class="st">"g__"</span>, genus))</span>
<span id="cb58-36"><a href="#cb58-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-37"><a href="#cb58-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate the Bray-Curtis distance</span></span>
<span id="cb58-38"><a href="#cb58-38" aria-hidden="true" tabindex="-1"></a>  bray_dist <span class="ot">=</span> <span class="fu">vegdist</span>(subset_data, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb58-39"><a href="#cb58-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-40"><a href="#cb58-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run PERMANOVA using adonis2</span></span>
<span id="cb58-41"><a href="#cb58-41" aria-hidden="true" tabindex="-1"></a>  adonis_result <span class="ot">=</span> <span class="fu">adonis2</span>(bray_dist <span class="sc">~</span> sample_type, <span class="at">data =</span> groups)</span>
<span id="cb58-42"><a href="#cb58-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-43"><a href="#cb58-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value for the sample_type factor (usually in the first row)</span></span>
<span id="cb58-44"><a href="#cb58-44" aria-hidden="true" tabindex="-1"></a>  pval <span class="ot">=</span> adonis_result<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb58-45"><a href="#cb58-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-46"><a href="#cb58-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Append the result to the results data frame</span></span>
<span id="cb58-47"><a href="#cb58-47" aria-hidden="true" tabindex="-1"></a>  permanova_taxa_results <span class="ot">=</span> <span class="fu">rbind</span>(permanova_taxa_results, <span class="fu">data.frame</span>(<span class="at">Genus =</span> genus, <span class="at">pvalue =</span> pval))</span>
<span id="cb58-48"><a href="#cb58-48" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb58-49"><a href="#cb58-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-50"><a href="#cb58-50" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(permanova_taxa_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             Genus pvalue
1       Prevotella  0.979
2        Treponema  0.617
3        Neisseria  0.853
4      Bacteroides  0.867
5       Filifactor  0.260
6    Porphyromonas  0.953
7    Fusobacterium  0.924
8      Escherichia  0.524
9      Selenomonas  0.928
10 Aggregatibacter  0.234
11  Capnocytophaga  0.803</code></pre>
</div>
</div>
</section>
</section>
<section id="relative-abundance" class="level1">
<h1>4. Relative Abundance</h1>
<p>The top 20 most abundant genera were selected from across the entire dataset and visualised with the plotAbundance function of miaViz.</p>
<section id="plotting-relative-abundance-of-genera-across-samples" class="level2">
<h2 class="anchored" data-anchor-id="plotting-relative-abundance-of-genera-across-samples">4.1 Plotting relative abundance of genera across samples</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a> <span class="co"># Check taxonomy ranks </span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">taxonomyRanks</span>(tse_metaphlan_sal_swb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make an assay for abundance</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  tse_metaphlan_sal_swb <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_metaphlan_sal_swb, <span class="at">assay.type=</span><span class="st">"counts"</span>, <span class="at">method=</span><span class="st">"relabundance"</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make an altExp and matrix for Genus</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">altExp</span>(tse_metaphlan_sal_swb,<span class="st">"Genus"</span>) <span class="ot">=</span> <span class="fu">agglomerateByRank</span>(tse_metaphlan_sal_swb,<span class="st">"Genus"</span>)</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a dataframe of relative abundance </span></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  relabundance_df_Genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan_sal_swb, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>))</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a matric of relative abundance </span></span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>  relabundance_matrix_Genus <span class="ot">=</span> <span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan_sal_swb, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>)</span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># calculate the total relative abundance of each Genus (row sums)</span></span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>  total_relabundance_Genus <span class="ot">=</span> <span class="fu">rowSums</span>(relabundance_matrix_Genus)</span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify the top 20 top Genuss</span></span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(total_relabundance_Genus, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>])</span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Delete everything from start to Genus</span></span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*_g__"</span>,<span class="st">""</span>,top_Genus)</span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Genus back in </span></span>
<span id="cb62-21"><a href="#cb62-21" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"g__"</span>, <span class="fu">length</span>(top_Genus)), top_Genus))</span>
<span id="cb62-22"><a href="#cb62-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Delete the space introduced by this </span></span>
<span id="cb62-23"><a href="#cb62-23" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">" "</span>,<span class="st">""</span>,top_Genus)</span>
<span id="cb62-24"><a href="#cb62-24" aria-hidden="true" tabindex="-1"></a>  top_Genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "g__Prevotella"      "g__Neisseria"       "g__Fusobacterium"  
 [4] "g__Escherichia"     "g__Treponema"       "g__Haemophilus"    
 [7] "g__Aggregatibacter" "g__Streptococcus"   "g__Selenomonas"    
[10] "g__Actinomyces"     "g__Capnocytophaga"  "g__Porphyromonas"  
[13] "g__Bacteroides"     "g__Leptotrichia"    "g__Campylobacter"  
[16] "g__Veillonella"     "g__Filifactor"      "g__Dialister"      
[19] "g__Tannerella"      "g__Rothia"         </code></pre>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a new tse_metaphlan_sal_swb where the top 14 Genuss are recognised, while others are "other"</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  tse_metaphlan_sal_swb_top_20_Genus <span class="ot">=</span> tse_metaphlan_sal_swb</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="fu">rowData</span>(tse_metaphlan_sal_swb_top_20_Genus)<span class="sc">$</span>Genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Coprothermobacter"  "g__Caldisericum"       "g__Desulfurispirillum"
[4] NA                      "g__Endomicrobium"      "g__Endomicrobium"     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowData</span>(tse_metaphlan_sal_swb_top_20_Genus)<span class="sc">$</span>Genus <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">rowData</span>(tse_metaphlan_sal_swb_top_20_Genus)<span class="sc">$</span>Genus <span class="sc">%in%</span> top_Genus, <span class="fu">rowData</span>(tse_metaphlan_sal_swb_top_20_Genus)<span class="sc">$</span>Genus, <span class="st">"-other"</span>)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  genus_colors <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">"-other"</span> <span class="ot">=</span> <span class="st">"#E41A1C"</span>,</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Actinomyces"</span> <span class="ot">=</span> <span class="st">"#377EB8"</span>,</span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Aggregatibacter"</span> <span class="ot">=</span> <span class="st">"#4DAF4A"</span>,</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Bacteroides"</span> <span class="ot">=</span> <span class="st">"#984EA3"</span>,</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Campylobacter"</span> <span class="ot">=</span> <span class="st">"#FF7F00"</span>,</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Capnocytophaga"</span> <span class="ot">=</span> <span class="st">"#FFFF33"</span>,</span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Dialister"</span> <span class="ot">=</span> <span class="st">"#E7298A"</span>,</span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Escherichia"</span> <span class="ot">=</span> <span class="st">"#A65628"</span>,</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Filifactor"</span> <span class="ot">=</span> <span class="st">"#F781BF"</span>,</span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Fusobacterium"</span> <span class="ot">=</span> <span class="st">"#999999"</span>,</span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g_Gemella"</span> <span class="ot">=</span> <span class="st">"#1B9E77"</span>,</span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Haemophilus"</span> <span class="ot">=</span> <span class="st">"#D95F02"</span>,</span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Leptotrichia"</span> <span class="ot">=</span> <span class="st">"#7570B3"</span>,</span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Neisseria"</span> <span class="ot">=</span> <span class="st">"#E7298A"</span>,</span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Porphyromonas"</span> <span class="ot">=</span> <span class="st">"#66A61E"</span>,</span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Prevotella"</span> <span class="ot">=</span> <span class="st">"#E6AB02"</span>,</span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Rothia"</span> <span class="ot">=</span> <span class="st">"#66C2A5"</span>,</span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Schaalia"</span> <span class="ot">=</span> <span class="st">"#FC8D62"</span>,</span>
<span id="cb66-22"><a href="#cb66-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Selenomonas"</span> <span class="ot">=</span> <span class="st">"#8DA0CB"</span>,</span>
<span id="cb66-23"><a href="#cb66-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Streptococcus"</span> <span class="ot">=</span> <span class="st">"#E78AC3"</span>,</span>
<span id="cb66-24"><a href="#cb66-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Tannerella"</span> <span class="ot">=</span> <span class="st">"#E41A1C"</span>,</span>
<span id="cb66-25"><a href="#cb66-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Treponema"</span> <span class="ot">=</span> <span class="st">"#A6D854"</span>,</span>
<span id="cb66-26"><a href="#cb66-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Veillonella"</span> <span class="ot">=</span> <span class="st">"#FFD92F"</span></span>
<span id="cb66-27"><a href="#cb66-27" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb66-28"><a href="#cb66-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb66-29"><a href="#cb66-29" aria-hidden="true" tabindex="-1"></a>  Genus_plot_sal_swb <span class="ot">=</span> <span class="fu">plotAbundance</span>(tse_metaphlan_sal_swb_top_20_Genus, </span>
<span id="cb66-30"><a href="#cb66-30" aria-hidden="true" tabindex="-1"></a>                              <span class="at">assay.type =</span> <span class="st">"relabundance"</span>, </span>
<span id="cb66-31"><a href="#cb66-31" aria-hidden="true" tabindex="-1"></a>                              <span class="at">rank =</span> <span class="st">"Genus"</span>, </span>
<span id="cb66-32"><a href="#cb66-32" aria-hidden="true" tabindex="-1"></a>                              <span class="at">add_x_text =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb66-33"><a href="#cb66-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">30</span>, <span class="at">r =</span> <span class="dv">10</span>, <span class="at">b =</span> <span class="dv">10</span>, <span class="at">l =</span> <span class="dv">10</span>))  </span>
<span id="cb66-34"><a href="#cb66-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb66-35"><a href="#cb66-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-36"><a href="#cb66-36" aria-hidden="true" tabindex="-1"></a>   Genus_plot_sal_swb_cols <span class="ot">=</span>  Genus_plot_sal_swb <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>genus_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for fill is already present.
Adding another scale for fill, which will replace the existing scale.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>   Genus_plot_sal_swb_cols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma---swab-vs-saliva_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Order by ID </span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>  metadata_3<span class="sc">$</span>sample_name</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "A10" "A11" "A13" "A14" "A16" "A17" "A18" "A19" "A1"  "A24" "A25" "A26"
[13] "A27" "A28" "A2"  "A30" "A34" "A35" "A38" "A39" "A3"  "A40" "A4"  "A5" 
[25] "A6"  "A7"  "A8"  "A9" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>  metadata_ID_order <span class="ot">=</span> metadata_3 <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(respondent_id)</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>  metadata_ID_order[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   sample_name respondent_id
1           A1            N1
2          A10           N10
3          A30           N10
4          A11           N11
5          A13           N12
6          A14           N13
7          A34           N13
8          A35           N14
9          A17           N15
10         A16           N16
11         A18           N17
12         A38           N17
13         A19           N18
14         A39           N18
15         A40           N19
16          A2            N2
17          A3            N3
18         A24            N4
19          A4            N4
20         A25            N5
21          A5            N5
22         A26            N6
23          A6            N6
24         A27            N7
25          A7            N7
26          A9            N8
27         A28            N9
28          A8            N9</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>  metadata_ID_order<span class="sc">$</span>sample_name</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "A1"  "A10" "A30" "A11" "A13" "A14" "A34" "A35" "A17" "A16" "A18" "A38"
[13] "A19" "A39" "A40" "A2"  "A3"  "A24" "A4"  "A25" "A5"  "A26" "A6"  "A27"
[25] "A7"  "A9"  "A28" "A8" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>  Ordered_Genus_plot <span class="ot">=</span> Genus_plot_sal_swb_cols <span class="sc">+</span> <span class="fu">scale_x_discrete</span>(<span class="at">limits =</span> metadata_ID_order<span class="sc">$</span>sample_name)</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_1A.png"</span>, <span class="at">plot =</span>  Ordered_Genus_plot, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="differential-analysis-with-deseq" class="level1">
<h1>5. Differential Analysis with Deseq</h1>
<p>Differential analysis used the DESeq2 model on normalised count data and determined fold-change and significant differences between the variables of the noma samples, such as noma stage, age and sex at the genera level.</p>
<p>We concentrated on noma stage for this analysis, there are four stages to noma</p>
<p>Stage 1: Gingivitis Stage 2: Oedema Stage 3: Gangrenous Stage 4: Scarring stage</p>
<section id="preparing-the-data-1" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data-1">5.1 Preparing the data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: TreeSummarizedExperiment 
dim: 6187 28 
metadata(0):
assays(2): counts relabundance
rownames(6187): s__Coprothermobacter_proteolyticus
  s__Caldisericum_exile ... s__Erysipelothrix_phage_SE-1
  s__Streptococcus_phage_PH10
rowData names(8): Kingdom Phylum ... Species clade_name
colnames(28): A10 A11 ... A8 A9
colData names(6): sample_name respondent_id ... noma_stage_on_admission
  sample_type
reducedDimNames(0):
mainExpName: NULL
altExpNames(5): Phylum Class Order Family Genus
rowLinks: NULL
rowTree: NULL
colLinks: NULL
colTree: NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>metadata_noma <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan_sal_swb))</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(metadata_noma<span class="sc">$</span>noma_stage_on_admission)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Stage_2" "Stage_4" "Stage_3" "Stage_1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co">#__________________________Makes into Genus________________________________________________</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="co"># make an assay for abundance</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_metaphlan_sal_swb, <span class="at">assay.type=</span><span class="st">"counts"</span>, <span class="at">method=</span><span class="st">"relabundance"</span>)</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyRanks</span>(tse_metaphlan_sal_swb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make an altExp and matrix for order</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse_metaphlan_sal_swb,<span class="st">"Genus"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: TreeSummarizedExperiment 
dim: 1805 28 
metadata(1): agglomerated_by_rank
assays(2): counts relabundance
rownames(1805):
  NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter
  NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum
  ... NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus
  NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus
rowData names(8): Kingdom Phylum ... Species clade_name
colnames(28): A10 A11 ... A8 A9
colData names(6): sample_name respondent_id ... noma_stage_on_admission
  sample_type
reducedDimNames(0):
mainExpName: NULL
altExpNames(0):
rowLinks: NULL
rowTree: NULL
colLinks: NULL
colTree: NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_sal_swb_genus <span class="ot">=</span> <span class="fu">altExp</span>(tse_metaphlan_sal_swb, <span class="st">"Genus"</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that colData was added successfully</span></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_sal_swb_genus) <span class="ot">=</span> metadata_df</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>metadata_noma_genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan_sal_swb_genus))</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Genus level</span></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan_noma <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_sal_swb_genus)</span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>deseq2_metaphlan_noma <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">phyloseq_to_deseq2</span>(phyloseq_metaphlan_noma, <span class="at">design =</span> <span class="sc">~</span>noma_stage_on_admission)</span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a><span class="co">#__________Remove or edit if other taxonomy class is needed_________________________________________</span></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Species level</span></span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan_noma <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_sal_swb)</span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a>deseq2_metaphlan_noma <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">phyloseq_to_deseq2</span>(phyloseq_metaphlan_noma, <span class="at">design =</span> <span class="sc">~</span>noma_stage_on_admission)</span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-24"><a href="#cb84-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Genus level</span></span>
<span id="cb84-25"><a href="#cb84-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb84-26"><a href="#cb84-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-27"><a href="#cb84-27" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan_noma <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_sal_swb_genus)</span>
<span id="cb84-28"><a href="#cb84-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-29"><a href="#cb84-29" aria-hidden="true" tabindex="-1"></a>deseq2_metaphlan_noma <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">phyloseq_to_deseq2</span>(phyloseq_metaphlan_noma, <span class="at">design =</span> <span class="sc">~</span>noma_stage_on_admission)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-analysis-with-deseq-1" class="level2">
<h2 class="anchored" data-anchor-id="differential-analysis-with-deseq-1">5.2 Differential analysis with Deseq</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>dds_noma <span class="ot">=</span> deseq2_metaphlan_noma</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dds_noma) <span class="ot">=</span> <span class="er">~</span> noma_stage_on_admission  <span class="co"># Replace with your column name for condition</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2 analysis</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>dds_stage <span class="ot">=</span> <span class="fu">DESeq</span>(dds_noma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extract-results-for-noma-stage" class="level2">
<h2 class="anchored" data-anchor-id="extract-results-for-noma-stage">5.3 Extract results for noma stage</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(metadata_noma<span class="sc">$</span>noma_stage_on_admission)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Stage_2" "Stage_4" "Stage_3" "Stage_1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results for diseased vs healthy</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>res_stage <span class="ot">=</span> <span class="fu">results</span>(dds_stage)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>res_stage</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma stage on admission Stage 4 vs Stage 1 
Wald test p-value: noma stage on admission Stage 4 vs Stage 1 
DataFrame with 1805 rows and 6 columns
                                                                                                                           baseMean
                                                                                                                          &lt;numeric&gt;
NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter   7.47996
NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum                                     28.61754
NA_p__Chrysiogenetes_c__Chrysiogenetes_o__Chrysiogenales_f__Chrysiogenaceae_g__Desulfurispirillum                          17.38504
NA_p__Elusimicrobia_NA_NA_NA_NA                                                                                             8.41601
NA_p__Elusimicrobia_c__Endomicrobia_o__Endomicrobiales_f__Endomicrobiaceae_g__Endomicrobium                                28.88837
...                                                                                                                             ...
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Bingvirus                                                                      4.681680
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Patiencevirus                                                                  0.215018
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Nonanavirus                                                                    2.366020
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus                                                                    0.846389
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus                                                                 0.584052
                                                                                                                          log2FoldChange
                                                                                                                               &lt;numeric&gt;
NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter      2.7866702
NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum                                         0.0926400
NA_p__Chrysiogenetes_c__Chrysiogenetes_o__Chrysiogenales_f__Chrysiogenaceae_g__Desulfurispirillum                              1.3038817
NA_p__Elusimicrobia_NA_NA_NA_NA                                                                                               -0.0261229
NA_p__Elusimicrobia_c__Endomicrobia_o__Endomicrobiales_f__Endomicrobiaceae_g__Endomicrobium                                   -1.6331178
...                                                                                                                                  ...
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Bingvirus                                                                           4.594973
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Patiencevirus                                                                      -0.110480
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Nonanavirus                                                                         1.179819
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus                                                                        -0.110494
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus                                                                     -0.110509
                                                                                                                              lfcSE
                                                                                                                          &lt;numeric&gt;
NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter  1.742500
NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum                                     1.178663
NA_p__Chrysiogenetes_c__Chrysiogenetes_o__Chrysiogenales_f__Chrysiogenaceae_g__Desulfurispirillum                          1.131759
NA_p__Elusimicrobia_NA_NA_NA_NA                                                                                            2.087037
NA_p__Elusimicrobia_c__Endomicrobia_o__Endomicrobiales_f__Endomicrobiaceae_g__Endomicrobium                                0.964105
...                                                                                                                             ...
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Bingvirus                                                                       4.01031
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Patiencevirus                                                                   7.90110
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Nonanavirus                                                                     2.16880
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus                                                                     4.87981
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus                                                                  7.90110
                                                                                                                                stat
                                                                                                                           &lt;numeric&gt;
NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter  1.5992367
NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum                                     0.0785975
NA_p__Chrysiogenetes_c__Chrysiogenetes_o__Chrysiogenales_f__Chrysiogenaceae_g__Desulfurispirillum                          1.1520846
NA_p__Elusimicrobia_NA_NA_NA_NA                                                                                           -0.0125168
NA_p__Elusimicrobia_c__Endomicrobia_o__Endomicrobiales_f__Endomicrobiaceae_g__Endomicrobium                               -1.6939218
...                                                                                                                              ...
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Bingvirus                                                                      1.1457903
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Patiencevirus                                                                 -0.0139828
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Nonanavirus                                                                    0.5439975
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus                                                                   -0.0226432
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus                                                                -0.0139866
                                                                                                                             pvalue
                                                                                                                          &lt;numeric&gt;
NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter 0.1097680
NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum                                    0.9373528
NA_p__Chrysiogenetes_c__Chrysiogenetes_o__Chrysiogenales_f__Chrysiogenaceae_g__Desulfurispirillum                         0.2492863
NA_p__Elusimicrobia_NA_NA_NA_NA                                                                                           0.9900133
NA_p__Elusimicrobia_c__Endomicrobia_o__Endomicrobiales_f__Endomicrobiaceae_g__Endomicrobium                               0.0902802
...                                                                                                                             ...
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Bingvirus                                                                      0.251882
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Patiencevirus                                                                  0.988844
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Nonanavirus                                                                    0.586443
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus                                                                    0.981935
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus                                                                 0.988841
                                                                                                                               padj
                                                                                                                          &lt;numeric&gt;
NA_p__Coprothermobacterota_c__Coprothermobacteria_o__Coprothermobacterales_f__Coprothermobacteraceae_g__Coprothermobacter  0.996147
NA_p__Caldiserica_c__Caldisericia_o__Caldisericales_f__Caldisericaceae_g__Caldisericum                                     0.996147
NA_p__Chrysiogenetes_c__Chrysiogenetes_o__Chrysiogenales_f__Chrysiogenaceae_g__Desulfurispirillum                          0.996147
NA_p__Elusimicrobia_NA_NA_NA_NA                                                                                            0.996147
NA_p__Elusimicrobia_c__Endomicrobia_o__Endomicrobiales_f__Endomicrobiaceae_g__Endomicrobium                                0.996147
...                                                                                                                             ...
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Bingvirus                                                                      0.996147
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Patiencevirus                                                                  0.996147
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Nonanavirus                                                                    0.996147
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Phicbkvirus                                                                    0.996147
NA_NA_NA_o__Caudovirales_f__Siphoviridae_g__Casadabanvirus                                                                 0.996147</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>string <span class="ot">=</span> <span class="st">"NA_p__Actinobacteria_c__Actinobacteria_o__Streptomycetales_f__Streptomycetaceae_g__Streptomyces"</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>string_result <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__Streptomyces)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(string_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Streptomyces"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up genus names for dds</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(dds_stage) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(dds_stage))</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up genus names for res</span></span>
<span id="cb92-5"><a href="#cb92-5" aria-hidden="true" tabindex="-1"></a>res_stage<span class="sc">@</span>rownames <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, res_stage<span class="sc">@</span>rownames)</span>
<span id="cb92-6"><a href="#cb92-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-7"><a href="#cb92-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb92-8"><a href="#cb92-8" aria-hidden="true" tabindex="-1"></a>res_stage_1_2 <span class="ot">=</span> <span class="fu">results</span>(dds_stage, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"noma_stage_on_admission"</span>, <span class="st">"Stage_1"</span>, <span class="st">"Stage_2"</span>))</span>
<span id="cb92-9"><a href="#cb92-9" aria-hidden="true" tabindex="-1"></a>res_stage_1_2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma_stage_on_admission Stage_1 vs Stage_2 
Wald test p-value: noma_stage_on_admission Stage_1 vs Stage_2 
DataFrame with 1805 rows and 6 columns
                                 baseMean log2FoldChange     lfcSE       stat
                                &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
g__Coprothermobacter              7.47996      -2.911303  1.528706   -1.90442
g__Caldisericum                  28.61754      -1.106190  0.884101   -1.25120
g__Desulfurispirillum            17.38504      -2.290095  0.922457   -2.48260
NA_p__Elusimicrobia_NA_NA_NA_NA   8.41601       1.684493  1.571201    1.07211
g__Endomicrobium                 28.88837      -0.262504  0.660272   -0.39757
...                                   ...            ...       ...        ...
g__Bingvirus                     4.681680      -0.718991   3.11878 -0.2305358
g__Patiencevirus                 0.215018      -0.536883   5.91932 -0.0907000
g__Nonanavirus                   2.366020       0.111032   1.71468  0.0647539
g__Phicbkvirus                   0.846389      -1.605449   3.64212 -0.4408008
g__Casadabanvirus                0.584052      -1.478896   5.91586 -0.2499884
                                   pvalue      padj
                                &lt;numeric&gt; &lt;numeric&gt;
g__Coprothermobacter            0.0568551        NA
g__Caldisericum                 0.2108604        NA
g__Desulfurispirillum           0.0130426        NA
NA_p__Elusimicrobia_NA_NA_NA_NA 0.2836725        NA
g__Endomicrobium                0.6909473        NA
...                                   ...       ...
g__Bingvirus                     0.817675        NA
g__Patiencevirus                 0.927731        NA
g__Nonanavirus                   0.948370        NA
g__Phicbkvirus                   0.659357        NA
g__Casadabanvirus                0.802596        NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>res_stage_1_2[<span class="fu">order</span>(res_stage_1_2<span class="sc">$</span>padj),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma_stage_on_admission Stage_1 vs Stage_2 
Wald test p-value: noma_stage_on_admission Stage_1 vs Stage_2 
DataFrame with 1805 rows and 6 columns
                   baseMean log2FoldChange     lfcSE       stat      pvalue
                  &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;   &lt;numeric&gt;
g__Mogibacterium  3581.1698       -2.85300  0.525510   -5.42901 5.66671e-08
g__Cellulophaga    170.2722        2.24655  0.449909    4.99335 5.93400e-07
g__Nitrospira       69.9556        2.36688  0.482007    4.91046 9.08610e-07
g__Murdochiella    242.5915       -2.94364  0.631600   -4.66061 3.15267e-06
g__Ndongobacter    182.0927       -2.79093  0.640210   -4.35940 1.30419e-05
...                     ...            ...       ...        ...         ...
g__Bingvirus       4.681680      -0.718991   3.11878 -0.2305358    0.817675
g__Patiencevirus   0.215018      -0.536883   5.91932 -0.0907000    0.927731
g__Nonanavirus     2.366020       0.111032   1.71468  0.0647539    0.948370
g__Phicbkvirus     0.846389      -1.605449   3.64212 -0.4408008    0.659357
g__Casadabanvirus  0.584052      -1.478896   5.91586 -0.2499884    0.802596
                         padj
                    &lt;numeric&gt;
g__Mogibacterium  4.08003e-05
g__Cellulophaga   2.13624e-04
g__Nitrospira     2.18066e-04
g__Murdochiella   5.67481e-04
g__Ndongobacter   1.87803e-03
...                       ...
g__Bingvirus               NA
g__Patiencevirus           NA
g__Nonanavirus             NA
g__Phicbkvirus             NA
g__Casadabanvirus          NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>res_stage_2_3 <span class="ot">=</span> <span class="fu">results</span>(dds_stage, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"noma_stage_on_admission"</span>, <span class="st">"Stage_2"</span>, <span class="st">"Stage_3"</span>))</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>res_stage_2_3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma_stage_on_admission Stage_2 vs Stage_3 
Wald test p-value: noma_stage_on_admission Stage_2 vs Stage_3 
DataFrame with 1805 rows and 6 columns
                                 baseMean log2FoldChange     lfcSE      stat
                                &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
g__Coprothermobacter              7.47996      0.0639333  0.495940  0.128914
g__Caldisericum                  28.61754      0.2541549  0.475294  0.534732
g__Desulfurispirillum            17.38504     -0.0735806  0.349031 -0.210814
NA_p__Elusimicrobia_NA_NA_NA_NA   8.41601      0.2393834  0.931148  0.257084
g__Endomicrobium                 28.88837      0.6025922  0.369975  1.628739
...                                   ...            ...       ...       ...
g__Bingvirus                     4.681680       2.063932  1.765658  1.168931
g__Patiencevirus                 0.215018       1.563184  3.402277  0.459452
g__Nonanavirus                   2.366020       0.711787  0.944862  0.753324
g__Phicbkvirus                   0.846389       1.445930  2.043016  0.707743
g__Casadabanvirus                0.584052       2.505227  3.396244  0.737646
                                   pvalue      padj
                                &lt;numeric&gt; &lt;numeric&gt;
g__Coprothermobacter             0.897426  0.950142
g__Caldisericum                  0.592835  0.795011
g__Desulfurispirillum            0.833032  0.918498
NA_p__Elusimicrobia_NA_NA_NA_NA  0.797114  0.908062
g__Endomicrobium                 0.103368  0.305995
...                                   ...       ...
g__Bingvirus                     0.242432  0.516453
g__Patiencevirus                 0.645909        NA
g__Nonanavirus                   0.451255  0.701377
g__Phicbkvirus                   0.479105  0.725474
g__Casadabanvirus                0.460729        NA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>res_stage_3_4 <span class="ot">=</span> <span class="fu">results</span>(dds_stage, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"noma_stage_on_admission"</span>, <span class="st">"Stage_3"</span>, <span class="st">"Stage_4"</span>))</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>res_stage_3_4</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma_stage_on_admission Stage_3 vs Stage_4 
Wald test p-value: noma_stage_on_admission Stage_3 vs Stage_4 
DataFrame with 1805 rows and 6 columns
                                 baseMean log2FoldChange     lfcSE       stat
                                &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt;
g__Coprothermobacter              7.47996       0.060699  0.972280  0.0624295
g__Caldisericum                  28.61754       0.759395  0.912973  0.8317826
g__Desulfurispirillum            17.38504       1.059794  0.742816  1.4267248
NA_p__Elusimicrobia_NA_NA_NA_NA   8.41601      -1.897754  1.659549 -1.1435359
g__Endomicrobium                 28.88837       1.293030  0.793990  1.6285213
...                                   ...            ...       ...        ...
g__Bingvirus                     4.681680      -5.939914   3.07791 -1.9298508
g__Patiencevirus                 0.215018       0.000000   6.24243  0.0000000
g__Nonanavirus                   2.366020      -2.002638   1.62983 -1.2287420
g__Phicbkvirus                   0.846389       0.270014   3.83691  0.0703726
g__Casadabanvirus                0.584052       0.000000   6.24243  0.0000000
                                   pvalue      padj
                                &lt;numeric&gt; &lt;numeric&gt;
g__Coprothermobacter             0.950221  1.000000
g__Caldisericum                  0.405532  0.971049
g__Desulfurispirillum            0.153659  0.824586
NA_p__Elusimicrobia_NA_NA_NA_NA  0.252816  0.875394
g__Endomicrobium                 0.103414  0.761428
...                                   ...       ...
g__Bingvirus                    0.0536253  0.746134
g__Patiencevirus                1.0000000  1.000000
g__Nonanavirus                  0.2191686  0.859956
g__Phicbkvirus                  0.9438971  1.000000
g__Casadabanvirus               1.0000000  1.000000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">results</span>(dds_stage, <span class="at">tidy=</span><span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                              row  baseMean log2FoldChange     lfcSE
1            g__Coprothermobacter  7.479963     2.78667021 1.7425001
2                 g__Caldisericum 28.617545     0.09263997 1.1786631
3           g__Desulfurispirillum 17.385038     1.30388172 1.1317587
4 NA_p__Elusimicrobia_NA_NA_NA_NA  8.416007    -0.02612293 2.0870365
5                g__Endomicrobium 28.888374    -1.63311780 0.9641046
6               g__Elusimicrobium 16.824150     0.81262987 1.1344232
         stat     pvalue      padj
1  1.59923673 0.10976801 0.9961471
2  0.07859750 0.93735278 0.9961471
3  1.15208458 0.24928632 0.9961471
4 -0.01251676 0.99001333 0.9961471
5 -1.69392176 0.09028015 0.9961471
6  0.71633747 0.47378300 0.9961471</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_stage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 1804 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 2, 0.11%
LFC &lt; 0 (down)     : 1, 0.055%
outliers [1]       : 0, 0%
low counts [2]     : 1, 0.055%
(mean count &lt; 0)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>res_stage_ordered <span class="ot">=</span> res_stage[<span class="fu">order</span>(res_stage<span class="sc">$</span>padj),]</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_stage_ordered, <span class="at">n =</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma stage on admission Stage 4 vs Stage 1 
Wald test p-value: noma stage on admission Stage 4 vs Stage 1 
DataFrame with 20 rows and 6 columns
                       baseMean log2FoldChange     lfcSE      stat      pvalue
                      &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
g__Desulfomicrobium    403.8747        7.46010  1.312825   5.68248 1.32759e-08
g__Propionibacterium   195.0813        4.62326  0.985293   4.69227 2.70190e-06
g__Nitrospira           69.9556       -2.55772  0.665799  -3.84157 1.22247e-04
g__Duncaniella         131.7423       20.48251  6.281503   3.26077 1.11112e-03
g__Desulfobulbus       235.4306        4.12281  1.382327   2.98251 2.85893e-03
...                         ...            ...       ...       ...         ...
g__Endomicrobium       28.88837      -1.633118  0.964105 -1.693922   0.0902802
g__Elusimicrobium      16.82415       0.812630  1.134423  0.716337   0.4737830
g__Dictyoglomus        42.00494      -0.369790  0.833218 -0.443810   0.6571802
g__Caldimicrobium      15.23297       0.212413  1.125505  0.188727   0.8503066
g__Thermodesulfatator   8.68596       0.617753  1.150849  0.536780   0.5914194
                             padj
                        &lt;numeric&gt;
g__Desulfomicrobium   2.39497e-05
g__Propionibacterium  2.43711e-03
g__Nitrospira         7.35115e-02
g__Duncaniella        5.01114e-01
g__Desulfobulbus      9.08700e-01
...                           ...
g__Endomicrobium         0.996147
g__Elusimicrobium        0.996147
g__Dictyoglomus          0.996147
g__Caldimicrobium        0.996147
g__Thermodesulfatator    0.996147</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>res_stage_ordered_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_stage_ordered)</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>res_stage_1_2_ordered <span class="ot">=</span> res_stage[<span class="fu">order</span>(res_stage_1_2<span class="sc">$</span>padj),]</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_stage_1_2_ordered, <span class="at">n =</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma stage on admission Stage 4 vs Stage 1 
Wald test p-value: noma stage on admission Stage 4 vs Stage 1 
DataFrame with 20 rows and 6 columns
                   baseMean log2FoldChange     lfcSE      stat      pvalue
                  &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
g__Mogibacterium  3581.1698        1.97011  0.698828   2.81917 0.004814851
g__Cellulophaga    170.2722       -1.67956  0.604107  -2.78024 0.005431899
g__Nitrospira       69.9556       -2.55772  0.665799  -3.84157 0.000122247
g__Murdochiella    242.5915        1.43749  0.825476   1.74140 0.081612815
g__Ndongobacter    182.0927        1.04213  0.838700   1.24256 0.214031712
...                     ...            ...       ...       ...         ...
g__Psychrobacter    375.849      -2.020202  1.057813 -1.909792    0.056160
g__Clostridioides  2032.123       0.214099  0.702750  0.304660    0.760625
g__Dialister      20986.584      -0.376389  1.305767 -0.288251    0.773155
g__Jeotgalibaca     174.110      -1.162595  0.984221 -1.181234    0.237510
g__Jonquetella      129.796       2.357468  1.605554  1.468321    0.142017
                       padj
                  &lt;numeric&gt;
g__Mogibacterium  0.9091111
g__Cellulophaga   0.9091111
g__Nitrospira     0.0735115
g__Murdochiella   0.9961471
g__Ndongobacter   0.9961471
...                     ...
g__Psychrobacter   0.996147
g__Clostridioides  0.996147
g__Dialister       0.996147
g__Jeotgalibaca    0.996147
g__Jonquetella     0.996147</code></pre>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>res_stage_2_3_ordered <span class="ot">=</span> res_stage[<span class="fu">order</span>(res_stage_2_3<span class="sc">$</span>padj),]</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_stage_2_3_ordered, <span class="at">n =</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma stage on admission Stage 4 vs Stage 1 
Wald test p-value: noma stage on admission Stage 4 vs Stage 1 
DataFrame with 20 rows and 6 columns
                     baseMean log2FoldChange     lfcSE       stat    pvalue
                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
g__Schaalia         7862.7080       3.767856   1.61670   2.330588 0.0197751
g__Aeromicrobium      57.6003       0.380902   1.22484   0.310981 0.7558150
g__Cryobacterium      28.7933       1.142165   1.10698   1.031787 0.3021718
g__Brachybacterium   163.1061       0.867287   1.23925   0.699848 0.4840224
g__Cellulomonas      109.3540       1.323469   1.30091   1.017341 0.3089911
...                       ...            ...       ...        ...       ...
g__Dictyoglomus       42.0049     -0.3697902  0.833218 -0.4438097  0.657180
g__Orientia           23.4129     -0.2811862  0.918582 -0.3061089  0.759522
g__Stenotrophomonas  311.3936      0.2442722  1.035628  0.2358688  0.813534
g__Paeniclostridium  293.2947     -0.0980988  0.628071 -0.1561906  0.875883
g__Rickettsia        152.7651     -0.0468430  0.887966 -0.0527532  0.957929
                         padj
                    &lt;numeric&gt;
g__Schaalia          0.996147
g__Aeromicrobium     0.996147
g__Cryobacterium     0.996147
g__Brachybacterium   0.996147
g__Cellulomonas      0.996147
...                       ...
g__Dictyoglomus      0.996147
g__Orientia          0.996147
g__Stenotrophomonas  0.996147
g__Paeniclostridium  0.996147
g__Rickettsia        0.996147</code></pre>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a>res_stage_3_4_ordered <span class="ot">=</span> res_stage[<span class="fu">order</span>(res_stage_3_4<span class="sc">$</span>padj),]</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_stage_2_3_ordered, <span class="at">n =</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): noma stage on admission Stage 4 vs Stage 1 
Wald test p-value: noma stage on admission Stage 4 vs Stage 1 
DataFrame with 20 rows and 6 columns
                     baseMean log2FoldChange     lfcSE       stat    pvalue
                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
g__Schaalia         7862.7080       3.767856   1.61670   2.330588 0.0197751
g__Aeromicrobium      57.6003       0.380902   1.22484   0.310981 0.7558150
g__Cryobacterium      28.7933       1.142165   1.10698   1.031787 0.3021718
g__Brachybacterium   163.1061       0.867287   1.23925   0.699848 0.4840224
g__Cellulomonas      109.3540       1.323469   1.30091   1.017341 0.3089911
...                       ...            ...       ...        ...       ...
g__Dictyoglomus       42.0049     -0.3697902  0.833218 -0.4438097  0.657180
g__Orientia           23.4129     -0.2811862  0.918582 -0.3061089  0.759522
g__Stenotrophomonas  311.3936      0.2442722  1.035628  0.2358688  0.813534
g__Paeniclostridium  293.2947     -0.0980988  0.628071 -0.1561906  0.875883
g__Rickettsia        152.7651     -0.0468430  0.887966 -0.0527532  0.957929
                         padj
                    &lt;numeric&gt;
g__Schaalia          0.996147
g__Aeromicrobium     0.996147
g__Cryobacterium     0.996147
g__Brachybacterium   0.996147
g__Cellulomonas      0.996147
...                       ...
g__Dictyoglomus      0.996147
g__Orientia          0.996147
g__Stenotrophomonas  0.996147
g__Paeniclostridium  0.996147
g__Rickettsia        0.996147</code></pre>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(res_stage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
out of 1804 with nonzero total read count
adjusted p-value &lt; 0.1
LFC &gt; 0 (up)       : 2, 0.11%
LFC &lt; 0 (down)     : 1, 0.055%
outliers [1]       : 0, 0%
low counts [2]     : 1, 0.055%
(mean count &lt; 0)
[1] see 'cooksCutoff' argument of ?results
[2] see 'independentFiltering' argument of ?results</code></pre>
</div>
</div>
</section>
<section id="inspect-genera-that-are-significantly-different-between-noma-stages" class="level2">
<h2 class="anchored" data-anchor-id="inspect-genera-that-are-significantly-different-between-noma-stages">5.4 Inspect genera that are significantly different between noma stages</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>significant_stage <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_stage) <span class="sc">%&gt;%</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_stage) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE     stat       pvalue
g__Desulfomicrobium  403.8747       7.460098 1.3128254 5.682476 1.327588e-08
g__Propionibacterium 195.0813       4.623263 0.9852933 4.692270 2.701900e-06
                             padj
g__Desulfomicrobium  2.394969e-05
g__Propionibacterium 2.437113e-03</code></pre>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a> significant_stage_1_2 <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_stage_1_2) <span class="sc">%&gt;%</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_stage_1_2) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   baseMean log2FoldChange     lfcSE      stat       pvalue
g__Nitrospira      69.95558       2.366876 0.4820066  4.910464 9.086102e-07
g__Jonquetella    129.79597      -4.254723 1.2420148 -3.425662 6.133036e-04
g__Mariprofundus   47.19083      -2.355499 0.7411882 -3.178004 1.482928e-03
g__Psychrobacter  375.84925       2.877568 0.7925132  3.630940 2.823904e-04
g__Actinobacillus 616.12402       2.241077 0.7243168  3.094056 1.974403e-03
g__Alicycliphilus  73.68619       3.091691 0.9429976  3.278578 1.043316e-03
                          padj
g__Nitrospira     0.0002180664
g__Jonquetella    0.0220695776
g__Mariprofundus  0.0343369089
g__Psychrobacter  0.0127075686
g__Actinobacillus 0.0406162827
g__Alicycliphilus 0.0293744999</code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>significant_stage_2_3 <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_stage_2_3) <span class="sc">%&gt;%</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_stage_2_3) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                baseMean log2FoldChange
g__Dictyoglomus                                 42.00494       1.536272
g__Thermovibrio                                 19.47635      -1.494045
g__Thermocrinis                                 18.29961       1.306161
g__Thermodesulfovibrio                          34.83078       1.082793
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA 2244.19436      -1.783110
g__Orientia                                     23.41292       1.715422
                                                  lfcSE      stat       pvalue
g__Dictyoglomus                               0.3573907  4.298579 1.718967e-05
g__Thermovibrio                               0.4586414 -3.257544 1.123807e-03
g__Thermocrinis                               0.3646499  3.581958 3.410283e-04
g__Thermodesulfovibrio                        0.3215428  3.367491 7.585545e-04
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA 0.5373296 -3.318466 9.051345e-04
g__Orientia                                   0.4103212  4.180681 2.906377e-05
                                                     padj
g__Dictyoglomus                               0.001599713
g__Thermovibrio                               0.022017740
g__Thermocrinis                               0.009765215
g__Thermodesulfovibrio                        0.016610112
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA 0.019253505
g__Orientia                                   0.002461660</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>significant_stage_3_4 <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_stage_3_4) <span class="sc">%&gt;%</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_stage_3_4) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE      stat       pvalue
g__Sneathia          562.9476      -3.708609 0.8568309 -4.328286 1.502746e-05
g__Desulfomicrobium  403.8747      -7.076476 1.0173671 -6.955676 3.508747e-12
g__Propionibacterium 195.0813      -3.442139 0.7660950 -4.493097 7.019467e-06
                             padj
g__Sneathia          9.036511e-03
g__Desulfomicrobium  6.329779e-09
g__Propionibacterium 6.331560e-03</code></pre>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results </span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage <span class="ot">=</span> significant_stage[<span class="fu">order</span>(significant_stage<span class="sc">$</span>padj),]</span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>sig_res_stage<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage)</span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE     stat       pvalue
g__Desulfomicrobium  403.8747       7.460098 1.3128254 5.682476 1.327588e-08
g__Propionibacterium 195.0813       4.623263 0.9852933 4.692270 2.701900e-06
                             padj                genus
g__Desulfomicrobium  2.394969e-05  g__Desulfomicrobium
g__Propionibacterium 2.437113e-03 g__Propionibacterium</code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE     stat       pvalue
g__Desulfomicrobium  403.8747       7.460098 1.3128254 5.682476 1.327588e-08
g__Propionibacterium 195.0813       4.623263 0.9852933 4.692270 2.701900e-06
                             padj                genus
g__Desulfomicrobium  2.394969e-05  g__Desulfomicrobium
g__Propionibacterium 2.437113e-03 g__Propionibacterium</code></pre>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 1 to 2 </span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_1_2 <span class="ot">=</span> significant_stage_2_3[<span class="fu">order</span>(significant_stage_1_2<span class="sc">$</span>padj),]</span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage_1_2<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage_1_2)</span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_1_2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 baseMean log2FoldChange     lfcSE      stat       pvalue
g__Arsenophonus  48.58743       1.156151 0.3497172  3.305960 9.465155e-04
g__Massilia     172.70323      -1.251125 0.2667741 -4.689830 2.734322e-06
g__Dictyoglomus  42.00494       1.536272 0.3573907  4.298579 1.718967e-05
g__Cetia         26.75105       1.783599 0.4617477  3.862714 1.121344e-04
g__Ochrobactrum 130.21463      -1.413041 0.3917376 -3.607111 3.096247e-04
g__Orrella       44.93631      -3.081929 0.6977098 -4.417207 9.998460e-06
                        padj           genus
g__Arsenophonus 0.0195744662 g__Arsenophonus
g__Massilia     0.0004860202     g__Massilia
g__Dictyoglomus 0.0015997133 g__Dictyoglomus
g__Cetia        0.0053860667        g__Cetia
g__Ochrobactrum 0.0092206240 g__Ochrobactrum
g__Orrella      0.0010634077      g__Orrella</code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_1_2, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                          baseMean log2FoldChange     lfcSE      stat
g__Arsenophonus           48.58743      1.1561513 0.3497172  3.305960
g__Massilia              172.70323     -1.2511250 0.2667741 -4.689830
g__Dictyoglomus           42.00494      1.5362721 0.3573907  4.298579
g__Cetia                  26.75105      1.7835992 0.4617477  3.862714
g__Ochrobactrum          130.21463     -1.4130412 0.3917376 -3.607111
g__Orrella                44.93631     -3.0819286 0.6977098 -4.417207
g__Sulfurimonas          111.24233      0.8398344 0.2885532  2.910501
g__Methylovirgula         10.75281     -2.2143084 0.6527240 -3.392412
g__Candidatus_Nardonella  31.46252      1.7007462 0.4281576  3.972244
g__Pantoea               270.51401     -1.2001352 0.3383904 -3.546599
g__Serratia              372.55986     -1.1619336 0.3263009 -3.560927
g__Salipiger              11.25125     -1.8278985 0.5795822 -3.153821
g__Azotobacter            24.11827     -1.3752698 0.3963851 -3.469530
g__Salmonella            794.72429      1.7763798 0.5530651  3.211882
g__Liberibacter           60.66628      1.1867133 0.3415184  3.474815
                               pvalue         padj                    genus
g__Arsenophonus          9.465155e-04 0.0195744662          g__Arsenophonus
g__Massilia              2.734322e-06 0.0004860202              g__Massilia
g__Dictyoglomus          1.718967e-05 0.0015997133          g__Dictyoglomus
g__Cetia                 1.121344e-04 0.0053860667                 g__Cetia
g__Ochrobactrum          3.096247e-04 0.0092206240          g__Ochrobactrum
g__Orrella               9.998460e-06 0.0010634077               g__Orrella
g__Sulfurimonas          3.608501e-03 0.0478371125          g__Sulfurimonas
g__Methylovirgula        6.928027e-04 0.0158705100        g__Methylovirgula
g__Candidatus_Nardonella 7.119880e-05 0.0044172924 g__Candidatus_Nardonella
g__Pantoea               3.902375e-04 0.0104802196               g__Pantoea
g__Serratia              3.695480e-04 0.0103822075              g__Serratia
g__Salipiger             1.611479e-03 0.0266610196             g__Salipiger
g__Azotobacter           5.213702e-04 0.0131579712           g__Azotobacter
g__Salmonella            1.318684e-03 0.0239453798            g__Salmonella
g__Liberibacter          5.112052e-04 0.0131238719          g__Liberibacter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(sig_res_stage_1_2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 40</code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 2 to 3</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_2_3 <span class="ot">=</span> significant_stage_2_3[<span class="fu">order</span>(significant_stage_2_3<span class="sc">$</span>padj),]</span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage_2_3<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage_2_3)</span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_2_3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE      stat       pvalue
g__Schaalia        7862.70798      -4.005672 0.6993562 -5.727656 1.018277e-08
g__Aeromicrobium     57.60032      -2.766741 0.5119240 -5.404593 6.495570e-08
g__Cryobacterium     28.79334      -2.313397 0.4397841 -5.260302 1.438188e-07
g__Brachybacterium  163.10607      -2.669830 0.5208725 -5.125687 2.964546e-07
g__Cellulomonas     109.35396      -2.705990 0.5443962 -4.970626 6.673725e-07
g__Xanthomonas      294.27326      -1.904727 0.4120628 -4.622420 3.792892e-06
                           padj              genus
g__Schaalia        1.516215e-05        g__Schaalia
g__Aeromicrobium   4.835952e-05   g__Aeromicrobium
g__Cryobacterium   7.138206e-05   g__Cryobacterium
g__Brachybacterium 1.103552e-04 g__Brachybacterium
g__Cellulomonas    1.987435e-04    g__Cellulomonas
g__Xanthomonas     4.860202e-04     g__Xanthomonas</code></pre>
</div>
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_2_3, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      baseMean log2FoldChange     lfcSE      stat       pvalue
g__Schaalia         7862.70798      -4.005672 0.6993562 -5.727656 1.018277e-08
g__Aeromicrobium      57.60032      -2.766741 0.5119240 -5.404593 6.495570e-08
g__Cryobacterium      28.79334      -2.313397 0.4397841 -5.260302 1.438188e-07
g__Brachybacterium   163.10607      -2.669830 0.5208725 -5.125687 2.964546e-07
g__Cellulomonas      109.35396      -2.705990 0.5443962 -4.970626 6.673725e-07
g__Xanthomonas       294.27326      -1.904727 0.4120628 -4.622420 3.792892e-06
g__Pseudomonas      3335.06812      -1.764557 0.3717582 -4.746518 2.069486e-06
g__Massilia          172.70323      -1.251125 0.2667741 -4.689830 2.734322e-06
g__Allokutzneria      14.10462      -2.180526 0.4689454 -4.649851 3.321746e-06
g__Corynebacterium  3248.23979      -2.488681 0.5342191 -4.658541 3.184587e-06
g__Rothia          22447.70119      -4.621913 0.9790636 -4.720749 2.349782e-06
g__Microbacterium    411.34235      -2.148616 0.4654972 -4.615745 3.916885e-06
g__Georgenia          41.07669      -2.213157 0.4923847 -4.494772 6.964453e-06
g__Orrella            44.93631      -3.081929 0.6977098 -4.417207 9.998460e-06
g__Beutenbergia       12.96593      -2.599389 0.5962120 -4.359839 1.301581e-05
                           padj              genus
g__Schaalia        1.516215e-05        g__Schaalia
g__Aeromicrobium   4.835952e-05   g__Aeromicrobium
g__Cryobacterium   7.138206e-05   g__Cryobacterium
g__Brachybacterium 1.103552e-04 g__Brachybacterium
g__Cellulomonas    1.987435e-04    g__Cellulomonas
g__Xanthomonas     4.860202e-04     g__Xanthomonas
g__Pseudomonas     4.860202e-04     g__Pseudomonas
g__Massilia        4.860202e-04        g__Massilia
g__Allokutzneria   4.860202e-04   g__Allokutzneria
g__Corynebacterium 4.860202e-04 g__Corynebacterium
g__Rothia          4.860202e-04          g__Rothia
g__Microbacterium  4.860202e-04  g__Microbacterium
g__Georgenia       7.976977e-04       g__Georgenia
g__Orrella         1.063408e-03         g__Orrella
g__Beutenbergia    1.292036e-03    g__Beutenbergia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(sig_res_stage_2_3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 120</code></pre>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 3 to 4 </span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4 <span class="ot">=</span> significant_stage_2_3[<span class="fu">order</span>(significant_stage_3_4<span class="sc">$</span>padj),]</span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage_3_4)</span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_3_4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                baseMean log2FoldChange     lfcSE      stat       pvalue
g__Thermovibrio 19.47635      -1.494045 0.4586414 -3.257544 1.123807e-03
g__Thermocrinis 18.29961       1.306161 0.3646499  3.581958 3.410283e-04
g__Dictyoglomus 42.00494       1.536272 0.3573907  4.298579 1.718967e-05
                       padj           genus
g__Thermovibrio 0.022017740 g__Thermovibrio
g__Thermocrinis 0.009765215 g__Thermocrinis
g__Dictyoglomus 0.001599713 g__Dictyoglomus</code></pre>
</div>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_3_4, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                baseMean log2FoldChange     lfcSE      stat       pvalue
g__Thermovibrio 19.47635      -1.494045 0.4586414 -3.257544 1.123807e-03
g__Thermocrinis 18.29961       1.306161 0.3646499  3.581958 3.410283e-04
g__Dictyoglomus 42.00494       1.536272 0.3573907  4.298579 1.718967e-05
                       padj           genus
g__Thermovibrio 0.022017740 g__Thermovibrio
g__Thermocrinis 0.009765215 g__Thermocrinis
g__Dictyoglomus 0.001599713 g__Dictyoglomus</code></pre>
</div>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(sig_res_stage_3_4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3</code></pre>
</div>
</div>
</section>
<section id="look-for-highly-abundant-significant-ones" class="level2">
<h2 class="anchored" data-anchor-id="look-for-highly-abundant-significant-ones">5.5 Look for highly abundant significant ones</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyRanks</span>(tse_metaphlan_sal_swb)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make an altExp and matrix for Genus</span></span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse_metaphlan_sal_swb,<span class="st">"Genus"</span>) <span class="ot">=</span> <span class="fu">agglomerateByRank</span>(tse_metaphlan_sal_swb,<span class="st">"Genus"</span>)</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="co"># make a dataframe of relative abundance </span></span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>relabundance_df_Genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan_sal_swb, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>))</span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-7"><a href="#cb146-7" aria-hidden="true" tabindex="-1"></a><span class="co"># make a matric of relative abundance </span></span>
<span id="cb146-8"><a href="#cb146-8" aria-hidden="true" tabindex="-1"></a>relabundance_matrix_Genus <span class="ot">=</span> <span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan_sal_swb, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>)</span>
<span id="cb146-9"><a href="#cb146-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-10"><a href="#cb146-10" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the total relative abundance of each Genus (row sums)</span></span>
<span id="cb146-11"><a href="#cb146-11" aria-hidden="true" tabindex="-1"></a>total_relabundance_Genus <span class="ot">=</span> <span class="fu">rowSums</span>(relabundance_matrix_Genus)</span>
<span id="cb146-12"><a href="#cb146-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-13"><a href="#cb146-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the top highly abundant genera based on relative abundance </span></span>
<span id="cb146-14"><a href="#cb146-14" aria-hidden="true" tabindex="-1"></a>top_Genus_numbers_basic <span class="ot">=</span> <span class="fu">sort</span>(total_relabundance_Genus, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb146-15"><a href="#cb146-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-16"><a href="#cb146-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Make into dataframe</span></span>
<span id="cb146-17"><a href="#cb146-17" aria-hidden="true" tabindex="-1"></a>top_Genus_numbers_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(top_Genus_numbers_basic)</span>
<span id="cb146-18"><a href="#cb146-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-19"><a href="#cb146-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Make into tibble</span></span>
<span id="cb146-20"><a href="#cb146-20" aria-hidden="true" tabindex="-1"></a>top_Genus_numbers <span class="ot">=</span> <span class="fu">as.tibble</span>(top_Genus_numbers_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `as.tibble()` was deprecated in tibble 2.0.0.
ℹ Please use `as_tibble()` instead.
ℹ The signature and semantics have changed, see `?as_tibble`.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename genera to remove any higher taxonomic names (a quirk of the metaphaln style)</span></span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(top_Genus_numbers_df) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(top_Genus_numbers_df))</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get percenatge by dividing by total number of samples (21) and * by 100</span></span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a>top_Genus_pc <span class="ot">=</span> top_Genus_numbers <span class="sc">%&gt;%</span> </span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">top_Genus_percentage =</span> (top_Genus_numbers_basic<span class="sc">/</span><span class="fu">length</span>(<span class="fu">colnames</span>(tse_metaphlan_sal_swb))) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">top_Genus =</span> <span class="fu">rownames</span>(top_Genus_numbers_df))</span>
<span id="cb148-8"><a href="#cb148-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-9"><a href="#cb148-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only the top 20 genera by relative abundance </span></span>
<span id="cb148-10"><a href="#cb148-10" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_20_Genus <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus <span class="sc">%in%</span> top_Genus, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb148-11"><a href="#cb148-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-12"><a href="#cb148-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only the top genera with a relative abundance above 1%</span></span>
<span id="cb148-13"><a href="#cb148-13" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_Genus_above_0<span class="fl">.1</span> <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus_percentage <span class="sc">&gt;</span> <span class="fl">0.1</span>, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb148-14"><a href="#cb148-14" aria-hidden="true" tabindex="-1"></a>top_Genus_above_0<span class="fl">.1</span> <span class="ot">=</span> <span class="fu">unique</span>(top_Genus_pc<span class="sc">$</span>top_Genus_above_0<span class="fl">.1</span>)</span>
<span id="cb148-15"><a href="#cb148-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-16"><a href="#cb148-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Check significant genera </span></span>
<span id="cb148-17"><a href="#cb148-17" aria-hidden="true" tabindex="-1"></a>sig_res_stage<span class="sc">$</span>genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Desulfomicrobium"  "g__Propionibacterium"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a>sig_res_stage<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab <span class="ot">=</span> <span class="fu">ifelse</span>(sig_res_stage<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_0<span class="fl">.1</span>, sig_res_stage<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "-other" "-other"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check which genera are both signifiantly different and highly abundant </span></span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sig_res_stage<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "-other"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results  and look for highly abundant ones </span></span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage <span class="ot">=</span> significant_stage[<span class="fu">order</span>(significant_stage<span class="sc">$</span>padj),]</span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage)</span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE     stat       pvalue
g__Desulfomicrobium  403.8747       7.460098 1.3128254 5.682476 1.327588e-08
g__Propionibacterium 195.0813       4.623263 0.9852933 4.692270 2.701900e-06
                             padj                genus
g__Desulfomicrobium  2.394969e-05  g__Desulfomicrobium
g__Propionibacterium 2.437113e-03 g__Propionibacterium</code></pre>
</div>
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE     stat       pvalue
g__Desulfomicrobium  403.8747       7.460098 1.3128254 5.682476 1.327588e-08
g__Propionibacterium 195.0813       4.623263 0.9852933 4.692270 2.701900e-06
                             padj                genus
g__Desulfomicrobium  2.394969e-05  g__Desulfomicrobium
g__Propionibacterium 2.437113e-03 g__Propionibacterium</code></pre>
</div>
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 1 to 2 </span></span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_1_2 <span class="ot">=</span> significant_stage_2_3[<span class="fu">order</span>(significant_stage_1_2<span class="sc">$</span>padj),]</span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage_1_2<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage_1_2)</span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_1_2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 baseMean log2FoldChange     lfcSE      stat       pvalue
g__Arsenophonus  48.58743       1.156151 0.3497172  3.305960 9.465155e-04
g__Massilia     172.70323      -1.251125 0.2667741 -4.689830 2.734322e-06
g__Dictyoglomus  42.00494       1.536272 0.3573907  4.298579 1.718967e-05
g__Cetia         26.75105       1.783599 0.4617477  3.862714 1.121344e-04
g__Ochrobactrum 130.21463      -1.413041 0.3917376 -3.607111 3.096247e-04
g__Orrella       44.93631      -3.081929 0.6977098 -4.417207 9.998460e-06
                        padj           genus
g__Arsenophonus 0.0195744662 g__Arsenophonus
g__Massilia     0.0004860202     g__Massilia
g__Dictyoglomus 0.0015997133 g__Dictyoglomus
g__Cetia        0.0053860667        g__Cetia
g__Ochrobactrum 0.0092206240 g__Ochrobactrum
g__Orrella      0.0010634077      g__Orrella</code></pre>
</div>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_1_2, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                          baseMean log2FoldChange     lfcSE      stat
g__Arsenophonus           48.58743      1.1561513 0.3497172  3.305960
g__Massilia              172.70323     -1.2511250 0.2667741 -4.689830
g__Dictyoglomus           42.00494      1.5362721 0.3573907  4.298579
g__Cetia                  26.75105      1.7835992 0.4617477  3.862714
g__Ochrobactrum          130.21463     -1.4130412 0.3917376 -3.607111
g__Orrella                44.93631     -3.0819286 0.6977098 -4.417207
g__Sulfurimonas          111.24233      0.8398344 0.2885532  2.910501
g__Methylovirgula         10.75281     -2.2143084 0.6527240 -3.392412
g__Candidatus_Nardonella  31.46252      1.7007462 0.4281576  3.972244
g__Pantoea               270.51401     -1.2001352 0.3383904 -3.546599
g__Serratia              372.55986     -1.1619336 0.3263009 -3.560927
g__Salipiger              11.25125     -1.8278985 0.5795822 -3.153821
g__Azotobacter            24.11827     -1.3752698 0.3963851 -3.469530
g__Salmonella            794.72429      1.7763798 0.5530651  3.211882
g__Liberibacter           60.66628      1.1867133 0.3415184  3.474815
                               pvalue         padj                    genus
g__Arsenophonus          9.465155e-04 0.0195744662          g__Arsenophonus
g__Massilia              2.734322e-06 0.0004860202              g__Massilia
g__Dictyoglomus          1.718967e-05 0.0015997133          g__Dictyoglomus
g__Cetia                 1.121344e-04 0.0053860667                 g__Cetia
g__Ochrobactrum          3.096247e-04 0.0092206240          g__Ochrobactrum
g__Orrella               9.998460e-06 0.0010634077               g__Orrella
g__Sulfurimonas          3.608501e-03 0.0478371125          g__Sulfurimonas
g__Methylovirgula        6.928027e-04 0.0158705100        g__Methylovirgula
g__Candidatus_Nardonella 7.119880e-05 0.0044172924 g__Candidatus_Nardonella
g__Pantoea               3.902375e-04 0.0104802196               g__Pantoea
g__Serratia              3.695480e-04 0.0103822075              g__Serratia
g__Salipiger             1.611479e-03 0.0266610196             g__Salipiger
g__Azotobacter           5.213702e-04 0.0131579712           g__Azotobacter
g__Salmonella            1.318684e-03 0.0239453798            g__Salmonella
g__Liberibacter          5.112052e-04 0.0131238719          g__Liberibacter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(sig_res_stage_1_2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 40</code></pre>
</div>
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 1 to 2 and find out which are highly abundant (above 0.1%)</span></span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_1_2<span class="sc">$</span>genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "g__Arsenophonus"                              
 [2] "g__Massilia"                                  
 [3] "g__Dictyoglomus"                              
 [4] "g__Cetia"                                     
 [5] "g__Ochrobactrum"                              
 [6] "g__Orrella"                                   
 [7] "g__Sulfurimonas"                              
 [8] "g__Methylovirgula"                            
 [9] "g__Candidatus_Nardonella"                     
[10] "g__Pantoea"                                   
[11] "g__Serratia"                                  
[12] "g__Salipiger"                                 
[13] "g__Azotobacter"                               
[14] "g__Salmonella"                                
[15] "g__Liberibacter"                              
[16] "g__Thermodesulfovibrio"                       
[17] "g__Methylobacillus"                           
[18] "g__Thermomonas"                               
[19] "g__Arcobacter"                                
[20] "g__Thermovibrio"                              
[21] "g__Xanthomonas"                               
[22] "g__Methyloversatilis"                         
[23] "g__Stenotrophomonas"                          
[24] "g__Orientia"                                  
[25] "g__Francisella"                               
[26] "g__Pseudomonas"                               
[27] "g__Thermocrinis"                              
[28] "g__Rickettsia"                                
[29] "g__Komagataeibacter"                          
[30] "g__Candidatus_Endolissoclinum"                
[31] "g__Sphingobium"                               
[32] "g__Paracoccus"                                
[33] "g__Lonsdalea"                                 
[34] "g__Candidatus_Moranella"                      
[35] "NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA"
[36] "g__Caminibacter"                              
[37] "g__Halomonas"                                 
[38] "g__Luteimonas"                                
[39] "g__Lysobacter"                                
[40] "g__Candidatus_Purcelliella"                   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>sig_res_stage_1_2<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab <span class="ot">=</span> <span class="fu">ifelse</span>(sig_res_stage_1_2<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_0<span class="fl">.1</span>, sig_res_stage_1_2<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sig_res_stage_1_2<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "-other"         "g__Pseudomonas"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 2 to 3</span></span>
<span id="cb168-2"><a href="#cb168-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_2_3 <span class="ot">=</span> significant_stage_2_3[<span class="fu">order</span>(significant_stage_2_3<span class="sc">$</span>padj),]</span>
<span id="cb168-3"><a href="#cb168-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage_2_3<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage_2_3)</span>
<span id="cb168-4"><a href="#cb168-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_2_3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE      stat       pvalue
g__Schaalia        7862.70798      -4.005672 0.6993562 -5.727656 1.018277e-08
g__Aeromicrobium     57.60032      -2.766741 0.5119240 -5.404593 6.495570e-08
g__Cryobacterium     28.79334      -2.313397 0.4397841 -5.260302 1.438188e-07
g__Brachybacterium  163.10607      -2.669830 0.5208725 -5.125687 2.964546e-07
g__Cellulomonas     109.35396      -2.705990 0.5443962 -4.970626 6.673725e-07
g__Xanthomonas      294.27326      -1.904727 0.4120628 -4.622420 3.792892e-06
                           padj              genus
g__Schaalia        1.516215e-05        g__Schaalia
g__Aeromicrobium   4.835952e-05   g__Aeromicrobium
g__Cryobacterium   7.138206e-05   g__Cryobacterium
g__Brachybacterium 1.103552e-04 g__Brachybacterium
g__Cellulomonas    1.987435e-04    g__Cellulomonas
g__Xanthomonas     4.860202e-04     g__Xanthomonas</code></pre>
</div>
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_2_3, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      baseMean log2FoldChange     lfcSE      stat       pvalue
g__Schaalia         7862.70798      -4.005672 0.6993562 -5.727656 1.018277e-08
g__Aeromicrobium      57.60032      -2.766741 0.5119240 -5.404593 6.495570e-08
g__Cryobacterium      28.79334      -2.313397 0.4397841 -5.260302 1.438188e-07
g__Brachybacterium   163.10607      -2.669830 0.5208725 -5.125687 2.964546e-07
g__Cellulomonas      109.35396      -2.705990 0.5443962 -4.970626 6.673725e-07
g__Xanthomonas       294.27326      -1.904727 0.4120628 -4.622420 3.792892e-06
g__Pseudomonas      3335.06812      -1.764557 0.3717582 -4.746518 2.069486e-06
g__Massilia          172.70323      -1.251125 0.2667741 -4.689830 2.734322e-06
g__Allokutzneria      14.10462      -2.180526 0.4689454 -4.649851 3.321746e-06
g__Corynebacterium  3248.23979      -2.488681 0.5342191 -4.658541 3.184587e-06
g__Rothia          22447.70119      -4.621913 0.9790636 -4.720749 2.349782e-06
g__Microbacterium    411.34235      -2.148616 0.4654972 -4.615745 3.916885e-06
g__Georgenia          41.07669      -2.213157 0.4923847 -4.494772 6.964453e-06
g__Orrella            44.93631      -3.081929 0.6977098 -4.417207 9.998460e-06
g__Beutenbergia       12.96593      -2.599389 0.5962120 -4.359839 1.301581e-05
                           padj              genus
g__Schaalia        1.516215e-05        g__Schaalia
g__Aeromicrobium   4.835952e-05   g__Aeromicrobium
g__Cryobacterium   7.138206e-05   g__Cryobacterium
g__Brachybacterium 1.103552e-04 g__Brachybacterium
g__Cellulomonas    1.987435e-04    g__Cellulomonas
g__Xanthomonas     4.860202e-04     g__Xanthomonas
g__Pseudomonas     4.860202e-04     g__Pseudomonas
g__Massilia        4.860202e-04        g__Massilia
g__Allokutzneria   4.860202e-04   g__Allokutzneria
g__Corynebacterium 4.860202e-04 g__Corynebacterium
g__Rothia          4.860202e-04          g__Rothia
g__Microbacterium  4.860202e-04  g__Microbacterium
g__Georgenia       7.976977e-04       g__Georgenia
g__Orrella         1.063408e-03         g__Orrella
g__Beutenbergia    1.292036e-03    g__Beutenbergia</code></pre>
</div>
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(sig_res_stage_2_3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 120</code></pre>
</div>
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 2 to 3 and find out which are highly abundant (above 0.1%)</span></span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_2_3<span class="sc">$</span>genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] "g__Schaalia"                                                                
  [2] "g__Aeromicrobium"                                                           
  [3] "g__Cryobacterium"                                                           
  [4] "g__Brachybacterium"                                                         
  [5] "g__Cellulomonas"                                                            
  [6] "g__Xanthomonas"                                                             
  [7] "g__Pseudomonas"                                                             
  [8] "g__Massilia"                                                                
  [9] "g__Allokutzneria"                                                           
 [10] "g__Corynebacterium"                                                         
 [11] "g__Rothia"                                                                  
 [12] "g__Microbacterium"                                                          
 [13] "g__Georgenia"                                                               
 [14] "g__Orrella"                                                                 
 [15] "g__Beutenbergia"                                                            
 [16] "g__Dictyoglomus"                                                            
 [17] "g__Orientia"                                                                
 [18] "g__Stenotrophomonas"                                                        
 [19] "g__Paeniclostridium"                                                        
 [20] "g__Rickettsia"                                                              
 [21] "NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Peptostreptococcaceae_NA"
 [22] "g__Lysobacter"                                                              
 [23] "g__Amycolatopsis"                                                           
 [24] "g__Candidatus_Nardonella"                                                   
 [25] "g__Ruania"                                                                  
 [26] "g__Mycolicibacterium"                                                       
 [27] "g__Xylanimonas"                                                             
 [28] "g__Paraclostridium"                                                         
 [29] "g__Methyloversatilis"                                                       
 [30] "g__Sextaecvirus"                                                            
 [31] "g__Cetia"                                                                   
 [32] "g__Sphingobium"                                                             
 [33] "g__Isoptericola"                                                            
 [34] "g__Curtobacterium"                                                          
 [35] "g__Candidatus_Nitrosopelagicus"                                             
 [36] "g__Candidatus_Moranella"                                                    
 [37] "g__Prochlorococcus"                                                         
 [38] "g__Cutibacterium"                                                           
 [39] "g__Streptococcus"                                                           
 [40] "g__Leucobacter"                                                             
 [41] "g__Achromobacter"                                                           
 [42] "g__Solibacillus"                                                            
 [43] "g__Actinobaculum"                                                           
 [44] "g__Kribbella"                                                               
 [45] "g__Streptomyces"                                                            
 [46] "g__Sanguibacter"                                                            
 [47] "g__Psychromicrobium"                                                        
 [48] "g__Pseudonocardia"                                                          
 [49] "g__Nocardioides"                                                            
 [50] "g__Ochrobactrum"                                                            
 [51] "g__Thermocrinis"                                                            
 [52] "g__Agromyces"                                                               
 [53] "g__Serratia"                                                                
 [54] "g__Halomonas"                                                               
 [55] "g__Pantoea"                                                                 
 [56] "g__Cellulosimicrobium"                                                      
 [57] "g__Thermomonas"                                                             
 [58] "g__Liberibacter"                                                            
 [59] "g__Azotobacter"                                                             
 [60] "g__Dermabacter"                                                             
 [61] "g__Hathewaya"                                                               
 [62] "g__Candidatus_Nitrosocosmicus"                                              
 [63] "g__Luteimonas"                                                              
 [64] "g__Methanococcus"                                                           
 [65] "g__Methylovirgula"                                                          
 [66] "g__Dietzia"                                                                 
 [67] "g__Rhodococcus"                                                             
 [68] "g__Thermodesulfovibrio"                                                     
 [69] "g__Arthrobacter"                                                            
 [70] "NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA"                              
 [71] "g__Arsenophonus"                                                            
 [72] "g__Bordetella"                                                              
 [73] "g__Delftia"                                                                 
 [74] "g__Pseudarthrobacter"                                                       
 [75] "g__Francisella"                                                             
 [76] "g__Thermovibrio"                                                            
 [77] "g__Komagataeibacter"                                                        
 [78] "g__Paracoccus"                                                              
 [79] "g__Caminibacter"                                                            
 [80] "g__Pseudanabaena"                                                           
 [81] "g__Salmonella"                                                              
 [82] "g__Zhihengliuella"                                                          
 [83] "g__Candidatus_Purcelliella"                                                 
 [84] "g__Micromonospora"                                                          
 [85] "g__Actinopolymorpha"                                                        
 [86] "g__Miniimonas"                                                              
 [87] "g__Peptoniphilus"                                                           
 [88] "g__Jiangella"                                                               
 [89] "g__Clostridioides"                                                          
 [90] "g__Salipiger"                                                               
 [91] "g__Sporanaerobacter"                                                        
 [92] "g__Cepunavirus"                                                             
 [93] "g__Cupriavidus"                                                             
 [94] "NA_p__Firmicutes_c__Tissierellia_o__Tissierellales_f__Tissierellaceae_NA"   
 [95] "g__Nocardia"                                                                
 [96] "g__Oerskovia"                                                               
 [97] "g__Actinoalloteichus"                                                       
 [98] "g__Enterococcus"                                                            
 [99] "g__Peptoclostridium"                                                        
[100] "g__Lonsdalea"                                                               
[101] "g__Candidatus_Nitrosomarinus"                                               
[102] "NA_NA_NA_NA_f__Marseilleviridae_NA"                                         
[103] "g__Candidatus_Endolissoclinum"                                              
[104] "g__Chromobacterium"                                                         
[105] "g__Glutamicibacter"                                                         
[106] "g__Vagococcus"                                                              
[107] "g__Gordonia"                                                                
[108] "g__Parvimonas"                                                              
[109] "NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Lachnospiraceae_NA"      
[110] "g__Sulfurimonas"                                                            
[111] "g__Methylobacillus"                                                         
[112] "g__Variovorax"                                                              
[113] "g__Blastococcus"                                                            
[114] "g__Phycicoccus"                                                             
[115] "g__Microterricola"                                                          
[116] "g__Acetohalobium"                                                           
[117] "g__Caldanaerobacter"                                                        
[118] "g__Methanobacterium"                                                        
[119] "g__Empedobacter"                                                            
[120] "g__Arcobacter"                                                              </code></pre>
</div>
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a>sig_res_stage_2_3<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab <span class="ot">=</span> <span class="fu">ifelse</span>(sig_res_stage_2_3<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_0<span class="fl">.1</span>, sig_res_stage_2_3<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sig_res_stage_2_3<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Schaalia"                                                          
[2] "-other"                                                               
[3] "g__Pseudomonas"                                                       
[4] "g__Corynebacterium"                                                   
[5] "g__Rothia"                                                            
[6] "g__Streptococcus"                                                     
[7] "g__Parvimonas"                                                        
[8] "NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Lachnospiraceae_NA"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 3 to 4 </span></span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4 <span class="ot">=</span> significant_stage_2_3[<span class="fu">order</span>(significant_stage_3_4<span class="sc">$</span>padj),]</span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_stage_3_4)</span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_3_4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                baseMean log2FoldChange     lfcSE      stat       pvalue
g__Thermovibrio 19.47635      -1.494045 0.4586414 -3.257544 1.123807e-03
g__Thermocrinis 18.29961       1.306161 0.3646499  3.581958 3.410283e-04
g__Dictyoglomus 42.00494       1.536272 0.3573907  4.298579 1.718967e-05
                       padj           genus
g__Thermovibrio 0.022017740 g__Thermovibrio
g__Thermocrinis 0.009765215 g__Thermocrinis
g__Dictyoglomus 0.001599713 g__Dictyoglomus</code></pre>
</div>
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_stage_3_4, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                baseMean log2FoldChange     lfcSE      stat       pvalue
g__Thermovibrio 19.47635      -1.494045 0.4586414 -3.257544 1.123807e-03
g__Thermocrinis 18.29961       1.306161 0.3646499  3.581958 3.410283e-04
g__Dictyoglomus 42.00494       1.536272 0.3573907  4.298579 1.718967e-05
                       padj           genus
g__Thermovibrio 0.022017740 g__Thermovibrio
g__Thermocrinis 0.009765215 g__Thermocrinis
g__Dictyoglomus 0.001599713 g__Dictyoglomus</code></pre>
</div>
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(sig_res_stage_3_4)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results for stages 3 to 4 and find out which are highly abundant (above 0.1%)</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4<span class="sc">$</span>genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Thermovibrio" "g__Thermocrinis" "g__Dictyoglomus"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb186-1"><a href="#cb186-1" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab <span class="ot">=</span> <span class="fu">ifelse</span>(sig_res_stage_3_4<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_0<span class="fl">.1</span>, sig_res_stage_3_4<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb186-2"><a href="#cb186-2" aria-hidden="true" tabindex="-1"></a>sig_res_stage_3_4<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "-other" "-other" "-other"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb188"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb188-1"><a href="#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sig_res_stage_3_4<span class="sc">$</span>genera_above_0<span class="fl">.1</span>pc_relab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "-other"</code></pre>
</div>
</div>
</section>
<section id="plot-counts-of-genera-across-noma-stages" class="level2">
<h2 class="anchored" data-anchor-id="plot-counts-of-genera-across-noma-stages">5.6 Plot counts of genera across noma stages</h2>
<p>First we create a function</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a>plotCountsGGanysig <span class="ot">=</span> <span class="cf">function</span>(dds, gene, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">normalized =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">transform =</span> <span class="cn">TRUE</span>, main, <span class="at">xlab =</span> <span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">FALSE</span>,  </span>
<span id="cb190-3"><a href="#cb190-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">replaced =</span> <span class="cn">FALSE</span>, pc, <span class="at">plot =</span> <span class="st">"point"</span>, <span class="at">text =</span> <span class="cn">TRUE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, ...) {  </span>
<span id="cb190-4"><a href="#cb190-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check input gene validity</span></span>
<span id="cb190-5"><a href="#cb190-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(gene) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> (<span class="fu">is.character</span>(gene) <span class="sc">|</span> (<span class="fu">is.numeric</span>(gene) <span class="sc">&amp;</span>  </span>
<span id="cb190-6"><a href="#cb190-6" aria-hidden="true" tabindex="-1"></a>                                                         (gene <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> gene <span class="sc">&lt;=</span> <span class="fu">nrow</span>(dds)))))  </span>
<span id="cb190-7"><a href="#cb190-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-8"><a href="#cb190-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if all intgroup columns exist in colData</span></span>
<span id="cb190-9"><a href="#cb190-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(intgroup <span class="sc">%in%</span> <span class="fu">names</span>(<span class="fu">colData</span>(dds))))  </span>
<span id="cb190-10"><a href="#cb190-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"all variables in 'intgroup' must be columns of colData"</span>)  </span>
<span id="cb190-11"><a href="#cb190-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-12"><a href="#cb190-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If not returning data, ensure intgroup variables are factors</span></span>
<span id="cb190-13"><a href="#cb190-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>returnData) {  </span>
<span id="cb190-14"><a href="#cb190-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">sapply</span>(intgroup, <span class="cf">function</span>(v) <span class="fu">is</span>(<span class="fu">colData</span>(dds)[[v]], <span class="st">"factor"</span>)))) {  </span>
<span id="cb190-15"><a href="#cb190-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">"all variables in 'intgroup' should be factors, or choose returnData=TRUE and plot manually"</span>)  </span>
<span id="cb190-16"><a href="#cb190-16" aria-hidden="true" tabindex="-1"></a>    }  </span>
<span id="cb190-17"><a href="#cb190-17" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb190-18"><a href="#cb190-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-19"><a href="#cb190-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set pseudo count if not provided</span></span>
<span id="cb190-20"><a href="#cb190-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">missing</span>(pc)) {  </span>
<span id="cb190-21"><a href="#cb190-21" aria-hidden="true" tabindex="-1"></a>    pc <span class="ot">=</span> <span class="cf">if</span> (transform) <span class="fl">0.5</span> <span class="cf">else</span> <span class="dv">0</span>  </span>
<span id="cb190-22"><a href="#cb190-22" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb190-23"><a href="#cb190-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-24"><a href="#cb190-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Estimate size factors if missing</span></span>
<span id="cb190-25"><a href="#cb190-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">sizeFactors</span>(dds)) <span class="sc">&amp;</span> <span class="fu">is.null</span>(<span class="fu">normalizationFactors</span>(dds))) {  </span>
<span id="cb190-26"><a href="#cb190-26" aria-hidden="true" tabindex="-1"></a>    dds <span class="ot">=</span> <span class="fu">estimateSizeFactors</span>(dds)  </span>
<span id="cb190-27"><a href="#cb190-27" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb190-28"><a href="#cb190-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-29"><a href="#cb190-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the counts for the gene</span></span>
<span id="cb190-30"><a href="#cb190-30" aria-hidden="true" tabindex="-1"></a>  cnts <span class="ot">=</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> normalized, <span class="at">replaced =</span> replaced)[gene, ]  </span>
<span id="cb190-31"><a href="#cb190-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-32"><a href="#cb190-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate grouping variable</span></span>
<span id="cb190-33"><a href="#cb190-33" aria-hidden="true" tabindex="-1"></a>  group <span class="ot">=</span> <span class="cf">if</span> (<span class="fu">length</span>(intgroup) <span class="sc">==</span> <span class="dv">1</span>) {  </span>
<span id="cb190-34"><a href="#cb190-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colData</span>(dds)[[intgroup]]  </span>
<span id="cb190-35"><a href="#cb190-35" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">length</span>(intgroup) <span class="sc">==</span> <span class="dv">2</span>) {  </span>
<span id="cb190-36"><a href="#cb190-36" aria-hidden="true" tabindex="-1"></a>    lvls <span class="ot">=</span> <span class="fu">as.vector</span>(<span class="fu">t</span>(<span class="fu">outer</span>(<span class="fu">levels</span>(<span class="fu">colData</span>(dds)[[intgroup[<span class="dv">1</span>]]]),  </span>
<span id="cb190-37"><a href="#cb190-37" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">levels</span>(<span class="fu">colData</span>(dds)[[intgroup[<span class="dv">2</span>]]]), <span class="cf">function</span>(x, y) <span class="fu">paste</span>(x, y, <span class="at">sep =</span> <span class="st">":"</span>))))  </span>
<span id="cb190-38"><a href="#cb190-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">droplevels</span>(<span class="fu">factor</span>(<span class="fu">apply</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds)[, intgroup, <span class="at">drop =</span> <span class="cn">FALSE</span>]), <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">":"</span>),  </span>
<span id="cb190-39"><a href="#cb190-39" aria-hidden="true" tabindex="-1"></a>                      <span class="at">levels =</span> lvls))  </span>
<span id="cb190-40"><a href="#cb190-40" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {  </span>
<span id="cb190-41"><a href="#cb190-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(<span class="fu">apply</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds)[, intgroup, <span class="at">drop =</span> <span class="cn">FALSE</span>]), <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">":"</span>))  </span>
<span id="cb190-42"><a href="#cb190-42" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb190-43"><a href="#cb190-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-44"><a href="#cb190-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the data frame with counts, group, and sample names</span></span>
<span id="cb190-45"><a href="#cb190-45" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">count =</span> cnts <span class="sc">+</span> pc, <span class="at">group =</span> group, <span class="at">sample =</span> <span class="fu">colnames</span>(dds), <span class="at">condition =</span> group)  </span>
<span id="cb190-46"><a href="#cb190-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-47"><a href="#cb190-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set log scale if necessary</span></span>
<span id="cb190-48"><a href="#cb190-48" aria-hidden="true" tabindex="-1"></a>  logxy <span class="ot">=</span> <span class="cf">if</span> (transform) <span class="st">"y"</span> <span class="cf">else</span> <span class="st">""</span>  </span>
<span id="cb190-49"><a href="#cb190-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-50"><a href="#cb190-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the plot title</span></span>
<span id="cb190-51"><a href="#cb190-51" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">missing</span>(main)) {  </span>
<span id="cb190-52"><a href="#cb190-52" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">=</span> <span class="cf">if</span> (<span class="fu">is.numeric</span>(gene)) {  </span>
<span id="cb190-53"><a href="#cb190-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(dds)[gene]  </span>
<span id="cb190-54"><a href="#cb190-54" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {  </span>
<span id="cb190-55"><a href="#cb190-55" aria-hidden="true" tabindex="-1"></a>      gene  </span>
<span id="cb190-56"><a href="#cb190-56" aria-hidden="true" tabindex="-1"></a>    }  </span>
<span id="cb190-57"><a href="#cb190-57" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb190-58"><a href="#cb190-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-59"><a href="#cb190-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the y-axis label based on normalization</span></span>
<span id="cb190-60"><a href="#cb190-60" aria-hidden="true" tabindex="-1"></a>  ylab <span class="ot">=</span> <span class="fu">ifelse</span>(normalized, <span class="st">"normalized count"</span>, <span class="st">"count"</span>)  </span>
<span id="cb190-61"><a href="#cb190-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-62"><a href="#cb190-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the data if requested</span></span>
<span id="cb190-63"><a href="#cb190-63" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (returnData)  </span>
<span id="cb190-64"><a href="#cb190-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">count =</span> data<span class="sc">$</span>count, <span class="fu">colData</span>(dds)[intgroup]))  </span>
<span id="cb190-65"><a href="#cb190-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-66"><a href="#cb190-66" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the base ggplot object with data and aesthetic mappings</span></span>
<span id="cb190-67"><a href="#cb190-67" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> count, <span class="at">label =</span> sample, <span class="at">color =</span> condition)) <span class="sc">+</span></span>
<span id="cb190-68"><a href="#cb190-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> xlab, <span class="at">y =</span> ylab, <span class="at">title =</span> main) <span class="sc">+</span>  <span class="co"># Labels and title</span></span>
<span id="cb190-69"><a href="#cb190-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="co"># Clean theme</span></span>
<span id="cb190-70"><a href="#cb190-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">ifelse</span>(transform, <span class="st">"log10"</span>, <span class="st">"identity"</span>)) <span class="sc">+</span>  <span class="co"># Apply log transformation if needed</span></span>
<span id="cb190-71"><a href="#cb190-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>)  <span class="co"># Optional: use color brewer for nice color scheme</span></span>
<span id="cb190-72"><a href="#cb190-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-73"><a href="#cb190-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select the type of plot based on the 'plot' argument</span></span>
<span id="cb190-74"><a href="#cb190-74" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"point"</span>) {</span>
<span id="cb190-75"><a href="#cb190-75" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb190-76"><a href="#cb190-76" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb190-77"><a href="#cb190-77" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"jitter"</span>) {</span>
<span id="cb190-78"><a href="#cb190-78" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">width =</span> <span class="fl">0.2</span>)</span>
<span id="cb190-79"><a href="#cb190-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb190-80"><a href="#cb190-80" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"bar"</span>) {</span>
<span id="cb190-81"><a href="#cb190-81" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"summary"</span>, <span class="at">fun =</span> <span class="st">"mean"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">width =</span> <span class="fl">0.7</span>) <span class="sc">+</span>  <span class="co"># Bar plot with whiskers</span></span>
<span id="cb190-82"><a href="#cb190-82" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_errorbar</span>(<span class="at">stat =</span> <span class="st">"summary"</span>, <span class="at">fun.data =</span> <span class="st">"mean_se"</span>, <span class="at">width =</span> <span class="fl">0.2</span>)</span>
<span id="cb190-83"><a href="#cb190-83" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"violin"</span>) {</span>
<span id="cb190-84"><a href="#cb190-84" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">width =</span> <span class="fl">0.2</span>)</span>
<span id="cb190-85"><a href="#cb190-85" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb190-86"><a href="#cb190-86" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"box"</span>) {</span>
<span id="cb190-87"><a href="#cb190-87" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_boxplot</span>()</span>
<span id="cb190-88"><a href="#cb190-88" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb190-89"><a href="#cb190-89" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb190-90"><a href="#cb190-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Invalid plot type. Choose from 'point', 'jitter', 'bar', 'violin', or 'box'."</span>)</span>
<span id="cb190-91"><a href="#cb190-91" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb190-92"><a href="#cb190-92" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-93"><a href="#cb190-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add significance annotation if requested</span></span>
<span id="cb190-94"><a href="#cb190-94" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (showSignificance) {</span>
<span id="cb190-95"><a href="#cb190-95" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get DESeq2 results for gene using Wald test and BH adjustment</span></span>
<span id="cb190-96"><a href="#cb190-96" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(intgroup, <span class="fu">levels</span>(group)[<span class="dv">1</span>], <span class="fu">levels</span>(group)[<span class="dv">2</span>]), <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb190-97"><a href="#cb190-97" aria-hidden="true" tabindex="-1"></a>    res_gene <span class="ot">=</span> res[gene, ]</span>
<span id="cb190-98"><a href="#cb190-98" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb190-99"><a href="#cb190-99" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check significance and add stars/annotations</span></span>
<span id="cb190-100"><a href="#cb190-100" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(res_gene<span class="sc">$</span>padj) <span class="sc">&amp;&amp;</span> res_gene<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb190-101"><a href="#cb190-101" aria-hidden="true" tabindex="-1"></a>      p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>, <span class="at">y =</span> <span class="fu">max</span>(data<span class="sc">$</span>count), <span class="at">label =</span> <span class="st">"*"</span>, <span class="at">size =</span> <span class="dv">8</span>)</span>
<span id="cb190-102"><a href="#cb190-102" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb190-103"><a href="#cb190-103" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb190-104"><a href="#cb190-104" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb190-105"><a href="#cb190-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb190-106"><a href="#cb190-106" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next we use the function to plot any highly abundant significantly different across noma stages</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a>Trep_stage <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_stage, <span class="at">gene=</span> <span class="st">"g__Treponema"</span>, <span class="at">intgroup =</span> <span class="st">"noma_stage_on_admission"</span>, <span class="at">plot =</span> <span class="st">"violin"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma---swab-vs-saliva_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Supplementary_Figure_1.png"</span>, <span class="at">plot =</span>  Trep_stage, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>