<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.55">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Richard Goodman">
<meta name="dcterms.date" content="2025-10-01">

<title>Noma Metagenomics: Short-read taxonomic based metagenomic analysis of noma vs healthy human saliva samples in R</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Noma_vs_healthy_files/libs/clipboard/clipboard.min.js"></script>
<script src="Noma_vs_healthy_files/libs/quarto-html/quarto.js"></script>
<script src="Noma_vs_healthy_files/libs/quarto-html/popper.min.js"></script>
<script src="Noma_vs_healthy_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Noma_vs_healthy_files/libs/quarto-html/anchor.min.js"></script>
<link href="Noma_vs_healthy_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Noma_vs_healthy_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Noma_vs_healthy_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Noma_vs_healthy_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Noma_vs_healthy_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Noma Metagenomics: Short-read taxonomic based metagenomic analysis of noma vs healthy human saliva samples in R</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Richard Goodman </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 1, 2025</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>This analysis was used in the short-read (Illumina) taxonomic based metagenomic analysis of noma vs healthy human saliva samples in R posted as a preprint on bioRxiv:</p>
<p><strong>Shotgun metagenomic analysis of the oral microbiomes of children with noma reveals a novel disease-associated organism</strong></p>
<p>Michael Olaleye, Angus M O’Ferrall, Richard N. Goodman, Deogracia W Kabila, Miriam Peters, Gregoire Falq, Joseph Samuel, Donal Doyle, Diana Gomez, Gbemisola Oloruntuyi, Shafiu Isah, Adeniyi S Adetunji, Elise N. Farley, Nicholas J Evans, Mark Sherlock, Adam P. Roberts, Mohana Amirtharajah, Stuart Ainsworth</p>
<p>bioRxiv 2025.06.24.661267; doi: https://doi.org/10.1101/2025.06.24.661267</p>
<section id="getting-started-in-r" class="level1">
<h1>1. Getting Started in R</h1>
<section id="installing-and-loading-packages" class="level2">
<h2 class="anchored" data-anchor-id="installing-and-loading-packages">1.1 Installing and Loading Packages</h2>
<p>Install all necessary packages into R.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure these are all installed as packages first </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Load necessary libraries</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(miaViz)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scater)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TreeSummarizedExperiment)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsignif)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gridExtra)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(randomForest)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(e1071)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pROC)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ROCR)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret) </span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="import-and-clean-data" class="level1">
<h1>2. Import and Clean Data</h1>
<p>We will be importing MetaPhlan style Bracken data</p>
<p>First we’ll write a function to import MetaPhlan style bracken data</p>
<section id="load-taxonomic-data" class="level2">
<h2 class="anchored" data-anchor-id="load-taxonomic-data">2.1 Load taxonomic data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>file_path <span class="ot">=</span>  <span class="st">"../data/noma_HMP_saliva_bracken_MetaPhlan_style_report_bacteria_noma_v_healthy_saliva_A1_A19_corrGB.txt"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sample_data_path <span class="ot">=</span> <span class="st">"../data/Samples_healthy_v_noma_saliva_A1_A19_corrGB.csv"</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_noma <span class="ot">=</span> <span class="fu">loadFromMetaphlan</span>(file_path)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Defining the TSE for the rest of the script</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>tse_metaphlan <span class="ot">=</span> tse_metaphlan_noma</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="add-metadata" class="level2">
<h2 class="anchored" data-anchor-id="add-metadata">2.2 Add metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>patient_metadata <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"../data/micro_study_metadata.xlsx"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>sample_to_patient <span class="ot">=</span> <span class="fu">read_excel</span>(<span class="st">"../data/sample_to_patient_A1_A40.xlsx"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(patient_metadata, sample_to_patient, <span class="at">by =</span> <span class="st">"respondent_id"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>metadata_2 <span class="ot">=</span> metadata <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample_name  <span class="sc">%in%</span> <span class="fu">colnames</span>(tse_metaphlan_noma))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>coldata <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">sample_name =</span> <span class="fu">colnames</span>(tse_metaphlan_noma))</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>metadata_3 <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(coldata, metadata_2, <span class="at">by =</span> <span class="st">"sample_name"</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with this information</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">=</span> <span class="fu">DataFrame</span>(metadata_3)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata_df) <span class="ot">=</span> metadata_3<span class="sc">$</span>sample_name</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>t_metadata_df <span class="ot">=</span> <span class="fu">t</span>(metadata_df)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="fu">ncol</span>(t_metadata_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 37</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add this DataFrame as colData to your TreeSummarizedExperiment object</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_noma) <span class="ot">=</span> metadata_df</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>tse_metaphlan <span class="ot">=</span> tse_metaphlan_noma</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="adding-diseased-or-healthy-condition-metadata" class="level2">
<h2 class="anchored" data-anchor-id="adding-diseased-or-healthy-condition-metadata">2.3 Adding diseased or healthy condition metadata</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assume your TreeSummarizedExperiment object is called `tse`</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the current sample names (column names)</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>sample_names <span class="ot">=</span> <span class="fu">colnames</span>(tse_metaphlan)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>sample_names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "A10"        "A11"        "A13"        "A14"        "A16"       
 [6] "A17"        "A18"        "A19"        "A1"         "A2"        
[11] "A3"         "A4"         "A5"         "A6"         "A7"        
[16] "A8"         "A9"         "DRR214959"  "DRR214960"  "DRR214961" 
[21] "DRR214962"  "DRR241310"  "SRR5892208" "SRR5892209" "SRR5892210"
[26] "SRR5892211" "SRR5892212" "SRR5892213" "SRR5892214" "SRR5892215"
[31] "SRR5892216" "SRR5892217" "SRS013942"  "SRS014468"  "SRS014692" 
[36] "SRS015055"  "SRS019120" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a vector indicating the condition (diseased or healthy)</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_1 to sample_17 are diseased, and sample_18 to sample_37 are healthy</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>condition <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"diseased"</span>, <span class="dv">17</span>), <span class="fu">rep</span>(<span class="st">"healthy"</span>, <span class="dv">20</span>))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with this information</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>sample_metadata_disease <span class="ot">=</span> <span class="fu">DataFrame</span>(<span class="at">condition =</span> condition)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_metadata_disease) <span class="ot">=</span> sample_names</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>sample_metadata_disease</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 37 rows and 1 column
            condition
          &lt;character&gt;
A10          diseased
A11          diseased
A13          diseased
A14          diseased
A16          diseased
...               ...
SRS013942     healthy
SRS014468     healthy
SRS014692     healthy
SRS015055     healthy
SRS019120     healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add this DataFrame as colData to your TreeSummarizedExperiment object</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan) <span class="ot">=</span> sample_metadata_disease</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that colData was added successfully</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan)), <span class="at">n =</span> <span class="dv">37</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           condition
A10         diseased
A11         diseased
A13         diseased
A14         diseased
A16         diseased
A17         diseased
A18         diseased
A19         diseased
A1          diseased
A2          diseased
A3          diseased
A4          diseased
A5          diseased
A6          diseased
A7          diseased
A8          diseased
A9          diseased
DRR214959    healthy
DRR214960    healthy
DRR214961    healthy
DRR214962    healthy
DRR241310    healthy
SRR5892208   healthy
SRR5892209   healthy
SRR5892210   healthy
SRR5892211   healthy
SRR5892212   healthy
SRR5892213   healthy
SRR5892214   healthy
SRR5892215   healthy
SRR5892216   healthy
SRR5892217   healthy
SRS013942    healthy
SRS014468    healthy
SRS014692    healthy
SRS015055    healthy
SRS019120    healthy</code></pre>
</div>
</div>
</section>
<section id="inspecting-the-data" class="level2">
<h2 class="anchored" data-anchor-id="inspecting-the-data">2.4 Inspecting the Data</h2>
<p>You can inspect the data with these functions.</p>
<p>The output is not printed here as it would be too large.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>(<span class="at">count =</span> <span class="fu">assays</span>(tse_metaphlan)[[<span class="dv">1</span>]])</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(tse_metaphlan))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(tse_metaphlan))</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">metadata</span>(tse_metaphlan))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="converting-tse-to-other-common-data-formats-e.g.-phyloseq" class="level2">
<h2 class="anchored" data-anchor-id="converting-tse-to-other-common-data-formats-e.g.-phyloseq">2.5 Converting TSE to other common data formats e.g.&nbsp;Phyloseq</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>phyloseq_obj <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="non-parametric-statistical-tests" class="level1">
<h1>3. Non-parametric statistical tests</h1>
<p>Non-parametric methods were used to determine the difference between samples based on the categorical variable of disease status. Non-parametric tests are used for metagenomic data due to the non-normal distribution of the data. We used the Bray-Curtis dissimilarity matrix for all these tests.</p>
<section id="preparing-the-data" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data">3.1 Preparing the data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># See above "Converting TSE to other common data formats e.g. Phyloseq"</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_noma <span class="ot">=</span> tse_metaphlan</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># make an assay for abundance</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_noma <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_metaphlan_noma, <span class="at">assay.type=</span><span class="st">"counts"</span>, <span class="at">method=</span><span class="st">"relabundance"</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyRanks</span>(tse_metaphlan_noma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_noma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 37 rows and 1 column
            condition
          &lt;character&gt;
A10          diseased
A11          diseased
A13          diseased
A14          diseased
A16          diseased
...               ...
SRS013942     healthy
SRS014468     healthy
SRS014692     healthy
SRS015055     healthy
SRS019120     healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make an altExp and matrix for order</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_noma_genus <span class="ot">=</span> <span class="fu">altExp</span>(tse_metaphlan_noma, <span class="st">"Genus"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Add metadata to colData </span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_noma_genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 37 rows and 0 columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_noma_genus) <span class="ot">=</span> sample_metadata_disease</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">=</span> <span class="fu">colData</span>(tse_metaphlan_noma_genus)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>metadata_noma_genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan_noma_genus))</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>metadata_noma_genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           condition
A10         diseased
A11         diseased
A13         diseased
A14         diseased
A16         diseased
A17         diseased
A18         diseased
A19         diseased
A1          diseased
A2          diseased
A3          diseased
A4          diseased
A5          diseased
A6          diseased
A7          diseased
A8          diseased
A9          diseased
DRR214959    healthy
DRR214960    healthy
DRR214961    healthy
DRR214962    healthy
DRR241310    healthy
SRR5892208   healthy
SRR5892209   healthy
SRR5892210   healthy
SRR5892211   healthy
SRR5892212   healthy
SRR5892213   healthy
SRR5892214   healthy
SRR5892215   healthy
SRR5892216   healthy
SRR5892217   healthy
SRS013942    healthy
SRS014468    healthy
SRS014692    healthy
SRS015055    healthy
SRS019120    healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># genus</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>phyloseq_noma <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_noma_genus)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>phyloseq_noma_esd <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(phyloseq_noma, <span class="cf">function</span>(x) <span class="fl">1E6</span> <span class="sc">*</span> x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ntaxa</span>(phyloseq_noma_esd) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1757</code></pre>
</div>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nsamples</span>(phyloseq_noma_esd) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 37</code></pre>
</div>
</div>
</section>
<section id="permanova-across-entire-dataset" class="level2">
<h2 class="anchored" data-anchor-id="permanova-across-entire-dataset">3.1 Permanova across entire dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate bray curtis distance matrix on main variables </span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>noma.bray <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_noma_esd, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>sample.noma.df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phyloseq_noma_esd))</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>permanova_all <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(noma.bray <span class="sc">~</span> condition , <span class="at">data =</span> sample.noma.df)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>permanova_all</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = noma.bray ~ condition, data = sample.noma.df)
         Df SumOfSqs      R2      F Pr(&gt;F)    
Model     1   2.5163 0.40556 23.879  0.001 ***
Residual 35   3.6881 0.59444                  
Total    36   6.2044 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Next we will test the beta dispersion</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># All together now</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">adonis2</span>(noma.bray <span class="sc">~</span> condition, <span class="at">data =</span> sample.noma.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 999

vegan::adonis2(formula = noma.bray ~ condition, data = sample.noma.df)
         Df SumOfSqs      R2      F Pr(&gt;F)    
Model     1   2.5163 0.40556 23.879  0.001 ***
Residual 35   3.6881 0.59444                  
Total    36   6.2044 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">=</span> <span class="fu">betadisper</span>(noma.bray, sample.noma.df<span class="sc">$</span>condition)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">permutest</span>(beta)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Permutation test for homogeneity of multivariate dispersions
Permutation: free
Number of permutations: 999

Response: Distances
          Df  Sum Sq  Mean Sq      F N.Perm Pr(&gt;F)  
Groups     1 0.07827 0.078269 3.8016    999  0.056 .
Residuals 35 0.72060 0.020589                       
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we don't want this to be significant </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="anosim-across-entire-dataset" class="level2">
<h2 class="anchored" data-anchor-id="anosim-across-entire-dataset">3.2 Anosim across entire dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>condition_group <span class="ot">=</span> <span class="fu">get_variable</span>(phyloseq_noma_esd, <span class="st">"condition"</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span> (<span class="dv">123456</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">anosim</span>(<span class="fu">distance</span>(phyloseq_noma_esd, <span class="st">"bray"</span>), condition_group)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
anosim(x = distance(phyloseq_noma_esd, "bray"), grouping = condition_group) 
Dissimilarity: bray 

ANOSIM statistic R: 0.734 
      Significance: 0.001 

Permutation: free
Number of permutations: 999</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>condition_ano <span class="ot">=</span> <span class="fu">anosim</span>(<span class="fu">distance</span>(phyloseq_noma_esd, <span class="st">"bray"</span>), condition_group)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>condition_ano</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
anosim(x = distance(phyloseq_noma_esd, "bray"), grouping = condition_group) 
Dissimilarity: bray 

ANOSIM statistic R: 0.734 
      Significance: 0.001 

Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</section>
<section id="mrpp-across-entire-dataset" class="level2">
<h2 class="anchored" data-anchor-id="mrpp-across-entire-dataset">3.3 MRPP across entire dataset</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#condition</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>noma.bray <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_noma_esd, <span class="at">method =</span> <span class="st">"bray"</span>) <span class="co"># Calculate bray curtis distance matrix</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>condition_group <span class="ot">=</span> <span class="fu">get_variable</span>(phyloseq_noma_esd, <span class="st">"condition"</span>) <span class="co"># Make condition Grouping</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Run MRPP</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">mrpp</span>(noma.bray, condition_group, <span class="at">permutations =</span> <span class="dv">999</span>, </span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">weight.type =</span> <span class="dv">1</span>, <span class="at">strata =</span> <span class="cn">NULL</span>, <span class="at">parallel =</span> <span class="fu">getOption</span>(<span class="st">"mc.cores"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
vegan::mrpp(dat = noma.bray, grouping = condition_group, permutations = 999,      weight.type = 1, strata = NULL, parallel = getOption("mc.cores")) 

Dissimilarity index: bray 
Weights for groups:  n 

Class means and counts:

      diseased healthy
delta 0.4931   0.3533 
n     17       20     

Chance corrected within-group agreement A: 0.2364 
Based on observed delta 0.4176 and expected delta 0.5468 

Significance of delta: 0.001 
Permutation: free
Number of permutations: 999</code></pre>
</div>
</div>
</section>
<section id="outputting-non-parametric-tests-into-table" class="level2">
<h2 class="anchored" data-anchor-id="outputting-non-parametric-tests-into-table">3.4 Outputting non parametric tests into table</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the list of metadata variables you want to test</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>variables_to_test <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"condition"</span>)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Set a seed for reproducibility of permutation-based tests</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123456</span>)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>bray_dist <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_noma_esd, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the sample data into a data frame for use with adonis2</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>sample_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phyloseq_noma_esd))</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an empty list to store the results from each iteration</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>results_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each variable name in the 'variables_to_test' vector</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (variable <span class="cf">in</span> variables_to_test) {</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">"Running tests for variable:"</span>, variable))</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># PERMANOVA (adonis2) </span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the statistical formula dynamically for the current variable</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>  formula <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">"bray_dist ~"</span>, variable))</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run the PERMANOVA test using the adonis2 function</span></span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>  permanova_res <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">adonis2</span>(formula, <span class="at">data =</span> sample_df, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value from the results. It's in the 'Pr(&gt;F)' column.</span></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>  p_permanova <span class="ot">=</span> permanova_res<span class="sc">$</span><span class="st">`</span><span class="at">Pr(&gt;F)</span><span class="st">`</span>[<span class="dv">1</span>]</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ANOSIM </span></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the grouping factor (the actual variable data) from the phyloseq object</span></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>  grouping_factor <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">get_variable</span>(phyloseq_noma_esd, variable)</span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run the ANOSIM test</span></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a>  anosim_res <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">anosim</span>(bray_dist, grouping_factor, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value (significance) from the ANOSIM result</span></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>  p_anosim <span class="ot">=</span> anosim_res<span class="sc">$</span>signif</span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># MRPP </span></span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The grouping factor is the same as for ANOSIM</span></span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run the MRPP test</span></span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a>  mrpp_res <span class="ot">=</span> vegan<span class="sc">::</span><span class="fu">mrpp</span>(bray_dist, grouping_factor, <span class="at">permutations =</span> <span class="dv">999</span>)</span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the p-value from the MRPP result</span></span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a>  p_mrpp <span class="ot">=</span> mrpp_res<span class="sc">$</span>Pvalue</span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb39-48"><a href="#cb39-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store Results </span></span>
<span id="cb39-49"><a href="#cb39-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Store the p-values for the current variable in our results list.</span></span>
<span id="cb39-50"><a href="#cb39-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># We create a small data frame for this iteration's results.</span></span>
<span id="cb39-51"><a href="#cb39-51" aria-hidden="true" tabindex="-1"></a>  results_list[[variable]] <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb39-52"><a href="#cb39-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">Variable =</span> <span class="fu">paste0</span>(variable, <span class="st">"."</span>),</span>
<span id="cb39-53"><a href="#cb39-53" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">permanova.</span><span class="st">`</span> <span class="ot">=</span> p_permanova,</span>
<span id="cb39-54"><a href="#cb39-54" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">anosim.</span><span class="st">`</span> <span class="ot">=</span> p_anosim,</span>
<span id="cb39-55"><a href="#cb39-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">mrpp.</span><span class="st">`</span> <span class="ot">=</span> p_mrpp,</span>
<span id="cb39-56"><a href="#cb39-56" aria-hidden="true" tabindex="-1"></a>    <span class="co"># 'check.names = FALSE' prevents R from changing our column names</span></span>
<span id="cb39-57"><a href="#cb39-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">check.names =</span> <span class="cn">FALSE</span> </span>
<span id="cb39-58"><a href="#cb39-58" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb39-59"><a href="#cb39-59" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Running tests for variable: condition</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the list of individual data frames into one final table</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>final_results_table <span class="ot">=</span> <span class="fu">do.call</span>(rbind, results_list)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up the row names of the final table</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(final_results_table) <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the final, consolidated table to the console</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(final_results_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    Variable permanova. anosim. mrpp.
1 condition.      0.001   0.001 0.001</code></pre>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(final_results_table, <span class="at">file =</span><span class="st">"../tbls/Table_1A.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="relative-abundance" class="level1">
<h1>4. Relative Abundance</h1>
<p>The top 20 most abundant genera were selected from across the entire dataset and visualised with the plotAbundance function of miaViz.</p>
<section id="plotting-relative-abundance-of-genera-across-samples" class="level2">
<h2 class="anchored" data-anchor-id="plotting-relative-abundance-of-genera-across-samples">4.1 Plotting relative abundance of genera across samples</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a> <span class="co"># Check taxonomy ranks </span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">taxonomyRanks</span>(tse_metaphlan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make an assay for abundance</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>  tse_metaphlan <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_metaphlan, <span class="at">assay.type=</span><span class="st">"counts"</span>, <span class="at">method=</span><span class="st">"relabundance"</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make an altExp and matrix for Genus</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">altExp</span>(tse_metaphlan,<span class="st">"Genus"</span>) <span class="ot">=</span> <span class="fu">agglomerateByRank</span>(tse_metaphlan,<span class="st">"Genus"</span>)</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a dataframe of relative abundance </span></span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a>  relabundance_df_Genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>))</span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a matrix of relative abundance </span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  relabundance_matrix_Genus <span class="ot">=</span> <span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>)</span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># calculate the total relative abundance of each Genus (row sums)</span></span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>  total_relabundance_Genus <span class="ot">=</span> <span class="fu">rowSums</span>(relabundance_matrix_Genus)</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify the top 20 top Genuss</span></span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(total_relabundance_Genus, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>])</span>
<span id="cb46-18"><a href="#cb46-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Delete everything from start to Genus</span></span>
<span id="cb46-19"><a href="#cb46-19" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*_g__"</span>,<span class="st">""</span>,top_Genus)</span>
<span id="cb46-20"><a href="#cb46-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Genus back in </span></span>
<span id="cb46-21"><a href="#cb46-21" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"g__"</span>, <span class="fu">length</span>(top_Genus)), top_Genus))</span>
<span id="cb46-22"><a href="#cb46-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Delete the space introduced by this </span></span>
<span id="cb46-23"><a href="#cb46-23" aria-hidden="true" tabindex="-1"></a>  top_Genus <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">" "</span>,<span class="st">""</span>,top_Genus)</span>
<span id="cb46-24"><a href="#cb46-24" aria-hidden="true" tabindex="-1"></a>  top_Genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "g__Prevotella"      "g__Streptococcus"   "g__Neisseria"      
 [4] "g__Haemophilus"     "g__Veillonella"     "g__Rothia"         
 [7] "g__Fusobacterium"   "g__Treponema"       "g__Schaalia"       
[10] "g__Escherichia"     "g__Aggregatibacter" "g__Selenomonas"    
[13] "g__Leptotrichia"    "g__Actinomyces"     "g__Campylobacter"  
[16] "g__Capnocytophaga"  "g__Porphyromonas"   "g__Bacteroides"    
[19] "g__Gemella"         "g__Filifactor"     </code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a new tse_metaphlan where the top 14 Genuss are recognised, while others are "other"</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>  tse_metaphlan_top_20_Genus <span class="ot">=</span> tse_metaphlan</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowData</span>(tse_metaphlan_top_20_Genus)<span class="sc">$</span>Genus <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">rowData</span>(tse_metaphlan_top_20_Genus)<span class="sc">$</span>Genus <span class="sc">%in%</span> top_Genus, <span class="fu">rowData</span>(tse_metaphlan_top_20_Genus)<span class="sc">$</span>Genus, <span class="st">"-other"</span>)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>  genus_colors <span class="ot">=</span> <span class="fu">c</span>(</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">"-other"</span> <span class="ot">=</span> <span class="st">"#E41A1C"</span>,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Actinomyces"</span> <span class="ot">=</span> <span class="st">"#377EB8"</span>,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Aggregatibacter"</span> <span class="ot">=</span> <span class="st">"#4DAF4A"</span>,</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Bacteroides"</span> <span class="ot">=</span> <span class="st">"#984EA3"</span>,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Campylobacter"</span> <span class="ot">=</span> <span class="st">"#FF7F00"</span>,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Capnocytophaga"</span> <span class="ot">=</span> <span class="st">"#FFFF33"</span>,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Dialister"</span> <span class="ot">=</span> <span class="st">"#E7298A"</span>,</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Escherichia"</span> <span class="ot">=</span> <span class="st">"#A65628"</span>,</span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Filifactor"</span> <span class="ot">=</span> <span class="st">"#F781BF"</span>,</span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Fusobacterium"</span> <span class="ot">=</span> <span class="st">"#999999"</span>,</span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g_Gemella"</span> <span class="ot">=</span> <span class="st">"#1B9E77"</span>,</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Haemophilus"</span> <span class="ot">=</span> <span class="st">"#D95F02"</span>,</span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Leptotrichia"</span> <span class="ot">=</span> <span class="st">"#7570B3"</span>,</span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Neisseria"</span> <span class="ot">=</span> <span class="st">"#E7298A"</span>,</span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Porphyromonas"</span> <span class="ot">=</span> <span class="st">"#66A61E"</span>,</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Prevotella"</span> <span class="ot">=</span> <span class="st">"#E6AB02"</span>,</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Rothia"</span> <span class="ot">=</span> <span class="st">"#66C2A5"</span>,</span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Schaalia"</span> <span class="ot">=</span> <span class="st">"#FC8D62"</span>,</span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Selenomonas"</span> <span class="ot">=</span> <span class="st">"#8DA0CB"</span>,</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Streptococcus"</span> <span class="ot">=</span> <span class="st">"#E78AC3"</span>,</span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Tannerella"</span> <span class="ot">=</span> <span class="st">"#E41A1C"</span>,</span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Treponema"</span> <span class="ot">=</span> <span class="st">"#A6D854"</span>,</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">"g__Veillonella"</span> <span class="ot">=</span> <span class="st">"#FFD92F"</span></span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a>  Genus_plot <span class="ot">=</span> <span class="fu">plotAbundance</span>(tse_metaphlan_top_20_Genus, </span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>                              <span class="at">assay.type =</span> <span class="st">"relabundance"</span>, </span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a>                              <span class="at">rank =</span> <span class="st">"Genus"</span>, </span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a>                              <span class="at">add_x_text =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">30</span>, <span class="at">r =</span> <span class="dv">10</span>, <span class="at">b =</span> <span class="dv">10</span>, <span class="at">l =</span> <span class="dv">10</span>))  </span>
<span id="cb48-37"><a href="#cb48-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-38"><a href="#cb48-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-39"><a href="#cb48-39" aria-hidden="true" tabindex="-1"></a>  Genus_plot_cols <span class="ot">=</span> Genus_plot <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>genus_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for fill is already present.
Adding another scale for fill, which will replace the existing scale.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>  Genus_plot_cols</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_1B.png"</span>, <span class="at">plot =</span>  Genus_plot_cols, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="creating-a-table-of-average-relative-abundance" class="level2">
<h2 class="anchored" data-anchor-id="creating-a-table-of-average-relative-abundance">4.2 Creating a table of average relative abundance</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get numbers in table </span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Total </span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  top_Genus_numbers_basic <span class="ot">=</span> <span class="fu">sort</span>(total_relabundance_Genus, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  top_Genus_numbers_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(top_Genus_numbers_basic)</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  top_Genus_numbers <span class="ot">=</span> <span class="fu">as.tibble</span>(top_Genus_numbers_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `as.tibble()` was deprecated in tibble 2.0.0.
ℹ Please use `as_tibble()` instead.
ℹ The signature and semantics have changed, see `?as_tibble`.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(top_Genus_numbers_df) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(top_Genus_numbers_df))</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>top_Genus_pc <span class="ot">=</span> top_Genus_numbers <span class="sc">%&gt;%</span> </span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">top_Genus_percentage =</span> (top_Genus_numbers_basic<span class="sc">/</span><span class="fu">sum</span>(top_Genus_numbers<span class="sc">$</span>top_Genus_numbers_basic)) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">top_Genus =</span> <span class="fu">rownames</span>(top_Genus_numbers_df))</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(top_Genus_pc<span class="sc">$</span>top_Genus_percentage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 100</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_20_Genus <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus <span class="sc">%in%</span> top_Genus, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_Genus_above_1 <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus_percentage <span class="sc">&gt;</span> <span class="dv">1</span>, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>top_Genus_above_1 <span class="ot">=</span> <span class="fu">unique</span>(top_Genus_pc<span class="sc">$</span>top_Genus_above_1)</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_Genus_above_0<span class="fl">.1</span> <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus_percentage <span class="sc">&gt;</span> <span class="fl">0.1</span>, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>top_Genus_above_0<span class="fl">.1</span> <span class="ot">=</span> <span class="fu">unique</span>(top_Genus_pc<span class="sc">$</span>top_Genus_above_0<span class="fl">.1</span>)</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>mat_colors <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Dark2"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set1"</span>), <span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the abundance plot with a single stacked bar for 1%</span></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ggplot</span>(top_Genus_pc, <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> top_Genus_percentage, <span class="at">fill =</span> top_Genus_above_1)) <span class="sc">+</span></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Abundance (%)"</span>, <span class="at">title =</span> <span class="st">"Abundance Plot of Genus"</span>) <span class="sc">+</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> genus_colors, <span class="at">name =</span> <span class="st">"Top genera above 0.1%"</span>) <span class="sc">+</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Supplementary_Figure_4.png"</span>, <span class="at">plot =</span>  p, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="statistical-differences-in-relative-abundance-with-wilcox-test-and-fdr" class="level2">
<h2 class="anchored" data-anchor-id="statistical-differences-in-relative-abundance-with-wilcox-test-and-fdr">4.3 Statistical differences in relative abundance with wilcox-test and FDR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Step 1: Prepare your data ---</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure 'groups$condition' is a factor with the correct reference level</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="co"># The reference level (e.g., "healthy") is important for the fold change calculation</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">=</span> <span class="fu">colData</span>(tse_metaphlan_noma_genus)</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>groups <span class="ot">=</span> <span class="fu">as.data.frame</span>(sample_data)</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="co"># make a dataframe of relative abundance of all genera for each sample</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>relative_abundances <span class="ot">=</span> relabundance_df_Genus</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove higher taxonomic names</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(relative_abundances) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(relative_abundances))</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose the relative abundances matrix</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>transposed_matrix <span class="ot">=</span> <span class="fu">t</span>(relative_abundances)</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert the resulting matrix back into a data frame</span></span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>relative_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(transposed_matrix)</span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Create factor at group level </span></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a>groups<span class="sc">$</span>condition <span class="ot">=</span> <span class="fu">factor</span>(groups<span class="sc">$</span>condition, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"healthy"</span>, <span class="st">"diseased"</span>))</span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine abundance and metadata into one data frame for easy access</span></span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>full_data <span class="ot">=</span> <span class="fu">cbind</span>(relative_df, groups)</span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the list of all genera you want to test</span></span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a>genera_to_test <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"g__Actinomyces"</span>, <span class="st">"g__Aggregatibacter"</span>, <span class="st">"g__Bacteroides"</span>, </span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Campylobacter"</span>, <span class="st">"g__Capnocytophaga"</span>, <span class="st">"g__Dialister"</span>,</span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Escherichia"</span>, <span class="st">"g__Filifactor"</span>, <span class="st">"g__Fusobacterium"</span>,</span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Gemella"</span>, <span class="st">"g__Haemophilus"</span>, <span class="st">"g__Leptotrichia"</span>,</span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Neisseria"</span>, <span class="st">"g__Porphyromonas"</span>, <span class="st">"g__Prevotella"</span>,</span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Rothia"</span>, <span class="st">"g__Schaalia"</span>, <span class="st">"g__Selenomonas"</span>,</span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Streptococcus"</span>, <span class="st">"g__Tannerella"</span>, <span class="st">"g__Treponema"</span>, </span>
<span id="cb58-35"><a href="#cb58-35" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"g__Veillonella"</span>)</span>
<span id="cb58-36"><a href="#cb58-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-37"><a href="#cb58-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate a pseudocount</span></span>
<span id="cb58-38"><a href="#cb58-38" aria-hidden="true" tabindex="-1"></a><span class="co"># This is a small number to add before calculating fold change to avoid log(0) issues.</span></span>
<span id="cb58-39"><a href="#cb58-39" aria-hidden="true" tabindex="-1"></a><span class="co"># A good pseudocount is half of the smallest non-zero value in your abundance data.</span></span>
<span id="cb58-40"><a href="#cb58-40" aria-hidden="true" tabindex="-1"></a>pseudocount <span class="ot">=</span> <span class="fu">min</span>(relative_df[relative_df <span class="sc">&gt;</span> <span class="dv">0</span>]) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb58-41"><a href="#cb58-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-42"><a href="#cb58-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop, Test, and Calculate </span></span>
<span id="cb58-43"><a href="#cb58-43" aria-hidden="true" tabindex="-1"></a><span class="co"># We will loop through each genus and perform all calculations</span></span>
<span id="cb58-44"><a href="#cb58-44" aria-hidden="true" tabindex="-1"></a>results_table <span class="ot">=</span> <span class="fu">map_dfr</span>(genera_to_test, <span class="sc">~</span>{</span>
<span id="cb58-45"><a href="#cb58-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-46"><a href="#cb58-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define the current genus</span></span>
<span id="cb58-47"><a href="#cb58-47" aria-hidden="true" tabindex="-1"></a>  current_genus <span class="ot">=</span> .x</span>
<span id="cb58-48"><a href="#cb58-48" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-49"><a href="#cb58-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 1. Perform the Wilcoxon Rank-Sum Test</span></span>
<span id="cb58-50"><a href="#cb58-50" aria-hidden="true" tabindex="-1"></a>  test_result <span class="ot">=</span> <span class="fu">wilcox.test</span>(</span>
<span id="cb58-51"><a href="#cb58-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="st">"`"</span>, current_genus, <span class="st">"` ~ condition"</span>)), </span>
<span id="cb58-52"><a href="#cb58-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> full_data</span>
<span id="cb58-53"><a href="#cb58-53" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb58-54"><a href="#cb58-54" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-55"><a href="#cb58-55" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 2. Calculate Median Abundances for each group</span></span>
<span id="cb58-56"><a href="#cb58-56" aria-hidden="true" tabindex="-1"></a>  medians <span class="ot">=</span> full_data <span class="sc">%&gt;%</span></span>
<span id="cb58-57"><a href="#cb58-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(condition) <span class="sc">%&gt;%</span></span>
<span id="cb58-58"><a href="#cb58-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">median_abund =</span> <span class="fu">median</span>(.data[[current_genus]]))</span>
<span id="cb58-59"><a href="#cb58-59" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-60"><a href="#cb58-60" aria-hidden="true" tabindex="-1"></a>  median_healthy <span class="ot">=</span> medians<span class="sc">$</span>median_abund[medians<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"healthy"</span>]</span>
<span id="cb58-61"><a href="#cb58-61" aria-hidden="true" tabindex="-1"></a>  median_diseased <span class="ot">=</span> medians<span class="sc">$</span>median_abund[medians<span class="sc">$</span>condition <span class="sc">==</span> <span class="st">"diseased"</span>]</span>
<span id="cb58-62"><a href="#cb58-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-63"><a href="#cb58-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 3. Calculate Log2 Fold Change using medians and the pseudocount</span></span>
<span id="cb58-64"><a href="#cb58-64" aria-hidden="true" tabindex="-1"></a>  log2_fc <span class="ot">=</span> <span class="fu">log2</span>((median_diseased <span class="sc">+</span> pseudocount) <span class="sc">/</span> (median_healthy <span class="sc">+</span> pseudocount))</span>
<span id="cb58-65"><a href="#cb58-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-66"><a href="#cb58-66" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 4. Return all results for this genus in a single row</span></span>
<span id="cb58-67"><a href="#cb58-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(</span>
<span id="cb58-68"><a href="#cb58-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">Genus =</span> <span class="fu">gsub</span>(<span class="st">"g__"</span>, <span class="st">""</span>, current_genus),</span>
<span id="cb58-69"><a href="#cb58-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> test_result<span class="sc">$</span>p.value,</span>
<span id="cb58-70"><a href="#cb58-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">log2_fold_change =</span> log2_fc</span>
<span id="cb58-71"><a href="#cb58-71" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb58-72"><a href="#cb58-72" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb58-73"><a href="#cb58-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-74"><a href="#cb58-74" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(results_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
  Genus            p_value log2_fold_change
  &lt;chr&gt;              &lt;dbl&gt;            &lt;dbl&gt;
1 Actinomyces     1.81e- 6           -4.03 
2 Aggregatibacter 9.75e- 4            3.19 
3 Bacteroides     8.80e-10            2.17 
4 Campylobacter   3.41e- 1            0.592
5 Capnocytophaga  1.84e- 2            1.97 
6 Dialister       2.01e- 7            5.42 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply FDR Correction and sort </span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>results_table <span class="ot">=</span> results_table <span class="sc">%&gt;%</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">q_value_BH =</span> <span class="fu">p.adjust</span>(p_value, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Reorder columns and sort by significance</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Genus, p_value, q_value_BH, log2_fold_change) <span class="sc">%&gt;%</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(q_value_BH)</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="co"># View your final results table </span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(results_table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 22 × 4
   Genus          p_value    q_value_BH log2_fold_change
   &lt;chr&gt;            &lt;dbl&gt;         &lt;dbl&gt;            &lt;dbl&gt;
 1 Streptococcus 1.26e-10 0.00000000138            -3.41
 2 Treponema     1.26e-10 0.00000000138             5.79
 3 Bacteroides   8.80e-10 0.00000000645             2.17
 4 Rothia        8.42e- 9 0.0000000463             -6.68
 5 Filifactor    6.39e- 8 0.000000234               4.84
 6 Schaalia      6.39e- 8 0.000000234              -6.59
 7 Porphyromonas 1.52e- 7 0.000000479               3.15
 8 Dialister     2.01e- 7 0.000000552               5.42
 9 Veillonella   1.45e- 6 0.00000353               -3.70
10 Actinomyces   1.81e- 6 0.00000398               -4.03
# ℹ 12 more rows</code></pre>
</div>
</div>
</section>
<section id="plotting-differences-in-relative-abundance-of-top-20-genera-as-boxplots" class="level2">
<h2 class="anchored" data-anchor-id="plotting-differences-in-relative-abundance-of-top-20-genera-as-boxplots">4.4 Plotting differences in relative abundance of top 20 genera as boxplots</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This new function takes q_val and lfc as arguments</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>plot_relab_boxplot_with_stats <span class="ot">=</span> <span class="cf">function</span>(genus_name, abundance_data, metadata, q_val, lfc) {</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Prepare the data frame for this specific plot</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  plot_df <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">abundance =</span> abundance_data[[<span class="fu">paste0</span>(<span class="st">"g__"</span>, genus_name)]],</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">condition =</span> metadata<span class="sc">$</span>condition <span class="co"># Assumes the condition column is named 'condition'</span></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the text for the annotation</span></span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># We format the numbers to look nice on the plot</span></span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>  annotation_text <span class="ot">=</span> <span class="fu">paste0</span>(</span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">"q = "</span>, <span class="fu">scientific</span>(q_val, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>    <span class="st">"</span><span class="sc">\n</span><span class="st">LFC = "</span>, <span class="fu">round</span>(lfc, <span class="dv">2</span>) <span class="co"># "\n" creates a new line</span></span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the boxplot</span></span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>(plot_df, <span class="fu">aes</span>(<span class="at">x =</span> condition, <span class="at">y =</span> abundance, <span class="at">fill =</span> condition)) <span class="sc">+</span></span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> <span class="co"># Hide outliers to prevent overplotting</span></span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">shape =</span> <span class="dv">16</span>, <span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span> <span class="co"># Show all points</span></span>
<span id="cb62-21"><a href="#cb62-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Add the custom text annotation to the plot</span></span>
<span id="cb62-22"><a href="#cb62-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(</span>
<span id="cb62-23"><a href="#cb62-23" aria-hidden="true" tabindex="-1"></a>      <span class="st">"text"</span>,</span>
<span id="cb62-24"><a href="#cb62-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fl">1.5</span>, <span class="co"># Position horizontally in the middle</span></span>
<span id="cb62-25"><a href="#cb62-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="cn">Inf</span>, <span class="co"># Position vertically at the top</span></span>
<span id="cb62-26"><a href="#cb62-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">vjust =</span> <span class="fl">1.2</span>, <span class="co"># Nudge it down from the very top</span></span>
<span id="cb62-27"><a href="#cb62-27" aria-hidden="true" tabindex="-1"></a>      <span class="at">label =</span> annotation_text,</span>
<span id="cb62-28"><a href="#cb62-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">size =</span> <span class="fl">3.5</span></span>
<span id="cb62-29"><a href="#cb62-29" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb62-30"><a href="#cb62-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb62-31"><a href="#cb62-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> genus_name,</span>
<span id="cb62-32"><a href="#cb62-32" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="cn">NULL</span>, <span class="co"># Remove x-axis label for cleaner grid</span></span>
<span id="cb62-33"><a href="#cb62-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Relative Abundance (%)"</span></span>
<span id="cb62-34"><a href="#cb62-34" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb62-35"><a href="#cb62-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb62-36"><a href="#cb62-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"healthy"</span> <span class="ot">=</span> <span class="st">"#4682B4"</span>, <span class="st">"diseased"</span> <span class="ot">=</span> <span class="st">"#E41A1C"</span>)) <span class="sc">+</span></span>
<span id="cb62-37"><a href="#cb62-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb62-38"><a href="#cb62-38" aria-hidden="true" tabindex="-1"></a>          <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"italic"</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb62-39"><a href="#cb62-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-40"><a href="#cb62-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the plot object instead of printing it</span></span>
<span id="cb62-41"><a href="#cb62-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb62-42"><a href="#cb62-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb62-43"><a href="#cb62-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-44"><a href="#cb62-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort the results table by log2_fold_change </span></span>
<span id="cb62-45"><a href="#cb62-45" aria-hidden="true" tabindex="-1"></a>results_table_sorted <span class="ot">=</span> results_table <span class="sc">%&gt;%</span></span>
<span id="cb62-46"><a href="#cb62-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(log2_fold_change))</span>
<span id="cb62-47"><a href="#cb62-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-48"><a href="#cb62-48" aria-hidden="true" tabindex="-1"></a><span class="co"># This loop creates all the plots and stores them in 'plot_list'</span></span>
<span id="cb62-49"><a href="#cb62-49" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">=</span> <span class="fu">map</span>(results_table_sorted<span class="sc">$</span>Genus, <span class="sc">~</span>{</span>
<span id="cb62-50"><a href="#cb62-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-51"><a href="#cb62-51" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the current genus name</span></span>
<span id="cb62-52"><a href="#cb62-52" aria-hidden="true" tabindex="-1"></a>  current_genus <span class="ot">=</span> .x</span>
<span id="cb62-53"><a href="#cb62-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-54"><a href="#cb62-54" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Find the stats for this genus from your results_table</span></span>
<span id="cb62-55"><a href="#cb62-55" aria-hidden="true" tabindex="-1"></a>  stats <span class="ot">=</span> results_table <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Genus <span class="sc">==</span> current_genus)</span>
<span id="cb62-56"><a href="#cb62-56" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-57"><a href="#cb62-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Call our new function, passing in the stats from the table</span></span>
<span id="cb62-58"><a href="#cb62-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_relab_boxplot_with_stats</span>(</span>
<span id="cb62-59"><a href="#cb62-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">genus_name =</span> current_genus,</span>
<span id="cb62-60"><a href="#cb62-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">abundance_data =</span> relative_df, <span class="co"># Your abundance data frame</span></span>
<span id="cb62-61"><a href="#cb62-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">metadata =</span> groups,           <span class="co"># Your metadata/groups data frame</span></span>
<span id="cb62-62"><a href="#cb62-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">q_val =</span> stats<span class="sc">$</span>q_value_BH,</span>
<span id="cb62-63"><a href="#cb62-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">lfc =</span> stats<span class="sc">$</span>log2_fold_change</span>
<span id="cb62-64"><a href="#cb62-64" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb62-65"><a href="#cb62-65" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb62-66"><a href="#cb62-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-67"><a href="#cb62-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-68"><a href="#cb62-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Arrange all the plots in the list into a grid with 6 columns</span></span>
<span id="cb62-69"><a href="#cb62-69" aria-hidden="true" tabindex="-1"></a>final_plot <span class="ot">=</span> <span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">6</span>)</span>
<span id="cb62-70"><a href="#cb62-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-71"><a href="#cb62-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a main title to the entire grid</span></span>
<span id="cb62-72"><a href="#cb62-72" aria-hidden="true" tabindex="-1"></a>final_plot <span class="sc">+</span> <span class="fu">plot_annotation</span>(<span class="at">title =</span> <span class="st">"Relative Abundance of Differentially Abundant Genera"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Supplementary_Figure_3.png"</span>, final_plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="differential-analysis-with-deseq" class="level1">
<h1>5. Differential Analysis with Deseq</h1>
<p>Differential analysis used the DESeq2 model on normalised count data and determined fold-change and significant differences between noma and healthy samples at the genera level.</p>
<section id="preparing-the-data-1" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data-1">5.1 Preparing the data</h2>
<p>Use makePhyloseqFromTreeSE from Miaverse to convert</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assume your TreeSummarizedExperiment object is called `tse`</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the current sample names (column names)</span></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_genus <span class="ot">=</span> <span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>) </span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>sample_names_genus <span class="ot">=</span> <span class="fu">colnames</span>(tse_metaphlan_genus)</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>sample_names_genus</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "A10"        "A11"        "A13"        "A14"        "A16"       
 [6] "A17"        "A18"        "A19"        "A1"         "A2"        
[11] "A3"         "A4"         "A5"         "A6"         "A7"        
[16] "A8"         "A9"         "DRR214959"  "DRR214960"  "DRR214961" 
[21] "DRR214962"  "DRR241310"  "SRR5892208" "SRR5892209" "SRR5892210"
[26] "SRR5892211" "SRR5892212" "SRR5892213" "SRR5892214" "SRR5892215"
[31] "SRR5892216" "SRR5892217" "SRS013942"  "SRS014468"  "SRS014692" 
[36] "SRS015055"  "SRS019120" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a vector indicating the condition (diseased or healthy)</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming sample_1 to sample_19 are diseased, and sample_20 to sample_22 are healthy</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>condition <span class="ot">=</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"diseased"</span>, <span class="dv">17</span>), <span class="fu">rep</span>(<span class="st">"healthy"</span>, <span class="dv">20</span>))</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a DataFrame with this information</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>sample_metadata_disease <span class="ot">=</span> <span class="fu">DataFrame</span>(<span class="at">condition =</span> condition)</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_metadata_disease) <span class="ot">=</span> sample_names</span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>sample_metadata_disease</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 37 rows and 1 column
            condition
          &lt;character&gt;
A10          diseased
A11          diseased
A13          diseased
A14          diseased
A16          diseased
...               ...
SRS013942     healthy
SRS014468     healthy
SRS014692     healthy
SRS015055     healthy
SRS019120     healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(sample_metadata_disease), <span class="at">n =</span> <span class="dv">37</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           condition
A10         diseased
A11         diseased
A13         diseased
A14         diseased
A16         diseased
A17         diseased
A18         diseased
A19         diseased
A1          diseased
A2          diseased
A3          diseased
A4          diseased
A5          diseased
A6          diseased
A7          diseased
A8          diseased
A9          diseased
DRR214959    healthy
DRR214960    healthy
DRR214961    healthy
DRR214962    healthy
DRR241310    healthy
SRR5892208   healthy
SRR5892209   healthy
SRR5892210   healthy
SRR5892211   healthy
SRR5892212   healthy
SRR5892213   healthy
SRR5892214   healthy
SRR5892215   healthy
SRR5892216   healthy
SRR5892217   healthy
SRS013942    healthy
SRS014468    healthy
SRS014692    healthy
SRS015055    healthy
SRS019120    healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add this DataFrame as colData to your TreeSummarizedExperiment object</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_genus) <span class="ot">=</span> sample_metadata_disease</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check that colData was added successfully</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 37 rows and 1 column
            condition
          &lt;character&gt;
A10          diseased
A11          diseased
A13          diseased
A14          diseased
A16          diseased
...               ...
SRS013942     healthy
SRS014468     healthy
SRS014692     healthy
SRS015055     healthy
SRS019120     healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan_genus <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_genus)</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>deseq2_metaphlan_genus <span class="ot">=</span> <span class="fu">phyloseq_to_deseq2</span>(phyloseq_metaphlan_genus, <span class="at">design =</span> <span class="sc">~</span>condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
</div>
</section>
<section id="running-differential-analysis-on-diseased-vs-healthy" class="level2">
<h2 class="anchored" data-anchor-id="running-differential-analysis-on-diseased-vs-healthy">5.2 Running differential analysis on diseased vs healthy</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>dds_genus_data <span class="ot">=</span> deseq2_metaphlan_genus</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dds_genus_data) <span class="ot">=</span> <span class="er">~</span> condition  <span class="co"># Replace with your column name for condition</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2 analysis</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>dds_genus <span class="ot">=</span> <span class="fu">DESeq</span>(dds_genus_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extracting-results-for-disease-state" class="level2">
<h2 class="anchored" data-anchor-id="extracting-results-for-disease-state">5.3 Extracting results for disease state</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results for diseased vs healthy</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>res_genus <span class="ot">=</span> <span class="fu">results</span>(dds_genus, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"condition"</span>, <span class="st">"diseased"</span>, <span class="st">"healthy"</span>))</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>string <span class="ot">=</span> <span class="st">"NA_p__Actinobacteria_c__Actinobacteria_o__Streptomycetales_f__Streptomycetaceae_g__Streptomyces"</span></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>string_result <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__Streptomyces)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, string)</span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(string_result)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Streptomyces"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up genus names for dds</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(dds_genus) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(dds_genus))</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up genus names for res</span></span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>res_genus<span class="sc">@</span>rownames <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, res_genus<span class="sc">@</span>rownames)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-output-as-volcano-plot" class="level2">
<h2 class="anchored" data-anchor-id="plotting-output-as-volcano-plot">5.4 Plotting output as volcano plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>deseq_volcano_g <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(res_genus), </span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, </span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(pvalue), </span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">color =</span> <span class="fu">ifelse</span>(<span class="sc">-</span><span class="fu">log10</span>(pvalue) <span class="sc">&gt;</span> <span class="fl">1.3</span>, log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"grey"</span>), </span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">ifelse</span>(<span class="sc">-</span><span class="fu">log10</span>(pvalue) <span class="sc">&gt;</span> <span class="fl">1.3</span>, <span class="fu">as.character</span>(<span class="fu">rownames</span>(<span class="fu">as.data.frame</span>(res_genus))), <span class="st">''</span>))) <span class="sc">+</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">1.3</span>) <span class="sc">+</span> </span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">name =</span> <span class="st">"Log2 fold change"</span>, </span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"TRUE"</span> <span class="ot">=</span> <span class="st">"#E47273"</span>, <span class="st">"FALSE"</span> <span class="ot">=</span> <span class="st">"#6DA6CE"</span>, <span class="st">"grey"</span> <span class="ot">=</span> <span class="st">"grey"</span>)) <span class="sc">+</span></span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Differential Abundance Analysis"</span>,</span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"-Log10 P-Value"</span>) <span class="sc">+</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>() </span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>deseq_volcano_g </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 723 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_2A.png"</span>, <span class="at">plot =</span>  deseq_volcano_g, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 648 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
</div>
</section>
<section id="inspecting-genera-that-are-significantly-different-between-disease-states" class="level2">
<h2 class="anchored" data-anchor-id="inspecting-genera-that-are-significantly-different-between-disease-states">5.5 Inspecting Genera that are significantly different between disease states</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sort summary list by p-value</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>res_ordered_genus <span class="ot">=</span> res_genus[<span class="fu">order</span>(res_genus<span class="sc">$</span>padj),]</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_ordered_genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition diseased vs healthy 
Wald test p-value: condition diseased vs healthy 
DataFrame with 6 rows and 6 columns
                    baseMean log2FoldChange     lfcSE      stat      pvalue
                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
g__Streptococcus 1.20291e+06       -4.55825  0.372011  -12.2530 1.61898e-34
g__Wolinella     8.26901e+01        4.31475  0.366735   11.7653 5.89076e-32
g__Atopobium     1.71802e+04       -5.01526  0.443219  -11.3155 1.09918e-29
g__Mesorhizobium 1.10485e+03        3.43460  0.323326   10.6227 2.33594e-26
g__Mogibacterium 1.99595e+04       -3.38475  0.319175  -10.6047 2.83487e-26
g__Treponema     5.17980e+04        4.65427  0.445544   10.4463 1.52395e-25
                        padj
                   &lt;numeric&gt;
g__Streptococcus 2.63084e-31
g__Wolinella     4.78624e-29
g__Atopobium     5.95388e-27
g__Mesorhizobium 9.21334e-24
g__Mogibacterium 9.21334e-24
g__Treponema     4.12737e-23</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_ordered_genus, <span class="at">n =</span><span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition diseased vs healthy 
Wald test p-value: condition diseased vs healthy 
DataFrame with 20 rows and 6 columns
                     baseMean log2FoldChange     lfcSE      stat      pvalue
                    &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
g__Streptococcus  1.20291e+06       -4.55825  0.372011  -12.2530 1.61898e-34
g__Wolinella      8.26901e+01        4.31475  0.366735   11.7653 5.89076e-32
g__Atopobium      1.71802e+04       -5.01526  0.443219  -11.3155 1.09918e-29
g__Mesorhizobium  1.10485e+03        3.43460  0.323326   10.6227 2.33594e-26
g__Mogibacterium  1.99595e+04       -3.38475  0.319175  -10.6047 2.83487e-26
...                       ...            ...       ...       ...         ...
g__Trueperella        369.261       -3.88747  0.446248  -8.71147 2.99966e-18
g__Brochothrix        104.056       -1.75310  0.202062  -8.67605 4.09762e-18
g__Cellulomonas       276.053       -3.45878  0.402625  -8.59058 8.65346e-18
g__Carnobacterium     965.031       -2.28987  0.267449  -8.56188 1.11039e-17
g__Helicobacter       466.085        1.54649  0.182290   8.48371 2.18116e-17
                         padj
                    &lt;numeric&gt;
g__Streptococcus  2.63084e-31
g__Wolinella      4.78624e-29
g__Atopobium      5.95388e-27
g__Mesorhizobium  9.21334e-24
g__Mogibacterium  9.21334e-24
...                       ...
g__Trueperella    3.04653e-16
g__Brochothrix    3.91684e-16
g__Cellulomonas   7.81215e-16
g__Carnobacterium 9.49675e-16
g__Helicobacter   1.77219e-15</code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for significant species </span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>significant_genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_genus) <span class="sc">%&gt;%</span></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                           baseMean log2FoldChange     lfcSE     stat
g__Coprothermobacter       5.335380      1.4410253 0.5116803 2.816261
g__Caldisericum           17.227644      1.8273856 0.3686445 4.957040
g__Endomicrobium          19.087437      1.1224406 0.3217218 3.488855
g__Thermodesulfobacterium 36.973878      0.6180491 0.2182316 2.832078
g__Thermovibrio            8.442273      1.2631596 0.4095794 3.084041
g__Hydrogenobaculum       10.621548      1.5927663 0.5268100 3.023417
                                pvalue         padj
g__Coprothermobacter      4.858615e-03 1.572759e-02
g__Caldisericum           7.157526e-07 5.512312e-06
g__Endomicrobium          4.850945e-04 2.096486e-03
g__Thermodesulfobacterium 4.624651e-03 1.518194e-02
g__Thermovibrio           2.042094e-03 7.357877e-03
g__Hydrogenobaculum       2.499376e-03 8.772108e-03</code></pre>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for significant species with higher abundance in healthy samples</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>significant_genus_less_thn_zero <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_genus) <span class="sc">%&gt;%</span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>)</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for significant species with higher abundance in diseased samples</span></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>significant_genus_grtr_thn_zero <span class="ot">=</span> <span class="fu">as.data.frame</span>(res_genus) <span class="sc">%&gt;%</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the results</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_genus_grtr_thn_zero)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                           baseMean log2FoldChange     lfcSE     stat
g__Coprothermobacter       5.335380      1.4410253 0.5116803 2.816261
g__Caldisericum           17.227644      1.8273856 0.3686445 4.957040
g__Endomicrobium          19.087437      1.1224406 0.3217218 3.488855
g__Thermodesulfobacterium 36.973878      0.6180491 0.2182316 2.832078
g__Thermovibrio            8.442273      1.2631596 0.4095794 3.084041
g__Hydrogenobaculum       10.621548      1.5927663 0.5268100 3.023417
                                pvalue         padj
g__Coprothermobacter      4.858615e-03 1.572759e-02
g__Caldisericum           7.157526e-07 5.512312e-06
g__Endomicrobium          4.850945e-04 2.096486e-03
g__Thermodesulfobacterium 4.624651e-03 1.518194e-02
g__Thermovibrio           2.042094e-03 7.357877e-03
g__Hydrogenobaculum       2.499376e-03 8.772108e-03</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(significant_genus_less_thn_zero)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                   baseMean
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA                    9950.98811
g__Candidatus_Saccharimonas                                       132.50097
g__Petrotoga                                                       22.53548
g__Luteitalea                                                      20.57572
NA_p__Planctomycetes_c__Planctomycetia_o__Planctomycetales_NA_NA   16.78372
g__Gemmata                                                         23.15259
                                                                 log2FoldChange
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA                        -3.2200083
g__Candidatus_Saccharimonas                                          -2.7674007
g__Petrotoga                                                         -0.6724982
g__Luteitalea                                                        -0.9780208
NA_p__Planctomycetes_c__Planctomycetia_o__Planctomycetales_NA_NA     -1.1875691
g__Gemmata                                                           -1.3204942
                                                                     lfcSE
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA                    0.4082855
g__Candidatus_Saccharimonas                                      0.4198980
g__Petrotoga                                                     0.2627007
g__Luteitalea                                                    0.3365252
NA_p__Planctomycetes_c__Planctomycetia_o__Planctomycetales_NA_NA 0.4243233
g__Gemmata                                                       0.3807214
                                                                      stat
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA                    -7.886658
g__Candidatus_Saccharimonas                                      -6.590649
g__Petrotoga                                                     -2.559940
g__Luteitalea                                                    -2.906234
NA_p__Planctomycetes_c__Planctomycetia_o__Planctomycetales_NA_NA -2.798737
g__Gemmata                                                       -3.468401
                                                                       pvalue
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA                    3.103870e-15
g__Candidatus_Saccharimonas                                      4.379075e-11
g__Petrotoga                                                     1.046901e-02
g__Luteitalea                                                    3.658079e-03
NA_p__Planctomycetes_c__Planctomycetia_o__Planctomycetales_NA_NA 5.130296e-03
g__Gemmata                                                       5.235659e-04
                                                                         padj
NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA                    1.441083e-13
g__Candidatus_Saccharimonas                                      8.471424e-10
g__Petrotoga                                                     3.016337e-02
g__Luteitalea                                                    1.230720e-02
NA_p__Planctomycetes_c__Planctomycetia_o__Planctomycetales_NA_NA 1.644250e-02
g__Gemmata                                                       2.250779e-03</code></pre>
</div>
</div>
<p>For noma samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results </span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>sig_res_genus <span class="ot">=</span> significant_genus[<span class="fu">order</span>(significant_genus<span class="sc">$</span>padj),]</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange     lfcSE      stat       pvalue
g__Streptococcus 1.202906e+06      -4.558247 0.3720108 -12.25300 1.618980e-34
g__Wolinella     8.269014e+01       4.314750 0.3667349  11.76531 5.890761e-32
g__Atopobium     1.718017e+04      -5.015261 0.4432187 -11.31554 1.099178e-29
g__Mesorhizobium 1.104854e+03       3.434604 0.3233255  10.62274 2.335943e-26
g__Mogibacterium 1.995953e+04      -3.384746 0.3191753 -10.60466 2.834874e-26
g__Treponema     5.179799e+04       4.654275 0.4455440  10.44627 1.523951e-25
                         padj
g__Streptococcus 2.630843e-31
g__Wolinella     4.786244e-29
g__Atopobium     5.953883e-27
g__Mesorhizobium 9.213340e-24
g__Mogibacterium 9.213340e-24
g__Treponema     4.127368e-23</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sig_res_genus, <span class="at">n=</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                              baseMean
g__Streptococcus                                                          1.202906e+06
g__Wolinella                                                              8.269014e+01
g__Atopobium                                                              1.718017e+04
g__Mesorhizobium                                                          1.104854e+03
g__Mogibacterium                                                          1.995953e+04
g__Treponema                                                              5.179799e+04
g__Staphylococcus                                                         4.277333e+03
g__Filifactor                                                             1.174252e+04
g__Cutibacterium                                                          9.676835e+02
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 1.769642e+02
g__Bradyrhizobium                                                         8.111957e+02
g__Lactococcus                                                            2.259492e+03
g__Kurthia                                                                1.268817e+02
g__Salinispira                                                            6.477155e+01
g__Kingella                                                               1.023084e+03
                                                                          log2FoldChange
g__Streptococcus                                                               -4.558247
g__Wolinella                                                                    4.314750
g__Atopobium                                                                   -5.015261
g__Mesorhizobium                                                                3.434604
g__Mogibacterium                                                               -3.384746
g__Treponema                                                                    4.654275
g__Staphylococcus                                                              -2.267672
g__Filifactor                                                                   3.876533
g__Cutibacterium                                                               -3.931289
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA       3.508700
g__Bradyrhizobium                                                               2.395926
g__Lactococcus                                                                 -3.677923
g__Kurthia                                                                     -1.632900
g__Salinispira                                                                  3.530439
g__Kingella                                                                     4.225752
                                                                              lfcSE
g__Streptococcus                                                          0.3720108
g__Wolinella                                                              0.3667349
g__Atopobium                                                              0.4432187
g__Mesorhizobium                                                          0.3233255
g__Mogibacterium                                                          0.3191753
g__Treponema                                                              0.4455440
g__Staphylococcus                                                         0.2269665
g__Filifactor                                                             0.3909906
g__Cutibacterium                                                          0.4055293
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 0.3682544
g__Bradyrhizobium                                                         0.2517317
g__Lactococcus                                                            0.3882346
g__Kurthia                                                                0.1740667
g__Salinispira                                                            0.3804755
g__Kingella                                                               0.4697250
                                                                                stat
g__Streptococcus                                                          -12.252998
g__Wolinella                                                               11.765313
g__Atopobium                                                              -11.315544
g__Mesorhizobium                                                           10.622743
g__Mogibacterium                                                          -10.604662
g__Treponema                                                               10.446274
g__Staphylococcus                                                          -9.991219
g__Filifactor                                                               9.914646
g__Cutibacterium                                                           -9.694218
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA   9.527923
g__Bradyrhizobium                                                           9.517779
g__Lactococcus                                                             -9.473456
g__Kurthia                                                                 -9.380891
g__Salinispira                                                              9.279020
g__Kingella                                                                 8.996226
                                                                                pvalue
g__Streptococcus                                                          1.618980e-34
g__Wolinella                                                              5.890761e-32
g__Atopobium                                                              1.099178e-29
g__Mesorhizobium                                                          2.335943e-26
g__Mogibacterium                                                          2.834874e-26
g__Treponema                                                              1.523951e-25
g__Staphylococcus                                                         1.665213e-23
g__Filifactor                                                             3.595287e-23
g__Cutibacterium                                                          3.190735e-22
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 1.604608e-21
g__Bradyrhizobium                                                         1.769193e-21
g__Lactococcus                                                            2.707342e-21
g__Kurthia                                                                6.541730e-21
g__Salinispira                                                            1.710452e-20
g__Kingella                                                               2.336101e-19
                                                                                  padj
g__Streptococcus                                                          2.630843e-31
g__Wolinella                                                              4.786244e-29
g__Atopobium                                                              5.953883e-27
g__Mesorhizobium                                                          9.213340e-24
g__Mogibacterium                                                          9.213340e-24
g__Treponema                                                              4.127368e-23
g__Staphylococcus                                                         3.865673e-21
g__Filifactor                                                             7.302927e-21
g__Cutibacterium                                                          5.761050e-20
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 2.607487e-19
g__Bradyrhizobium                                                         2.613581e-19
g__Lactococcus                                                            3.666192e-19
g__Kurthia                                                                8.177163e-19
g__Salinispira                                                            1.985347e-18
g__Kingella                                                               2.530776e-17</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results by significance greater than zero (Noma)</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>sig_res_genus_grtr_thn_zero <span class="ot">=</span> significant_genus_grtr_thn_zero[<span class="fu">order</span>(significant_genus_grtr_thn_zero<span class="sc">$</span>padj),]</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>sig_res_genus_grtr_thn_zero<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_genus_grtr_thn_zero)</span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(sig_res_genus_grtr_thn_zero, <span class="at">n =</span> <span class="dv">30</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                             baseMean
g__Wolinella                                                                 82.69014
g__Mesorhizobium                                                           1104.85379
g__Treponema                                                              51797.99394
g__Filifactor                                                             11742.51502
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA   176.96419
g__Bradyrhizobium                                                           811.19569
                                                                          log2FoldChange
g__Wolinella                                                                    4.314750
g__Mesorhizobium                                                                3.434604
g__Treponema                                                                    4.654275
g__Filifactor                                                                   3.876533
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA       3.508700
g__Bradyrhizobium                                                               2.395926
                                                                              lfcSE
g__Wolinella                                                              0.3667349
g__Mesorhizobium                                                          0.3233255
g__Treponema                                                              0.4455440
g__Filifactor                                                             0.3909906
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 0.3682544
g__Bradyrhizobium                                                         0.2517317
                                                                               stat
g__Wolinella                                                              11.765313
g__Mesorhizobium                                                          10.622743
g__Treponema                                                              10.446274
g__Filifactor                                                              9.914646
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA  9.527923
g__Bradyrhizobium                                                          9.517779
                                                                                pvalue
g__Wolinella                                                              5.890761e-32
g__Mesorhizobium                                                          2.335943e-26
g__Treponema                                                              1.523951e-25
g__Filifactor                                                             3.595287e-23
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 1.604608e-21
g__Bradyrhizobium                                                         1.769193e-21
                                                                                  padj
g__Wolinella                                                              4.786244e-29
g__Mesorhizobium                                                          9.213340e-24
g__Treponema                                                              4.127368e-23
g__Filifactor                                                             7.302927e-21
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 2.607487e-19
g__Bradyrhizobium                                                         2.613581e-19
                                                                                                                                              genus
g__Wolinella                                                                                                                           g__Wolinella
g__Mesorhizobium                                                                                                                   g__Mesorhizobium
g__Treponema                                                                                                                           g__Treponema
g__Filifactor                                                                                                                         g__Filifactor
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA
g__Bradyrhizobium                                                                                                                 g__Bradyrhizobium</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>deseq_genus_sig_diff_noma <span class="ot">=</span> <span class="fu">head</span>(sig_res_genus_grtr_thn_zero, <span class="at">n =</span> <span class="dv">30</span>)</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results by change greater than zero (Noma)</span></span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>change_res_genus_grtr_thn_zero <span class="ot">=</span> significant_genus_grtr_thn_zero[<span class="fu">order</span>(significant_genus_grtr_thn_zero<span class="sc">$</span>log2FoldChange),]</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(change_res_genus_grtr_thn_zero, <span class="at">n  =</span> <span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                           baseMean log2FoldChange     lfcSE     stat
g__Polynucleobacter       158.58906      0.3636115 0.1356301 2.680906
g__Geobacter              197.18389      0.4759009 0.1852261 2.569297
g__Pandoraea              112.41762      0.4765583 0.2021663 2.357259
g__Thermoanaerobacterium  199.00680      0.5044903 0.1998842 2.523913
g__Blattabacterium         91.60746      0.5085865 0.1965401 2.587698
g__Chlorobium              73.96961      0.5181272 0.1777769 2.914480
g__Salegentibacter         82.57517      0.5339047 0.2164407 2.466748
g__Thermosipho             57.03753      0.5379010 0.2015835 2.668379
g__Arcobacter             528.83484      0.5452385 0.1749888 3.115849
g__Echinicola             113.89700      0.5461497 0.1670004 3.270351
g__Brachyspira            254.89486      0.5813467 0.2025750 2.869784
g__Hungateiclostridium    251.56932      0.5858876 0.2157149 2.716027
g__Natranaerobius          20.39054      0.5919283 0.2369416 2.498203
g__Legionella             287.40179      0.6098754 0.1350509 4.515894
g__Thermodesulfobacterium  36.97388      0.6180491 0.2182316 2.832078
                                pvalue         padj
g__Polynucleobacter       7.342307e-03 2.234316e-02
g__Geobacter              1.019050e-02 2.951793e-02
g__Pandoraea              1.841040e-02 4.896383e-02
g__Thermoanaerobacterium  1.160566e-02 3.285575e-02
g__Blattabacterium        9.661964e-03 2.813744e-02
g__Chlorobium             3.562811e-03 1.206160e-02
g__Salegentibacter        1.363462e-02 3.761675e-02
g__Thermosipho            7.621828e-03 2.308301e-02
g__Arcobacter             1.834161e-03 6.697778e-03
g__Echinicola             1.074143e-03 4.236780e-03
g__Brachyspira            4.107519e-03 1.362187e-02
g__Hungateiclostridium    6.607043e-03 2.041149e-02
g__Natranaerobius         1.248245e-02 3.485221e-02
g__Legionella             6.305023e-06 3.955854e-05
g__Thermodesulfobacterium 4.624651e-03 1.518194e-02</code></pre>
</div>
</div>
<p>For healthy samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results by significance less than zero (healthy)</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>sig_res_genus_less_thn_zero <span class="ot">=</span> significant_genus_less_thn_zero[<span class="fu">order</span>(significant_genus_less_thn_zero<span class="sc">$</span>padj),]</span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>sig_res_genus_less_thn_zero<span class="sc">$</span>genus <span class="ot">=</span> <span class="fu">rownames</span>(sig_res_genus_less_thn_zero)</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(sig_res_genus_less_thn_zero, <span class="at">n =</span><span class="dv">30</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      baseMean log2FoldChange     lfcSE       stat       pvalue
g__Streptococcus  1202906.4900      -4.558247 0.3720108 -12.252998 1.618980e-34
g__Atopobium        17180.1727      -5.015261 0.4432187 -11.315544 1.099178e-29
g__Mogibacterium    19959.5262      -3.384746 0.3191753 -10.604662 2.834874e-26
g__Staphylococcus    4277.3332      -2.267672 0.2269665  -9.991219 1.665213e-23
g__Cutibacterium      967.6835      -3.931289 0.4055293  -9.694218 3.190735e-22
g__Lactococcus       2259.4921      -3.677923 0.3882346  -9.473456 2.707342e-21
                          padj             genus
g__Streptococcus  2.630843e-31  g__Streptococcus
g__Atopobium      5.953883e-27      g__Atopobium
g__Mogibacterium  9.213340e-24  g__Mogibacterium
g__Staphylococcus 3.865673e-21 g__Staphylococcus
g__Cutibacterium  5.761050e-20  g__Cutibacterium
g__Lactococcus    3.666192e-19    g__Lactococcus</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>deseq_genus_sig_diff_healthy <span class="ot">=</span> <span class="fu">head</span>(sig_res_genus_less_thn_zero, <span class="at">n =</span> <span class="dv">30</span>)</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the results by change greater than zero (healthy)</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>change_res_genus_less_thn_zero <span class="ot">=</span> significant_genus_less_thn_zero[<span class="fu">order</span>(significant_genus_less_thn_zero<span class="sc">$</span>log2FoldChange),]</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(change_res_genus_less_thn_zero, <span class="at">n =</span><span class="dv">15</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       baseMean log2FoldChange     lfcSE       stat
g__Rothia          3.111581e+05      -5.606861 0.8099458  -6.922513
g__Atopobium       1.718017e+04      -5.015261 0.4432187 -11.315544
g__Schaalia        1.275042e+05      -4.845840 0.6891937  -7.031174
g__Kochikohdavirus 4.688045e+00      -4.830583 0.7413594  -6.515845
g__Streptococcus   1.202906e+06      -4.558247 0.3720108 -12.252998
g__Ruania          6.387048e+01      -4.088696 0.5071776  -8.061665
                         pvalue         padj
g__Rothia          4.437010e-12 9.613521e-11
g__Atopobium       1.099178e-29 5.953883e-27
g__Schaalia        2.048035e-12 4.894202e-11
g__Kochikohdavirus 7.228141e-11 1.276710e-09
g__Streptococcus   1.618980e-34 2.630843e-31
g__Ruania          7.526211e-16 4.367890e-14</code></pre>
</div>
</div>
</section>
<section id="look-for-highly-abundant-significant-genera" class="level2">
<h2 class="anchored" data-anchor-id="look-for-highly-abundant-significant-genera">5.6 Look for highly abundant significant genera</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">taxonomyRanks</span>(tse_metaphlan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make an assay for abundance</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>tse_metaphlan <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_metaphlan, <span class="at">assay.type=</span><span class="st">"counts"</span>, <span class="at">method=</span><span class="st">"relabundance"</span>)</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co"># make an altExp and matrix for Genus</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="fu">altExp</span>(tse_metaphlan,<span class="st">"Genus"</span>) <span class="ot">=</span> <span class="fu">agglomerateByRank</span>(tse_metaphlan,<span class="st">"Genus"</span>)</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="co"># make a dataframe of relative abundance </span></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a>relabundance_df_Genus <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>))</span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="co"># make a matric of relative abundance </span></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a>relabundance_matrix_Genus <span class="ot">=</span> <span class="fu">assay</span>(<span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>), <span class="st">"relabundance"</span>)</span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate the total relative abundance of each Genus (row sums)</span></span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a>total_relabundance_Genus <span class="ot">=</span> <span class="fu">rowSums</span>(relabundance_matrix_Genus)</span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the top highly abundant genera based on relative abundance </span></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a>top_Genus_numbers_basic <span class="ot">=</span> <span class="fu">sort</span>(total_relabundance_Genus, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Make into dataframe</span></span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a>top_Genus_numbers_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(top_Genus_numbers_basic)</span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Make into tibble</span></span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a>top_Genus_numbers <span class="ot">=</span> <span class="fu">as.tibble</span>(top_Genus_numbers_df)</span>
<span id="cb106-24"><a href="#cb106-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename genera to remove any higher taxonomic names (a quirk of the metaphaln style)</span></span>
<span id="cb106-26"><a href="#cb106-26" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(top_Genus_numbers_df) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(top_Genus_numbers_df))</span>
<span id="cb106-27"><a href="#cb106-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-28"><a href="#cb106-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Get percenatge by dividing by totoal number of samples (21) and * by 100</span></span>
<span id="cb106-29"><a href="#cb106-29" aria-hidden="true" tabindex="-1"></a>top_Genus_pc <span class="ot">=</span> top_Genus_numbers <span class="sc">%&gt;%</span> </span>
<span id="cb106-30"><a href="#cb106-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">top_Genus_percentage =</span> (top_Genus_numbers_basic<span class="sc">/</span><span class="dv">21</span>) <span class="sc">*</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb106-31"><a href="#cb106-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">top_Genus =</span> <span class="fu">rownames</span>(top_Genus_numbers_df))</span>
<span id="cb106-32"><a href="#cb106-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-33"><a href="#cb106-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only the top 20 genera by relative abundance </span></span>
<span id="cb106-34"><a href="#cb106-34" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_20_Genus <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus <span class="sc">%in%</span> top_Genus, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb106-35"><a href="#cb106-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-36"><a href="#cb106-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Select only the top genera with a relative abundance above 1%</span></span>
<span id="cb106-37"><a href="#cb106-37" aria-hidden="true" tabindex="-1"></a>top_Genus_pc<span class="sc">$</span>top_Genus_above_1 <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus_percentage <span class="sc">&gt;</span> <span class="dv">1</span>, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb106-38"><a href="#cb106-38" aria-hidden="true" tabindex="-1"></a>top_Genus_above_1 <span class="ot">=</span> <span class="fu">unique</span>(top_Genus_pc<span class="sc">$</span>top_Genus_above_1)</span>
<span id="cb106-39"><a href="#cb106-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-40"><a href="#cb106-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Select </span></span>
<span id="cb106-41"><a href="#cb106-41" aria-hidden="true" tabindex="-1"></a>sig_res_genus_grtr_thn_zero<span class="sc">$</span>genera_above_1pc_relab <span class="ot">=</span> <span class="fu">ifelse</span>(sig_res_genus_grtr_thn_zero<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_1, sig_res_genus_grtr_thn_zero<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb106-42"><a href="#cb106-42" aria-hidden="true" tabindex="-1"></a>sig_res_genus_less_thn_zero<span class="sc">$</span>genera_above_1pc_relab <span class="ot">=</span> <span class="fu">ifelse</span>(sig_res_genus_less_thn_zero<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_1, sig_res_genus_less_thn_zero<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb106-43"><a href="#cb106-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-44"><a href="#cb106-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Check which genera are both signifiantly different and highly abundant </span></span>
<span id="cb106-45"><a href="#cb106-45" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sig_res_genus_grtr_thn_zero<span class="sc">$</span>genera_above_1pc_relab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "-other"           "g__Treponema"     "g__Porphyromonas" "g__Bacteroides"  
[5] "g__Selenomonas"  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(sig_res_genus_less_thn_zero<span class="sc">$</span>genera_above_1pc_relab)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "g__Streptococcus" "-other"           "g__Veillonella"   "g__Gemella"      
[5] "g__Schaalia"      "g__Rothia"        "g__Actinomyces"   "g__Haemophilus"  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(sig_res_genus_grtr_thn_zero, <span class="at">n=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                             baseMean
g__Wolinella                                                                 82.69014
g__Mesorhizobium                                                           1104.85379
g__Treponema                                                              51797.99394
g__Filifactor                                                             11742.51502
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA   176.96419
g__Bradyrhizobium                                                           811.19569
                                                                          log2FoldChange
g__Wolinella                                                                    4.314750
g__Mesorhizobium                                                                3.434604
g__Treponema                                                                    4.654275
g__Filifactor                                                                   3.876533
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA       3.508700
g__Bradyrhizobium                                                               2.395926
                                                                              lfcSE
g__Wolinella                                                              0.3667349
g__Mesorhizobium                                                          0.3233255
g__Treponema                                                              0.4455440
g__Filifactor                                                             0.3909906
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 0.3682544
g__Bradyrhizobium                                                         0.2517317
                                                                               stat
g__Wolinella                                                              11.765313
g__Mesorhizobium                                                          10.622743
g__Treponema                                                              10.446274
g__Filifactor                                                              9.914646
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA  9.527923
g__Bradyrhizobium                                                          9.517779
                                                                                pvalue
g__Wolinella                                                              5.890761e-32
g__Mesorhizobium                                                          2.335943e-26
g__Treponema                                                              1.523951e-25
g__Filifactor                                                             3.595287e-23
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 1.604608e-21
g__Bradyrhizobium                                                         1.769193e-21
                                                                                  padj
g__Wolinella                                                              4.786244e-29
g__Mesorhizobium                                                          9.213340e-24
g__Treponema                                                              4.127368e-23
g__Filifactor                                                             7.302927e-21
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA 2.607487e-19
g__Bradyrhizobium                                                         2.613581e-19
                                                                                                                                              genus
g__Wolinella                                                                                                                           g__Wolinella
g__Mesorhizobium                                                                                                                   g__Mesorhizobium
g__Treponema                                                                                                                           g__Treponema
g__Filifactor                                                                                                                         g__Filifactor
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA
g__Bradyrhizobium                                                                                                                 g__Bradyrhizobium
                                                                          genera_above_1pc_relab
g__Wolinella                                                                              -other
g__Mesorhizobium                                                                          -other
g__Treponema                                                                        g__Treponema
g__Filifactor                                                                             -other
NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA                 -other
g__Bradyrhizobium                                                                         -other</code></pre>
</div>
</div>
<p>Creating a table of the highly abundant significant genera</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare and Filter the Data </span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="co"># We'll convert the results to a clean data frame and filter for significance</span></span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>significant_results <span class="ot">=</span> res_ordered_genus <span class="sc">%&gt;%</span></span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"Genus"</span>) <span class="sc">%&gt;%</span>       <span class="co"># Convert row names to a column</span></span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span>                      <span class="co"># Filter for significant results</span></span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(padj)                                <span class="co"># Ensure it's sorted by significance</span></span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract Key Numbers </span></span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Total number of significant genera</span></span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>total_significant_count <span class="ot">=</span> <span class="fu">nrow</span>(significant_results)</span>
<span id="cb112-12"><a href="#cb112-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-13"><a href="#cb112-13" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(total_significant_count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 613</code></pre>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate into enriched (upregulated in diseased/noma) and depleted (downregulated)</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>noma_enriched_genera <span class="ot">=</span> significant_results <span class="sc">%&gt;%</span> <span class="fu">filter</span>(log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>noma_depleted_genera <span class="ot">=</span> significant_results <span class="sc">%&gt;%</span> <span class="fu">filter</span>(log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>)</span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Count each group</span></span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>enriched_count <span class="ot">=</span> <span class="fu">nrow</span>(noma_enriched_genera)</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>depleted_count <span class="ot">=</span> <span class="fu">nrow</span>(noma_depleted_genera)</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the results table to keep only these genera</span></span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a><span class="co"># If you named it something else, just change it here.</span></span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>selected_results <span class="ot">=</span> significant_results <span class="sc">%&gt;%</span></span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Genus <span class="sc">%in%</span> top_Genus_above_1)</span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>selected_results_noma_enriched <span class="ot">=</span> noma_enriched_genera <span class="sc">%&gt;%</span></span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Genus <span class="sc">%in%</span> top_Genus_above_1)</span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>selected_results_noma_depleted <span class="ot">=</span> noma_depleted_genera <span class="sc">%&gt;%</span></span>
<span id="cb114-19"><a href="#cb114-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Genus <span class="sc">%in%</span> top_Genus_above_1)</span>
<span id="cb114-20"><a href="#cb114-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-21"><a href="#cb114-21" aria-hidden="true" tabindex="-1"></a><span class="co"># View the new, filtered table </span></span>
<span id="cb114-22"><a href="#cb114-22" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(selected_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              Genus   baseMean log2FoldChange     lfcSE       stat       pvalue
1  g__Streptococcus 1202906.49      -4.558247 0.3720108 -12.252998 1.618980e-34
2      g__Treponema   51797.99       4.654275 0.4455440  10.446274 1.523951e-25
3    g__Veillonella  236654.73      -3.897011 0.4989804  -7.809947 5.721202e-15
4        g__Gemella   29807.49      -2.351051 0.3336915  -7.045584 1.846855e-12
5       g__Schaalia  127504.21      -4.845840 0.6891937  -7.031174 2.048035e-12
6         g__Rothia  311158.14      -5.606861 0.8099458  -6.922513 4.437010e-12
7  g__Porphyromonas   18711.58       2.681356 0.3952778   6.783473 1.173208e-11
8    g__Bacteroides   21160.09       2.037286 0.3629309   5.613426 1.983590e-08
9    g__Actinomyces   71349.39      -3.815649 0.8187922  -4.660094 3.160649e-06
10   g__Haemophilus  293639.68      -1.923842 0.5730789  -3.357028 7.878525e-04
11   g__Selenomonas   33035.69       1.024696 0.3916311   2.616483 8.884066e-03
           padj
1  2.630843e-31
2  4.127368e-23
3  2.446567e-13
4  4.479313e-11
5  4.894202e-11
6  9.613521e-11
7  2.413245e-10
8  2.207763e-07
9  2.140023e-05
10 3.232981e-03
11 2.624838e-02</code></pre>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(selected_results_noma_enriched)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             Genus baseMean log2FoldChange     lfcSE      stat       pvalue
1     g__Treponema 51797.99       4.654275 0.4455440 10.446274 1.523951e-25
2 g__Porphyromonas 18711.58       2.681356 0.3952778  6.783473 1.173208e-11
3   g__Bacteroides 21160.09       2.037286 0.3629309  5.613426 1.983590e-08
4   g__Selenomonas 33035.69       1.024696 0.3916311  2.616483 8.884066e-03
          padj
1 4.127368e-23
2 2.413245e-10
3 2.207763e-07
4 2.624838e-02</code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(selected_results_noma_depleted)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             Genus   baseMean log2FoldChange     lfcSE       stat       pvalue
1 g__Streptococcus 1202906.49      -4.558247 0.3720108 -12.252998 1.618980e-34
2   g__Veillonella  236654.73      -3.897011 0.4989804  -7.809947 5.721202e-15
3       g__Gemella   29807.49      -2.351051 0.3336915  -7.045584 1.846855e-12
4      g__Schaalia  127504.21      -4.845840 0.6891937  -7.031174 2.048035e-12
5        g__Rothia  311158.14      -5.606861 0.8099458  -6.922513 4.437010e-12
6   g__Actinomyces   71349.39      -3.815649 0.8187922  -4.660094 3.160649e-06
7   g__Haemophilus  293639.68      -1.923842 0.5730789  -3.357028 7.878525e-04
          padj
1 2.630843e-31
2 2.446567e-13
3 4.479313e-11
4 4.894202e-11
5 9.613521e-11
6 2.140023e-05
7 3.232981e-03</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Count table</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>selected_count <span class="ot">=</span> <span class="fu">nrow</span>(selected_results)</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(selected_count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 11</code></pre>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>selected_enriched_count <span class="ot">=</span> <span class="fu">nrow</span>(selected_results_noma_enriched)</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(selected_enriched_count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4</code></pre>
</div>
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>selected_depleted_count <span class="ot">=</span> <span class="fu">nrow</span>(selected_results_noma_depleted)</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(selected_depleted_count)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7</code></pre>
</div>
</div>
</section>
<section id="plotting-p-values" class="level2">
<h2 class="anchored" data-anchor-id="plotting-p-values">5.7 Plotting p-values</h2>
<p>Plotting for noma</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>sig_res_genus_grtr_thn_zero_above_1pc_relab <span class="ot">=</span> sig_res_genus_grtr_thn_zero <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genera_above_1pc_relab <span class="sc">!=</span> <span class="st">"-other"</span>)</span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>sig_res_genus_grtr_thn_zero_above_1pc_relab <span class="ot">=</span> sig_res_genus_grtr_thn_zero_above_1pc_relab[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Lachnospiraceae_NA"</span>, sig_res_genus_grtr_thn_zero_above_1pc_relab<span class="sc">$</span>genus), ]</span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>padj_plot_noma <span class="ot">=</span> <span class="fu">ggplot</span>(sig_res_genus_grtr_thn_zero_above_1pc_relab, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(genera_above_1pc_relab, <span class="st">`</span><span class="at">padj</span><span class="st">`</span>), <span class="at">y =</span> <span class="st">`</span><span class="at">padj</span><span class="st">`</span>)) <span class="sc">+</span> </span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"#D1352B"</span>) <span class="sc">+</span> </span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">padj</span><span class="st">`</span>, <span class="dv">4</span>)), </span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="fl">1.2</span>, <span class="co"># Adjust horizontal position (slightly outside the bars)</span></span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="co"># Center vertically</span></span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">6</span>, <span class="co"># Adjust text size</span></span>
<span id="cb126-11"><a href="#cb126-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb126-12"><a href="#cb126-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip to make the plot horizontal</span></span>
<span id="cb126-13"><a href="#cb126-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb126-14"><a href="#cb126-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Most significantly different genera above 1% abundance noma related"</span>,</span>
<span id="cb126-15"><a href="#cb126-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Genus"</span>,</span>
<span id="cb126-16"><a href="#cb126-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Adjusted p-value"</span></span>
<span id="cb126-17"><a href="#cb126-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb126-18"><a href="#cb126-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb126-19"><a href="#cb126-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb126-20"><a href="#cb126-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)<span class="co"># Center the title</span></span>
<span id="cb126-21"><a href="#cb126-21" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb126-22"><a href="#cb126-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb126-23"><a href="#cb126-23" aria-hidden="true" tabindex="-1"></a>padj_plot_noma</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_2E.png"</span>, <span class="at">plot =</span>  padj_plot_noma, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plotting for healthy</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot for Healthy</span></span>
<span id="cb128-2"><a href="#cb128-2" aria-hidden="true" tabindex="-1"></a>sig_res_genus_less_thn_zero_above_1pc_relab <span class="ot">=</span> sig_res_genus_less_thn_zero <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genera_above_1pc_relab <span class="sc">!=</span> <span class="st">"-other"</span>)</span>
<span id="cb128-3"><a href="#cb128-3" aria-hidden="true" tabindex="-1"></a>sig_res_genus_less_thn_zero_above_1pc_relab <span class="ot">=</span> sig_res_genus_less_thn_zero_above_1pc_relab[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Clostridiales_Family_XIII._Incertae_Sedis_NA"</span>, sig_res_genus_less_thn_zero_above_1pc_relab<span class="sc">$</span>genus), ]</span>
<span id="cb128-4"><a href="#cb128-4" aria-hidden="true" tabindex="-1"></a>sig_res_genus_less_thn_zero_above_1pc_relab <span class="ot">=</span> sig_res_genus_less_thn_zero_above_1pc_relab[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA"</span>, sig_res_genus_less_thn_zero_above_1pc_relab<span class="sc">$</span>genus), ]</span>
<span id="cb128-5"><a href="#cb128-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-6"><a href="#cb128-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-7"><a href="#cb128-7" aria-hidden="true" tabindex="-1"></a>padj_plot_healthy <span class="ot">=</span> <span class="fu">ggplot</span>(sig_res_genus_less_thn_zero_above_1pc_relab, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(genera_above_1pc_relab, <span class="st">`</span><span class="at">padj</span><span class="st">`</span>), <span class="at">y =</span> <span class="st">`</span><span class="at">padj</span><span class="st">`</span>)) <span class="sc">+</span> </span>
<span id="cb128-8"><a href="#cb128-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span> </span>
<span id="cb128-9"><a href="#cb128-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">padj</span><span class="st">`</span>, <span class="dv">4</span>)), </span>
<span id="cb128-10"><a href="#cb128-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="fl">1.2</span>, <span class="co"># Adjust horizontal position (slightly outside the bars)</span></span>
<span id="cb128-11"><a href="#cb128-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="co"># Center vertically</span></span>
<span id="cb128-12"><a href="#cb128-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> <span class="dv">6</span>, <span class="co"># Adjust text size</span></span>
<span id="cb128-13"><a href="#cb128-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb128-14"><a href="#cb128-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip to make the plot horizontal</span></span>
<span id="cb128-15"><a href="#cb128-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb128-16"><a href="#cb128-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Most significantly different genera above 1% abundance noma related"</span>,</span>
<span id="cb128-17"><a href="#cb128-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Genus"</span>,</span>
<span id="cb128-18"><a href="#cb128-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Adjusted p-value"</span></span>
<span id="cb128-19"><a href="#cb128-19" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span> </span>
<span id="cb128-20"><a href="#cb128-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb128-21"><a href="#cb128-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb128-22"><a href="#cb128-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)<span class="co"># Center the title</span></span>
<span id="cb128-23"><a href="#cb128-23" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb128-24"><a href="#cb128-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb128-25"><a href="#cb128-25" aria-hidden="true" tabindex="-1"></a>padj_plot_healthy</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_2D.png"</span>, <span class="at">plot =</span>  padj_plot_healthy, <span class="at">width =</span> <span class="dv">28</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-normalised-counts-as-boxplots" class="level2">
<h2 class="anchored" data-anchor-id="plotting-normalised-counts-as-boxplots">5.8 Plotting normalised counts as boxplots</h2>
<p>First we create a function</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>plotCountsGGanysig <span class="ot">=</span> <span class="cf">function</span>(dds, gene, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">normalized =</span> <span class="cn">TRUE</span>,  </span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">transform =</span> <span class="cn">TRUE</span>, main, <span class="at">xlab =</span> <span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">FALSE</span>,  </span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">replaced =</span> <span class="cn">FALSE</span>, pc, <span class="at">plot =</span> <span class="st">"point"</span>, <span class="at">text =</span> <span class="cn">TRUE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> <span class="cn">NULL</span>, ...) {  </span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check input gene validity</span></span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(gene) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> (<span class="fu">is.character</span>(gene) <span class="sc">|</span> (<span class="fu">is.numeric</span>(gene) <span class="sc">&amp;</span>  </span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>                                                         (gene <span class="sc">&gt;=</span> <span class="dv">1</span> <span class="sc">&amp;</span> gene <span class="sc">&lt;=</span> <span class="fu">nrow</span>(dds)))))  </span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if all intgroup columns exist in colData</span></span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(intgroup <span class="sc">%in%</span> <span class="fu">names</span>(<span class="fu">colData</span>(dds))))  </span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"all variables in 'intgroup' must be columns of colData"</span>)  </span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># If not returning data, ensure intgroup variables are factors</span></span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>returnData) {  </span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">sapply</span>(intgroup, <span class="cf">function</span>(v) <span class="fu">is</span>(<span class="fu">colData</span>(dds)[[v]], <span class="st">"factor"</span>)))) {  </span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">"all variables in 'intgroup' should be factors, or choose returnData=TRUE and plot manually"</span>)  </span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a>    }  </span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb130-18"><a href="#cb130-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-19"><a href="#cb130-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set pseudo count if not provided</span></span>
<span id="cb130-20"><a href="#cb130-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">missing</span>(pc)) {  </span>
<span id="cb130-21"><a href="#cb130-21" aria-hidden="true" tabindex="-1"></a>    pc <span class="ot">=</span> <span class="cf">if</span> (transform) <span class="fl">0.5</span> <span class="cf">else</span> <span class="dv">0</span>  </span>
<span id="cb130-22"><a href="#cb130-22" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb130-23"><a href="#cb130-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-24"><a href="#cb130-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Estimate size factors if missing</span></span>
<span id="cb130-25"><a href="#cb130-25" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">sizeFactors</span>(dds)) <span class="sc">&amp;</span> <span class="fu">is.null</span>(<span class="fu">normalizationFactors</span>(dds))) {  </span>
<span id="cb130-26"><a href="#cb130-26" aria-hidden="true" tabindex="-1"></a>    dds <span class="ot">=</span> <span class="fu">estimateSizeFactors</span>(dds)  </span>
<span id="cb130-27"><a href="#cb130-27" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb130-28"><a href="#cb130-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-29"><a href="#cb130-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the counts for the gene</span></span>
<span id="cb130-30"><a href="#cb130-30" aria-hidden="true" tabindex="-1"></a>  cnts <span class="ot">=</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> normalized, <span class="at">replaced =</span> replaced)[gene, ]  </span>
<span id="cb130-31"><a href="#cb130-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-32"><a href="#cb130-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate grouping variable</span></span>
<span id="cb130-33"><a href="#cb130-33" aria-hidden="true" tabindex="-1"></a>  group <span class="ot">=</span> <span class="cf">if</span> (<span class="fu">length</span>(intgroup) <span class="sc">==</span> <span class="dv">1</span>) {  </span>
<span id="cb130-34"><a href="#cb130-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">colData</span>(dds)[[intgroup]]  </span>
<span id="cb130-35"><a href="#cb130-35" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (<span class="fu">length</span>(intgroup) <span class="sc">==</span> <span class="dv">2</span>) {  </span>
<span id="cb130-36"><a href="#cb130-36" aria-hidden="true" tabindex="-1"></a>    lvls <span class="ot">=</span> <span class="fu">as.vector</span>(<span class="fu">t</span>(<span class="fu">outer</span>(<span class="fu">levels</span>(<span class="fu">colData</span>(dds)[[intgroup[<span class="dv">1</span>]]]),  </span>
<span id="cb130-37"><a href="#cb130-37" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">levels</span>(<span class="fu">colData</span>(dds)[[intgroup[<span class="dv">2</span>]]]), <span class="cf">function</span>(x, y) <span class="fu">paste</span>(x, y, <span class="at">sep =</span> <span class="st">":"</span>))))  </span>
<span id="cb130-38"><a href="#cb130-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">droplevels</span>(<span class="fu">factor</span>(<span class="fu">apply</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds)[, intgroup, <span class="at">drop =</span> <span class="cn">FALSE</span>]), <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">":"</span>),  </span>
<span id="cb130-39"><a href="#cb130-39" aria-hidden="true" tabindex="-1"></a>                      <span class="at">levels =</span> lvls))  </span>
<span id="cb130-40"><a href="#cb130-40" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {  </span>
<span id="cb130-41"><a href="#cb130-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">factor</span>(<span class="fu">apply</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds)[, intgroup, <span class="at">drop =</span> <span class="cn">FALSE</span>]), <span class="dv">1</span>, paste, <span class="at">collapse =</span> <span class="st">":"</span>))  </span>
<span id="cb130-42"><a href="#cb130-42" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb130-43"><a href="#cb130-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-44"><a href="#cb130-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the data frame with counts, group, and sample names</span></span>
<span id="cb130-45"><a href="#cb130-45" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">count =</span> cnts <span class="sc">+</span> pc, <span class="at">group =</span> group, <span class="at">sample =</span> <span class="fu">colnames</span>(dds), <span class="at">condition =</span> group)  </span>
<span id="cb130-46"><a href="#cb130-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-47"><a href="#cb130-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set log scale if necessary</span></span>
<span id="cb130-48"><a href="#cb130-48" aria-hidden="true" tabindex="-1"></a>  logxy <span class="ot">=</span> <span class="cf">if</span> (transform) <span class="st">"y"</span> <span class="cf">else</span> <span class="st">""</span>  </span>
<span id="cb130-49"><a href="#cb130-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-50"><a href="#cb130-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the plot title</span></span>
<span id="cb130-51"><a href="#cb130-51" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">missing</span>(main)) {  </span>
<span id="cb130-52"><a href="#cb130-52" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">=</span> <span class="cf">if</span> (<span class="fu">is.numeric</span>(gene)) {  </span>
<span id="cb130-53"><a href="#cb130-53" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(dds)[gene]  </span>
<span id="cb130-54"><a href="#cb130-54" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {  </span>
<span id="cb130-55"><a href="#cb130-55" aria-hidden="true" tabindex="-1"></a>      gene  </span>
<span id="cb130-56"><a href="#cb130-56" aria-hidden="true" tabindex="-1"></a>    }  </span>
<span id="cb130-57"><a href="#cb130-57" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb130-58"><a href="#cb130-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-59"><a href="#cb130-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the y-axis label based on normalization</span></span>
<span id="cb130-60"><a href="#cb130-60" aria-hidden="true" tabindex="-1"></a>  ylab <span class="ot">=</span> <span class="fu">ifelse</span>(normalized, <span class="st">"normalized count"</span>, <span class="st">"count"</span>)  </span>
<span id="cb130-61"><a href="#cb130-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-62"><a href="#cb130-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the data if requested</span></span>
<span id="cb130-63"><a href="#cb130-63" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (returnData)  </span>
<span id="cb130-64"><a href="#cb130-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">count =</span> data<span class="sc">$</span>count, <span class="fu">colData</span>(dds)[intgroup]))  </span>
<span id="cb130-65"><a href="#cb130-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-66"><a href="#cb130-66" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the base ggplot object with data and aesthetic mappings</span></span>
<span id="cb130-67"><a href="#cb130-67" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> count, <span class="at">label =</span> sample, <span class="at">color =</span> condition)) <span class="sc">+</span></span>
<span id="cb130-68"><a href="#cb130-68" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> xlab, <span class="at">y =</span> ylab, <span class="at">title =</span> main) <span class="sc">+</span>  <span class="co"># Labels and title</span></span>
<span id="cb130-69"><a href="#cb130-69" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="co"># Clean theme</span></span>
<span id="cb130-70"><a href="#cb130-70" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans =</span> <span class="fu">ifelse</span>(transform, <span class="st">"log10"</span>, <span class="st">"identity"</span>), <span class="at">limits =</span> y_limits) <span class="sc">+</span>  <span class="co"># Apply log transformation if needed</span></span>
<span id="cb130-71"><a href="#cb130-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>)  <span class="co"># Optional: use color brewer for nice color scheme</span></span>
<span id="cb130-72"><a href="#cb130-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-73"><a href="#cb130-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select the type of plot based on the 'plot' argument</span></span>
<span id="cb130-74"><a href="#cb130-74" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"point"</span>) {</span>
<span id="cb130-75"><a href="#cb130-75" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>)</span>
<span id="cb130-76"><a href="#cb130-76" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb130-77"><a href="#cb130-77" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"jitter"</span>) {</span>
<span id="cb130-78"><a href="#cb130-78" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">width =</span> <span class="fl">0.2</span>)</span>
<span id="cb130-79"><a href="#cb130-79" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb130-80"><a href="#cb130-80" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"bar"</span>) {</span>
<span id="cb130-81"><a href="#cb130-81" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"summary"</span>, <span class="at">fun =</span> <span class="st">"mean"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">width =</span> <span class="fl">0.7</span>) <span class="sc">+</span>  <span class="co"># Bar plot with whiskers</span></span>
<span id="cb130-82"><a href="#cb130-82" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_errorbar</span>(<span class="at">stat =</span> <span class="st">"summary"</span>, <span class="at">fun.data =</span> <span class="st">"mean_se"</span>, <span class="at">width =</span> <span class="fl">0.2</span>)</span>
<span id="cb130-83"><a href="#cb130-83" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"violin"</span>) {</span>
<span id="cb130-84"><a href="#cb130-84" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">2</span>, <span class="at">width =</span> <span class="fl">0.2</span>)</span>
<span id="cb130-85"><a href="#cb130-85" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb130-86"><a href="#cb130-86" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot <span class="sc">==</span> <span class="st">"box"</span>) {</span>
<span id="cb130-87"><a href="#cb130-87" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_boxplot</span>()</span>
<span id="cb130-88"><a href="#cb130-88" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust =</span> <span class="dv">0</span>)  <span class="co"># Add text if text = TRUE</span></span>
<span id="cb130-89"><a href="#cb130-89" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb130-90"><a href="#cb130-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Invalid plot type. Choose from 'point', 'jitter', 'bar', 'violin', or 'box'."</span>)</span>
<span id="cb130-91"><a href="#cb130-91" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb130-92"><a href="#cb130-92" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-93"><a href="#cb130-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add significance annotation if requested</span></span>
<span id="cb130-94"><a href="#cb130-94" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (showSignificance) {</span>
<span id="cb130-95"><a href="#cb130-95" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get DESeq2 results for gene using Wald test and BH adjustment</span></span>
<span id="cb130-96"><a href="#cb130-96" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(intgroup, <span class="fu">levels</span>(group)[<span class="dv">1</span>], <span class="fu">levels</span>(group)[<span class="dv">2</span>]), <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb130-97"><a href="#cb130-97" aria-hidden="true" tabindex="-1"></a>    res_gene <span class="ot">=</span> res[gene, ]</span>
<span id="cb130-98"><a href="#cb130-98" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb130-99"><a href="#cb130-99" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check significance and add stars/annotations</span></span>
<span id="cb130-100"><a href="#cb130-100" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.na</span>(res_gene<span class="sc">$</span>padj) <span class="sc">&amp;&amp;</span> res_gene<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb130-101"><a href="#cb130-101" aria-hidden="true" tabindex="-1"></a>      p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fl">1.5</span>, <span class="at">y =</span> <span class="fu">max</span>(data<span class="sc">$</span>count), <span class="at">label =</span> <span class="st">"*"</span>, <span class="at">size =</span> <span class="dv">8</span>)</span>
<span id="cb130-102"><a href="#cb130-102" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb130-103"><a href="#cb130-103" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb130-104"><a href="#cb130-104" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb130-105"><a href="#cb130-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb130-106"><a href="#cb130-106" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define all the genera you are going to plot</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>genera_to_plot <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"g__Treponema"</span>, <span class="st">"g__Streptococcus"</span>, <span class="st">"g__Porphyromonas"</span>, <span class="st">"g__Selenomonas"</span>)</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all the normalized counts for those genera</span></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>all_counts <span class="ot">=</span> <span class="fu">counts</span>(dds_genus, <span class="at">normalized =</span> <span class="cn">TRUE</span>)[genera_to_plot, ]</span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Determine the maximum value for the y-axis (add 10% for buffer)</span></span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a><span class="co"># We add a pseudocount of 0.5 for the log transformation</span></span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a>y_axis_max <span class="ot">=</span> <span class="fu">max</span>(all_counts) <span class="sc">*</span> <span class="fl">1.1</span></span>
<span id="cb131-10"><a href="#cb131-10" aria-hidden="true" tabindex="-1"></a>y_axis_min <span class="ot">=</span> <span class="fl">0.5</span> <span class="co"># A good minimum for log transformed data</span></span>
<span id="cb131-11"><a href="#cb131-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-12"><a href="#cb131-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Store your final range</span></span>
<span id="cb131-13"><a href="#cb131-13" aria-hidden="true" tabindex="-1"></a>overall_y_range <span class="ot">=</span> <span class="fu">c</span>(y_axis_min, y_axis_max)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Next we use the function to plot any highly abundant significantly associated with Noma</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Significant highly abundant for Noma above 1%</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>plot_1 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Treponema"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>plot_2 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Porphyromonas"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-30-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>plot_3 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Bacteroides"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-30-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>plot_4 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Selenomonas"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-30-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(plot_1, plot_2, plot_3, plot_4, <span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-30-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_2C.svg"</span>, </span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">arrangeGrob</span>(plot_1, plot_2, plot_3, plot_4, <span class="at">ncol=</span><span class="dv">2</span>),</span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>,</span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Then we use the function to plot any highly abundant signifcantly asscoiated with healthy global dataset</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Significant highly abundant for healthy above 1%</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>plot_1 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Streptococcus"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>,  <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>plot_2 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Veillonella"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a>plot_3 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Gemella"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a>plot_4 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Schaalia"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>plot_5 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Rothia"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>plot_6 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Actinomyces"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>plot_7 <span class="ot">=</span> <span class="fu">plotCountsGGanysig</span>(dds_genus, <span class="at">gene=</span><span class="st">"g__Haemophilus"</span>, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">y_limits =</span> overall_y_range)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(plot_1, plot_2, plot_3, plot_4,</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a>             plot_5, plot_6, plot_7, <span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-31-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_2B.svg"</span>, </span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">arrangeGrob</span>(plot_1, plot_2, plot_3, plot_4,</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a>             plot_5, plot_6, plot_7, <span class="at">ncol=</span><span class="dv">2</span>),</span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>,</span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="comparing-noma-samples-against-the-three-separate-healthy-cohorts" class="level2">
<h2 class="anchored" data-anchor-id="comparing-noma-samples-against-the-three-separate-healthy-cohorts">5.9 Comparing noma samples against the three separate healthy cohorts</h2>
<section id="add-additional-metadata-relating-to-specific-healthy-controls" class="level3">
<h3 class="anchored" data-anchor-id="add-additional-metadata-relating-to-specific-healthy-controls">5.9.1 Add additional metadata relating to specific healthy controls</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the function to update metadata</span></span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a>update_sample_metadata <span class="ot">=</span> <span class="cf">function</span>(tse_object) {</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract sample names (column names)</span></span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a>  sample_names <span class="ot">=</span> <span class="fu">colnames</span>(tse_object)</span>
<span id="cb147-5"><a href="#cb147-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-6"><a href="#cb147-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the "accession" column based on sample name prefix</span></span>
<span id="cb147-7"><a href="#cb147-7" aria-hidden="true" tabindex="-1"></a>  accession <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^SRS"</span>, sample_names), <span class="st">"SRS"</span>,</span>
<span id="cb147-8"><a href="#cb147-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^SRR"</span>, sample_names), <span class="st">"SRR"</span>,</span>
<span id="cb147-9"><a href="#cb147-9" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^A"</span>, sample_names), <span class="st">"noma"</span>,</span>
<span id="cb147-10"><a href="#cb147-10" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^DRR"</span>, sample_names), <span class="st">"DRR"</span>, <span class="cn">NA</span>))))</span>
<span id="cb147-11"><a href="#cb147-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-12"><a href="#cb147-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the "location" column based on sample name prefix</span></span>
<span id="cb147-13"><a href="#cb147-13" aria-hidden="true" tabindex="-1"></a>  location <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^SRS"</span>, sample_names), <span class="st">"USA"</span>,</span>
<span id="cb147-14"><a href="#cb147-14" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^SRR"</span>, sample_names), <span class="st">"Denmark"</span>,</span>
<span id="cb147-15"><a href="#cb147-15" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^A"</span>, sample_names), <span class="st">"Nigeria"</span>,</span>
<span id="cb147-16"><a href="#cb147-16" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^DRR"</span>, sample_names), <span class="st">"Japan"</span>, <span class="cn">NA</span>))))</span>
<span id="cb147-17"><a href="#cb147-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-18"><a href="#cb147-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a DataFrame with the new metadata columns</span></span>
<span id="cb147-19"><a href="#cb147-19" aria-hidden="true" tabindex="-1"></a>  sample_metadata <span class="ot">=</span> <span class="fu">DataFrame</span>(<span class="at">accession =</span> accession, <span class="at">location =</span> location)</span>
<span id="cb147-20"><a href="#cb147-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(sample_metadata) <span class="ot">=</span> sample_names</span>
<span id="cb147-21"><a href="#cb147-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-22"><a href="#cb147-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add the metadata as colData to the TreeSummarizedExperiment object</span></span>
<span id="cb147-23"><a href="#cb147-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colData</span>(tse_object) <span class="ot">=</span> sample_metadata</span>
<span id="cb147-24"><a href="#cb147-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-25"><a href="#cb147-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the updated object</span></span>
<span id="cb147-26"><a href="#cb147-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(tse_object)</span>
<span id="cb147-27"><a href="#cb147-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb147-28"><a href="#cb147-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-29"><a href="#cb147-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage:</span></span>
<span id="cb147-30"><a href="#cb147-30" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_genus_hc <span class="ot">=</span> <span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>) </span>
<span id="cb147-31"><a href="#cb147-31" aria-hidden="true" tabindex="-1"></a>tse_metaphlan_genus_hc <span class="ot">=</span> <span class="fu">update_sample_metadata</span>(tse_metaphlan_genus_hc)</span>
<span id="cb147-32"><a href="#cb147-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-33"><a href="#cb147-33" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan_genus_hc)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    accession location
A10      noma  Nigeria
A11      noma  Nigeria
A13      noma  Nigeria
A14      noma  Nigeria
A16      noma  Nigeria
A17      noma  Nigeria</code></pre>
</div>
</div>
</section>
<section id="convert-data" class="level3">
<h3 class="anchored" data-anchor-id="convert-data">5.9.2 Convert data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add this DataFrame as colData to your TreeSummarizedExperiment object</span></span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tse_metaphlan_genus_hc) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 37 rows and 2 columns
            accession    location
          &lt;character&gt; &lt;character&gt;
A10              noma     Nigeria
A11              noma     Nigeria
A13              noma     Nigeria
A14              noma     Nigeria
A16              noma     Nigeria
...               ...         ...
SRS013942         SRS         USA
SRS014468         SRS         USA
SRS014692         SRS         USA
SRS015055         SRS         USA
SRS019120         SRS         USA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(<span class="fu">colData</span>(tse_metaphlan_genus_hc))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 4 rows and 2 columns
             accession    location
           &lt;character&gt; &lt;character&gt;
A10               noma     Nigeria
DRR214959          DRR       Japan
SRR5892208         SRR     Denmark
SRS013942          SRS         USA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Use makePhyloseqFromTreeSE from Miaverse</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>phyloseq_metaphlan_hc <span class="ot">=</span> <span class="fu">makePhyloseqFromTreeSE</span>(tse_metaphlan_genus_hc)</span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>deseq2_metaphlan_hc <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">phyloseq_to_deseq2</span>(phyloseq_metaphlan_hc, <span class="at">design =</span> <span class="sc">~</span> accession)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
</div>
</section>
<section id="run-differential-analysis-between-seperate-datasets" class="level3">
<h3 class="anchored" data-anchor-id="run-differential-analysis-between-seperate-datasets">5.9.3 Run differential analysis between seperate datasets</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>dds_hc <span class="ot">=</span> deseq2_metaphlan_hc</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dds_hc) <span class="ot">=</span> <span class="er">~</span> accession  <span class="co"># Replace with your column name for condition</span></span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Run DESeq2 analysis</span></span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a>dds_hc <span class="ot">=</span> <span class="fu">DESeq</span>(dds_hc)</span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Clean up genus names for dds</span></span>
<span id="cb156-9"><a href="#cb156-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(dds_hc) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(dds_hc))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extract-results-for-disease-state" class="level3">
<h3 class="anchored" data-anchor-id="extract-results-for-disease-state">5.9.4 Extract results for disease state</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_metaphlan_genus_hc)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           accession location
A10             noma  Nigeria
DRR214959        DRR    Japan
SRR5892208       SRR  Denmark
SRS013942        SRS      USA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract results for diseased vs healthy</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>res_SRR_noma <span class="ot">=</span> <span class="fu">results</span>(dds_hc, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"accession"</span>, <span class="st">"noma"</span>, <span class="st">"SRR"</span>))</span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>res_DRR_noma <span class="ot">=</span> <span class="fu">results</span>(dds_hc, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"accession"</span>, <span class="st">"noma"</span>, <span class="st">"DRR"</span>))</span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>res_SRS_noma <span class="ot">=</span> <span class="fu">results</span>(dds_hc, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"accession"</span>, <span class="st">"noma"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-counts-of-genera-between-diseased-and-healthy" class="level3">
<h3 class="anchored" data-anchor-id="plot-counts-of-genera-between-diseased-and-healthy">5.9.5 Plot counts of genera between diseased and healthy</h3>
<p>Here we introduce another function which plots lines for significance</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a>plotCountsGGsigline <span class="ot">=</span> <span class="cf">function</span>(dds, gene, <span class="at">intgroup =</span> <span class="st">"condition"</span>, <span class="at">normalized =</span> <span class="cn">TRUE</span>,</span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">transform =</span> <span class="cn">TRUE</span>, main, <span class="at">xlab =</span> <span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">FALSE</span>,</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">replaced =</span> <span class="cn">FALSE</span>, pc, <span class="at">plot =</span> <span class="st">"point"</span>, <span class="at">text =</span> <span class="cn">TRUE</span>,</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>                               <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="fl">0.1</span>,</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a>                               <span class="at">groupOrder =</span> <span class="cn">NULL</span>, ...) {</span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># input checks</span></span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopifnot</span>(<span class="fu">length</span>(gene)<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;&amp;</span> (<span class="fu">is.character</span>(gene) <span class="sc">||</span></span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a>                                  (<span class="fu">is.numeric</span>(gene) <span class="sc">&amp;&amp;</span> gene<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;&amp;</span> gene<span class="sc">&lt;=</span><span class="fu">nrow</span>(dds))))</span>
<span id="cb160-10"><a href="#cb160-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(intgroup <span class="sc">%in%</span> <span class="fu">names</span>(<span class="fu">colData</span>(dds))))</span>
<span id="cb160-11"><a href="#cb160-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"all variables in 'intgroup' must be columns of colData"</span>)</span>
<span id="cb160-12"><a href="#cb160-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>returnData) {</span>
<span id="cb160-13"><a href="#cb160-13" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">sapply</span>(intgroup, <span class="cf">function</span>(v) <span class="fu">is</span>(<span class="fu">colData</span>(dds)[[v]], <span class="st">"factor"</span>))))</span>
<span id="cb160-14"><a href="#cb160-14" aria-hidden="true" tabindex="-1"></a>      <span class="fu">stop</span>(<span class="st">"all variables in 'intgroup' should be factors, or choose returnData=TRUE"</span>)</span>
<span id="cb160-15"><a href="#cb160-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-16"><a href="#cb160-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-17"><a href="#cb160-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pseudo-count</span></span>
<span id="cb160-18"><a href="#cb160-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">missing</span>(pc)) pc <span class="ot">=</span> <span class="cf">if</span> (transform) <span class="fl">0.5</span> <span class="cf">else</span> <span class="dv">0</span></span>
<span id="cb160-19"><a href="#cb160-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-20"><a href="#cb160-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ensure size factors</span></span>
<span id="cb160-21"><a href="#cb160-21" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(<span class="fu">sizeFactors</span>(dds)) <span class="sc">&amp;&amp;</span> <span class="fu">is.null</span>(<span class="fu">normalizationFactors</span>(dds)))</span>
<span id="cb160-22"><a href="#cb160-22" aria-hidden="true" tabindex="-1"></a>    dds <span class="ot">=</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb160-23"><a href="#cb160-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-24"><a href="#cb160-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-25"><a href="#cb160-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># extract counts + grouping</span></span>
<span id="cb160-26"><a href="#cb160-26" aria-hidden="true" tabindex="-1"></a>  cnts  <span class="ot">=</span> <span class="fu">counts</span>(dds, <span class="at">normalized=</span>normalized, <span class="at">replaced=</span>replaced)[gene,]</span>
<span id="cb160-27"><a href="#cb160-27" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(intgroup)<span class="sc">==</span><span class="dv">1</span>) {</span>
<span id="cb160-28"><a href="#cb160-28" aria-hidden="true" tabindex="-1"></a>    group <span class="ot">=</span> <span class="fu">colData</span>(dds)[[intgroup]]</span>
<span id="cb160-29"><a href="#cb160-29" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb160-30"><a href="#cb160-30" aria-hidden="true" tabindex="-1"></a>    lvls  <span class="ot">=</span> <span class="fu">as.vector</span>(<span class="fu">t</span>(<span class="fu">outer</span>(<span class="fu">levels</span>(<span class="fu">colData</span>(dds)[[intgroup[<span class="dv">1</span>]]]),</span>
<span id="cb160-31"><a href="#cb160-31" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">levels</span>(<span class="fu">colData</span>(dds)[[intgroup[<span class="dv">2</span>]]]),</span>
<span id="cb160-32"><a href="#cb160-32" aria-hidden="true" tabindex="-1"></a>                               paste, <span class="at">sep=</span><span class="st">":"</span>)))</span>
<span id="cb160-33"><a href="#cb160-33" aria-hidden="true" tabindex="-1"></a>    group <span class="ot">=</span> <span class="fu">droplevels</span>(<span class="fu">factor</span>(</span>
<span id="cb160-34"><a href="#cb160-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">apply</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(dds)[,intgroup]),<span class="dv">1</span>,paste,<span class="at">collapse=</span><span class="st">":"</span>),</span>
<span id="cb160-35"><a href="#cb160-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels=</span>lvls))</span>
<span id="cb160-36"><a href="#cb160-36" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-37"><a href="#cb160-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-38"><a href="#cb160-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set order of groups on the x-axis if specified</span></span>
<span id="cb160-39"><a href="#cb160-39" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(groupOrder)) {</span>
<span id="cb160-40"><a href="#cb160-40" aria-hidden="true" tabindex="-1"></a>    group <span class="ot">=</span> <span class="fu">factor</span>(group, <span class="at">levels =</span> groupOrder)</span>
<span id="cb160-41"><a href="#cb160-41" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-42"><a href="#cb160-42" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-43"><a href="#cb160-43" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">count=</span>cnts<span class="sc">+</span>pc,</span>
<span id="cb160-44"><a href="#cb160-44" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group=</span>group,</span>
<span id="cb160-45"><a href="#cb160-45" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sample=</span><span class="fu">colnames</span>(dds),</span>
<span id="cb160-46"><a href="#cb160-46" aria-hidden="true" tabindex="-1"></a>                     <span class="at">condition=</span>group)</span>
<span id="cb160-47"><a href="#cb160-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-48"><a href="#cb160-48" aria-hidden="true" tabindex="-1"></a>  <span class="co"># axis labels</span></span>
<span id="cb160-49"><a href="#cb160-49" aria-hidden="true" tabindex="-1"></a>  ylab <span class="ot">=</span> <span class="fu">ifelse</span>(normalized, <span class="st">"normalized count"</span>, <span class="st">"count"</span>)</span>
<span id="cb160-50"><a href="#cb160-50" aria-hidden="true" tabindex="-1"></a>  logxy <span class="ot">=</span> <span class="fu">ifelse</span>(transform, <span class="st">"y"</span>, <span class="st">""</span>)</span>
<span id="cb160-51"><a href="#cb160-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-52"><a href="#cb160-52" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">missing</span>(main)) {</span>
<span id="cb160-53"><a href="#cb160-53" aria-hidden="true" tabindex="-1"></a>    main <span class="ot">=</span> <span class="cf">if</span> (<span class="fu">is.numeric</span>(gene)) <span class="fu">rownames</span>(dds)[gene] <span class="cf">else</span> gene</span>
<span id="cb160-54"><a href="#cb160-54" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-55"><a href="#cb160-55" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (returnData) {</span>
<span id="cb160-56"><a href="#cb160-56" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">data.frame</span>(<span class="at">count=</span>data<span class="sc">$</span>count, <span class="fu">colData</span>(dds)[intgroup]))</span>
<span id="cb160-57"><a href="#cb160-57" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-58"><a href="#cb160-58" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-59"><a href="#cb160-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># base ggplot</span></span>
<span id="cb160-60"><a href="#cb160-60" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">=</span> <span class="fu">ggplot</span>(data, <span class="fu">aes</span>(<span class="at">x=</span>group,<span class="at">y=</span>count,<span class="at">label=</span>sample,</span>
<span id="cb160-61"><a href="#cb160-61" aria-hidden="true" tabindex="-1"></a>                        <span class="at">color=</span>condition,<span class="at">group=</span>group)) <span class="sc">+</span></span>
<span id="cb160-62"><a href="#cb160-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span>xlab, <span class="at">y=</span>ylab, <span class="at">title=</span>main) <span class="sc">+</span></span>
<span id="cb160-63"><a href="#cb160-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb160-64"><a href="#cb160-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">trans=</span><span class="fu">ifelse</span>(transform,<span class="st">"log10"</span>,<span class="st">"identity"</span>)) <span class="sc">+</span></span>
<span id="cb160-65"><a href="#cb160-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">"Set1"</span>)</span>
<span id="cb160-66"><a href="#cb160-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-67"><a href="#cb160-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># choose geom</span></span>
<span id="cb160-68"><a href="#cb160-68" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (plot<span class="sc">==</span><span class="st">"point"</span>) {</span>
<span id="cb160-69"><a href="#cb160-69" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>)</span>
<span id="cb160-70"><a href="#cb160-70" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust=</span><span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust=</span><span class="dv">0</span>)</span>
<span id="cb160-71"><a href="#cb160-71" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot<span class="sc">==</span><span class="st">"jitter"</span>) {</span>
<span id="cb160-72"><a href="#cb160-72" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_jitter</span>(<span class="at">size=</span><span class="dv">3</span>,<span class="at">width=</span><span class="fl">0.2</span>)</span>
<span id="cb160-73"><a href="#cb160-73" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust=</span><span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust=</span><span class="dv">0</span>)</span>
<span id="cb160-74"><a href="#cb160-74" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot<span class="sc">==</span><span class="st">"bar"</span>) {</span>
<span id="cb160-75"><a href="#cb160-75" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"summary"</span>, <span class="at">fun=</span><span class="st">"mean"</span>,</span>
<span id="cb160-76"><a href="#cb160-76" aria-hidden="true" tabindex="-1"></a>                      <span class="at">position=</span><span class="st">"dodge"</span>, <span class="at">width=</span><span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb160-77"><a href="#cb160-77" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_errorbar</span>(<span class="at">stat=</span><span class="st">"summary"</span>, <span class="at">fun.data=</span><span class="st">"mean_se"</span>, <span class="at">width=</span><span class="fl">0.2</span>)</span>
<span id="cb160-78"><a href="#cb160-78" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot<span class="sc">==</span><span class="st">"violin"</span>) {</span>
<span id="cb160-79"><a href="#cb160-79" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_violin</span>(<span class="at">trim=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb160-80"><a href="#cb160-80" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_jitter</span>(<span class="at">size=</span><span class="dv">2</span>,<span class="at">width=</span><span class="fl">0.2</span>)</span>
<span id="cb160-81"><a href="#cb160-81" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust=</span><span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust=</span><span class="dv">0</span>)</span>
<span id="cb160-82"><a href="#cb160-82" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (plot<span class="sc">==</span><span class="st">"box"</span>) {</span>
<span id="cb160-83"><a href="#cb160-83" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb160-84"><a href="#cb160-84" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_jitter</span>(<span class="at">size=</span><span class="dv">2</span>,<span class="at">width=</span><span class="fl">0.2</span>)</span>
<span id="cb160-85"><a href="#cb160-85" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (text) p <span class="ot">=</span> p <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">hjust=</span><span class="sc">-</span><span class="fl">0.2</span>, <span class="at">vjust=</span><span class="dv">0</span>)</span>
<span id="cb160-86"><a href="#cb160-86" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb160-87"><a href="#cb160-87" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">"Invalid plot type. Choose from 'point','jitter','bar','violin','box'."</span>)</span>
<span id="cb160-88"><a href="#cb160-88" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-89"><a href="#cb160-89" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-90"><a href="#cb160-90" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add significance lines &amp; stars</span></span>
<span id="cb160-91"><a href="#cb160-91" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (showSignificance) {</span>
<span id="cb160-92"><a href="#cb160-92" aria-hidden="true" tabindex="-1"></a>    max_y    <span class="ot">=</span> <span class="fu">max</span>(data<span class="sc">$</span>count)</span>
<span id="cb160-93"><a href="#cb160-93" aria-hidden="true" tabindex="-1"></a>    step     <span class="ot">=</span> max_y <span class="sc">*</span> lineSpacing</span>
<span id="cb160-94"><a href="#cb160-94" aria-hidden="true" tabindex="-1"></a>    counter  <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb160-95"><a href="#cb160-95" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb160-96"><a href="#cb160-96" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">nlevels</span>(group) <span class="sc">-</span> <span class="dv">1</span>)) {</span>
<span id="cb160-97"><a href="#cb160-97" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (j <span class="cf">in</span> (i <span class="sc">+</span> <span class="dv">1</span>)<span class="sc">:</span><span class="fu">nlevels</span>(group)) {</span>
<span id="cb160-98"><a href="#cb160-98" aria-hidden="true" tabindex="-1"></a>        contrast <span class="ot">=</span> <span class="fu">c</span>(intgroup, <span class="fu">levels</span>(group)[i], <span class="fu">levels</span>(group)[j])</span>
<span id="cb160-99"><a href="#cb160-99" aria-hidden="true" tabindex="-1"></a>        res      <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> contrast, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb160-100"><a href="#cb160-100" aria-hidden="true" tabindex="-1"></a>        pv       <span class="ot">=</span> <span class="fu">as.numeric</span>(res[gene, <span class="st">"padj"</span>])</span>
<span id="cb160-101"><a href="#cb160-101" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb160-102"><a href="#cb160-102" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Determine label and color</span></span>
<span id="cb160-103"><a href="#cb160-103" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">is.na</span>(pv) <span class="sc">||</span> pv <span class="sc">&gt;</span> <span class="fl">0.05</span>) {</span>
<span id="cb160-104"><a href="#cb160-104" aria-hidden="true" tabindex="-1"></a>          lab <span class="ot">=</span> <span class="st">"ns"</span></span>
<span id="cb160-105"><a href="#cb160-105" aria-hidden="true" tabindex="-1"></a>          col <span class="ot">=</span> <span class="st">"grey70"</span></span>
<span id="cb160-106"><a href="#cb160-106" aria-hidden="true" tabindex="-1"></a>          size <span class="ot">=</span> <span class="dv">3</span></span>
<span id="cb160-107"><a href="#cb160-107" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (pv <span class="sc">&lt;=</span> <span class="fl">0.001</span>) {</span>
<span id="cb160-108"><a href="#cb160-108" aria-hidden="true" tabindex="-1"></a>          lab <span class="ot">=</span> <span class="st">"***"</span></span>
<span id="cb160-109"><a href="#cb160-109" aria-hidden="true" tabindex="-1"></a>          col <span class="ot">=</span> <span class="st">"black"</span></span>
<span id="cb160-110"><a href="#cb160-110" aria-hidden="true" tabindex="-1"></a>          size <span class="ot">=</span> <span class="dv">6</span></span>
<span id="cb160-111"><a href="#cb160-111" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (pv <span class="sc">&lt;=</span> <span class="fl">0.01</span>) {</span>
<span id="cb160-112"><a href="#cb160-112" aria-hidden="true" tabindex="-1"></a>          lab <span class="ot">=</span> <span class="st">"**"</span></span>
<span id="cb160-113"><a href="#cb160-113" aria-hidden="true" tabindex="-1"></a>          col <span class="ot">=</span> <span class="st">"black"</span></span>
<span id="cb160-114"><a href="#cb160-114" aria-hidden="true" tabindex="-1"></a>          size <span class="ot">=</span> <span class="dv">6</span></span>
<span id="cb160-115"><a href="#cb160-115" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb160-116"><a href="#cb160-116" aria-hidden="true" tabindex="-1"></a>          lab <span class="ot">=</span> <span class="st">"*"</span></span>
<span id="cb160-117"><a href="#cb160-117" aria-hidden="true" tabindex="-1"></a>          col <span class="ot">=</span> <span class="st">"black"</span></span>
<span id="cb160-118"><a href="#cb160-118" aria-hidden="true" tabindex="-1"></a>          size <span class="ot">=</span> <span class="dv">6</span></span>
<span id="cb160-119"><a href="#cb160-119" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb160-120"><a href="#cb160-120" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb160-121"><a href="#cb160-121" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Positioning</span></span>
<span id="cb160-122"><a href="#cb160-122" aria-hidden="true" tabindex="-1"></a>        counter <span class="ot">=</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb160-123"><a href="#cb160-123" aria-hidden="true" tabindex="-1"></a>        y_line  <span class="ot">=</span> max_y <span class="sc">+</span> step <span class="sc">*</span> counter</span>
<span id="cb160-124"><a href="#cb160-124" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb160-125"><a href="#cb160-125" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Add line and label</span></span>
<span id="cb160-126"><a href="#cb160-126" aria-hidden="true" tabindex="-1"></a>        p <span class="ot">=</span> p <span class="sc">+</span></span>
<span id="cb160-127"><a href="#cb160-127" aria-hidden="true" tabindex="-1"></a>          <span class="fu">annotate</span>(<span class="st">"segment"</span>,</span>
<span id="cb160-128"><a href="#cb160-128" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x =</span> i, <span class="at">xend =</span> j,</span>
<span id="cb160-129"><a href="#cb160-129" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> y_line, <span class="at">yend =</span> y_line,</span>
<span id="cb160-130"><a href="#cb160-130" aria-hidden="true" tabindex="-1"></a>                   <span class="at">colour =</span> col) <span class="sc">+</span></span>
<span id="cb160-131"><a href="#cb160-131" aria-hidden="true" tabindex="-1"></a>          <span class="fu">annotate</span>(<span class="st">"text"</span>,</span>
<span id="cb160-132"><a href="#cb160-132" aria-hidden="true" tabindex="-1"></a>                   <span class="at">x =</span> <span class="fu">mean</span>(<span class="fu">c</span>(i, j)),</span>
<span id="cb160-133"><a href="#cb160-133" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> y_line <span class="sc">+</span> (step <span class="sc">*</span> <span class="fl">0.05</span>),</span>
<span id="cb160-134"><a href="#cb160-134" aria-hidden="true" tabindex="-1"></a>                   <span class="at">label =</span> lab,</span>
<span id="cb160-135"><a href="#cb160-135" aria-hidden="true" tabindex="-1"></a>                   <span class="at">size =</span> size,</span>
<span id="cb160-136"><a href="#cb160-136" aria-hidden="true" tabindex="-1"></a>                   <span class="at">colour =</span> col)</span>
<span id="cb160-137"><a href="#cb160-137" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb160-138"><a href="#cb160-138" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb160-139"><a href="#cb160-139" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb160-140"><a href="#cb160-140" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb160-141"><a href="#cb160-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb160-142"><a href="#cb160-142" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now plot the different genera</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>plot_1 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Treponema"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a>plot_2 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Bacteroides"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>plot_3 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Porphyromonas"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a>plot_4 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Selenomonas"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>plot_5 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Prevotella"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>plot_6 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Streptococcus"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-6.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>plot_7 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Rothia"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-7.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a>plot_8 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Veillonella"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-8.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a>plot_9 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Gemella"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-9.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a>plot_10 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Schaalia"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-10.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a>plot_11 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Actinomyces"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-11.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a>plot_12 <span class="ot">=</span> <span class="fu">plotCountsGGsigline</span>(dds_hc, <span class="at">gene=</span> <span class="st">"g__Haemophilus"</span>, <span class="at">intgroup =</span> <span class="st">"accession"</span>, <span class="at">plot =</span> <span class="st">"box"</span>, <span class="at">text =</span> <span class="cn">FALSE</span>, <span class="at">showSignificance =</span> <span class="cn">TRUE</span>, <span class="at">lineSpacing =</span> <span class="dv">5</span>, <span class="at">groupOrder =</span> <span class="fu">c</span>(<span class="st">"noma"</span>, <span class="st">"DRR"</span>, <span class="st">"SRR"</span>, <span class="st">"SRS"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-12.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(plot_1, plot_2, plot_3, plot_4, plot_5,</span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a>             plot_6, plot_7, plot_8, plot_9, plot_10,</span>
<span id="cb173-3"><a href="#cb173-3" aria-hidden="true" tabindex="-1"></a>             plot_11, plot_12, <span class="at">ncol=</span><span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-37-13.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Supplementary_Figure_2.png"</span>, </span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a><span class="fu">arrangeGrob</span>(plot_1, plot_2, plot_3, plot_4, plot_5,</span>
<span id="cb174-3"><a href="#cb174-3" aria-hidden="true" tabindex="-1"></a>             plot_6, plot_7, plot_8, plot_9, plot_10,</span>
<span id="cb174-4"><a href="#cb174-4" aria-hidden="true" tabindex="-1"></a>             plot_11, plot_12, <span class="at">ncol=</span><span class="dv">3</span>),</span>
<span id="cb174-5"><a href="#cb174-5" aria-hidden="true" tabindex="-1"></a><span class="at">width =</span> <span class="dv">20</span>,</span>
<span id="cb174-6"><a href="#cb174-6" aria-hidden="true" tabindex="-1"></a><span class="at">height =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="random-forests" class="level1">
<h1>6. Random Forests</h1>
<p>Random forest (RF) classifiers were applied based on 500 estimator trees to identify key predictors of microbial community composition and their association with noma or healthy states.</p>
<p>We also ran a cross validation of the dataset taking 20% of samples as test data and 80% as training data. This was used to assess the predictive power of the RF classifier, assessed as an area under the receiver operating characteristic (ROC) curve (AUC) between 0 and 1.</p>
<p>A permutation test (n = 1000) was conducted to evaluate the statistical significance of the observed classification accuracy, this allowed for a null distribution of accuracy to be plotted. The observed accuracy of the random forest model was overlayed on the null distribution.</p>
<section id="preparing-the-data-2" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data-2">6.1 Preparing the data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract metadata and OTU table</span></span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a>  phyloseq_obj <span class="ot">=</span> phyloseq_noma_esd</span>
<span id="cb175-3"><a href="#cb175-3" aria-hidden="true" tabindex="-1"></a>  metadata <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">sample_data</span>(phyloseq_obj))</span>
<span id="cb175-4"><a href="#cb175-4" aria-hidden="true" tabindex="-1"></a>  otu_data <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">otu_table</span>(phyloseq_obj))</span>
<span id="cb175-5"><a href="#cb175-5" aria-hidden="true" tabindex="-1"></a>  otu_data <span class="ot">=</span> <span class="fu">t</span>(otu_data)  <span class="co"># Transpose so samples are rows and species are columns</span></span>
<span id="cb175-6"><a href="#cb175-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb175-7"><a href="#cb175-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine OTU data with metadata</span></span>
<span id="cb175-8"><a href="#cb175-8" aria-hidden="true" tabindex="-1"></a>  ml_data <span class="ot">=</span> <span class="fu">cbind</span>(metadata, otu_data)</span>
<span id="cb175-9"><a href="#cb175-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb175-10"><a href="#cb175-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure the metadata column you're predicting (e.g., condition) is a factor</span></span>
<span id="cb175-11"><a href="#cb175-11" aria-hidden="true" tabindex="-1"></a>  ml_data<span class="sc">$</span>condition <span class="ot">=</span> <span class="fu">as.factor</span>(ml_data<span class="sc">$</span>condition)  <span class="co"># Replace "condition" with your actual column</span></span>
<span id="cb175-12"><a href="#cb175-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb175-13"><a href="#cb175-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove columns containing NA</span></span>
<span id="cb175-14"><a href="#cb175-14" aria-hidden="true" tabindex="-1"></a>  ml_data_clean <span class="ot">=</span> ml_data[, <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colnames</span>(ml_data))]</span>
<span id="cb175-15"><a href="#cb175-15" aria-hidden="true" tabindex="-1"></a>  ml_data_clean <span class="ot">=</span> ml_data[, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA"</span>, <span class="fu">colnames</span>(ml_data))]</span>
<span id="cb175-16"><a href="#cb175-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb175-17"><a href="#cb175-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb175-18"><a href="#cb175-18" aria-hidden="true" tabindex="-1"></a>  rf_model_cv <span class="ot">=</span> <span class="fu">randomForest</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> ml_data_clean, <span class="at">ntree =</span> <span class="dv">500</span>, <span class="at">importance =</span> <span class="cn">TRUE</span>)</span>
<span id="cb175-19"><a href="#cb175-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(rf_model_cv)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = condition ~ ., data = ml_data_clean, ntree = 500,      importance = TRUE) 
               Type of random forest: classification
                     Number of trees: 500
No. of variables tried at each split: 41

        OOB estimate of  error rate: 2.7%
Confusion matrix:
         diseased healthy class.error
diseased       16       1  0.05882353
healthy         0      20  0.00000000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a>  importance_rf_model_cv <span class="ot">=</span> <span class="fu">importance</span>(rf_model_cv)</span>
<span id="cb177-2"><a href="#cb177-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb177-3"><a href="#cb177-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb177-4"><a href="#cb177-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Delete everything from start to Genus</span></span>
<span id="cb177-5"><a href="#cb177-5" aria-hidden="true" tabindex="-1"></a>  ml_data_col_names <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">".*_g__"</span>,<span class="st">""</span>,<span class="fu">colnames</span>(ml_data))</span>
<span id="cb177-6"><a href="#cb177-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add Genus back in </span></span>
<span id="cb177-7"><a href="#cb177-7" aria-hidden="true" tabindex="-1"></a>  ml_data_clean <span class="ot">=</span> <span class="fu">paste0</span>(<span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"g__"</span>, <span class="fu">length</span>(ml_data_col_names)), ml_data_col_names))</span>
<span id="cb177-8"><a href="#cb177-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Delete the space introduced by this </span></span>
<span id="cb177-9"><a href="#cb177-9" aria-hidden="true" tabindex="-1"></a>  ml_data_col_names <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">" "</span>,<span class="st">""</span>,ml_data_col_names)</span>
<span id="cb177-10"><a href="#cb177-10" aria-hidden="true" tabindex="-1"></a>  ml_data_col_names <span class="ot">=</span> <span class="fu">sub</span>(<span class="st">"g__condition"</span>,<span class="st">"condition"</span>,ml_data_col_names)</span>
<span id="cb177-11"><a href="#cb177-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(ml_data_col_names)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "condition"             "g__Coprothermobacter"  "g__Caldisericum"      
[4] "g__Desulfurispirillum" "g__Endomicrobium"      "g__Elusimicrobium"    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a>  ml_data_clean <span class="ot">=</span> ml_data</span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(ml_data_clean) <span class="ot">=</span> ml_data_col_names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="running-random-forests" class="level2">
<h2 class="anchored" data-anchor-id="running-random-forests">6.2 running random forests</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run Random Forest to classify samples based on 'condition'</span></span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>  rf_model_condition <span class="ot">=</span> <span class="fu">randomForest</span>(condition <span class="sc">~</span> ., <span class="at">data =</span> ml_data_clean, <span class="at">ntree =</span> <span class="dv">500</span>, <span class="at">importance =</span> <span class="cn">TRUE</span>)</span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(rf_model_condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = condition ~ ., data = ml_data_clean, ntree = 500,      importance = TRUE) 
               Type of random forest: classification
                     Number of trees: 500
No. of variables tried at each split: 41

        OOB estimate of  error rate: 2.7%
Confusion matrix:
         diseased healthy class.error
diseased       16       1  0.05882353
healthy         0      20  0.00000000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Load the rf model object from paper</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>  rf_model_condition <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"../data/rf_model_condition.rds"</span>)</span>
<span id="cb182-3"><a href="#cb182-3" aria-hidden="true" tabindex="-1"></a>  importance_rf_model <span class="ot">=</span> <span class="fu">importance</span>(rf_model_condition)</span>
<span id="cb182-4"><a href="#cb182-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(importance_rf_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                diseased healthy MeanDecreaseAccuracy
Coprothermobacter               0.000000 0.00000             0.000000
Caldisericum                    1.663615 1.95912             1.874178
Desulfurispirillum              0.000000 0.00000             0.000000
NA_p__Elusimicrobia_NA_NA_NA_NA 0.000000 0.00000             0.000000
Endomicrobium                   0.000000 0.00000             0.000000
Elusimicrobium                  0.000000 0.00000             0.000000
                                MeanDecreaseGini
Coprothermobacter                    0.000000000
Caldisericum                         0.099614496
Desulfurispirillum                   0.000000000
NA_p__Elusimicrobia_NA_NA_NA_NA      0.000000000
Endomicrobium                        0.000000000
Elusimicrobium                       0.003764706</code></pre>
</div>
</div>
</section>
<section id="plot-variable-importance" class="level2">
<h2 class="anchored" data-anchor-id="plot-variable-importance">6.3 Plot variable importance</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot Variable Importance</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># To plot the variable importance, use the %IncMSE (or MeanDecreaseAccuracy) </span></span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># from the importance output. Below is an example of how to create a bar plot </span></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># of the top 10 most important predictors:</span></span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract variable importance and convert to a data frame</span></span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(importance_rf_model)</span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">=</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(importance_df, <span class="st">"genus"</span>)</span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">=</span> importance_df <span class="sc">%&gt;%</span></span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb184-14"><a href="#cb184-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">50</span>)  <span class="co"># Select top 10 most important predictors</span></span>
<span id="cb184-15"><a href="#cb184-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-16"><a href="#cb184-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># select by any above 1.7</span></span>
<span id="cb184-17"><a href="#cb184-17" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">=</span> importance_df <span class="sc">%&gt;%</span></span>
<span id="cb184-18"><a href="#cb184-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span><span class="sc">&gt;=</span> <span class="fl">1.7</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb184-19"><a href="#cb184-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span>))</span>
<span id="cb184-20"><a href="#cb184-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-21"><a href="#cb184-21" aria-hidden="true" tabindex="-1"></a>  importance_df_ggplot <span class="ot">=</span> importance_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA"</span>, genus))</span>
<span id="cb184-22"><a href="#cb184-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-23"><a href="#cb184-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot the top 10 important predictors</span></span>
<span id="cb184-24"><a href="#cb184-24" aria-hidden="true" tabindex="-1"></a>  gp <span class="ot">=</span> <span class="fu">ggplot</span>(importance_df_ggplot, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(genus, <span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span>), <span class="at">y =</span> <span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb184-25"><a href="#cb184-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb184-26"><a href="#cb184-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip to make the plot horizontal</span></span>
<span id="cb184-27"><a href="#cb184-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(</span>
<span id="cb184-28"><a href="#cb184-28" aria-hidden="true" tabindex="-1"></a>      <span class="at">title =</span> <span class="st">"Top Most Important Genus Predictors in Random Forest"</span>,</span>
<span id="cb184-29"><a href="#cb184-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="st">"Genus"</span>,</span>
<span id="cb184-30"><a href="#cb184-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="st">"Increase in Mean Squared Error (%IncMSE)"</span></span>
<span id="cb184-31"><a href="#cb184-31" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb184-32"><a href="#cb184-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>)</span>
<span id="cb184-33"><a href="#cb184-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb184-34"><a href="#cb184-34" aria-hidden="true" tabindex="-1"></a>  gp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select only the top genera with a relative abundance above 1%</span></span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a>  top_Genus_pc<span class="sc">$</span>top_Genus_above_1 <span class="ot">=</span> <span class="fu">ifelse</span>(top_Genus_pc<span class="sc">$</span>top_Genus_percentage <span class="sc">&gt;</span> <span class="fl">0.1</span>, top_Genus_pc<span class="sc">$</span>top_Genus, <span class="st">"-other"</span>)</span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a>  top_Genus_above_1 <span class="ot">=</span> <span class="fu">unique</span>(top_Genus_pc<span class="sc">$</span>top_Genus_above_1)</span>
<span id="cb185-4"><a href="#cb185-4" aria-hidden="true" tabindex="-1"></a>  top_Genus_above_1_nog <span class="ot">=</span>  <span class="fu">gsub</span>(<span class="st">"g__"</span>, <span class="st">""</span>, top_Genus_above_1)</span>
<span id="cb185-5"><a href="#cb185-5" aria-hidden="true" tabindex="-1"></a>  top_Genus_above_1_nog</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Prevotella"                                                                                     
 [2] "Streptococcus"                                                                                  
 [3] "Neisseria"                                                                                      
 [4] "Haemophilus"                                                                                    
 [5] "Veillonella"                                                                                    
 [6] "Rothia"                                                                                         
 [7] "Fusobacterium"                                                                                  
 [8] "Treponema"                                                                                      
 [9] "Schaalia"                                                                                       
[10] "Escherichia"                                                                                    
[11] "Aggregatibacter"                                                                                
[12] "Selenomonas"                                                                                    
[13] "Leptotrichia"                                                                                   
[14] "Actinomyces"                                                                                    
[15] "Campylobacter"                                                                                  
[16] "Capnocytophaga"                                                                                 
[17] "Porphyromonas"                                                                                  
[18] "Bacteroides"                                                                                    
[19] "Gemella"                                                                                        
[20] "Filifactor"                                                                                     
[21] "Mogibacterium"                                                                                  
[22] "Atopobium"                                                                                      
[23] "Tannerella"                                                                                     
[24] "Lautropia"                                                                                      
[25] "Dialister"                                                                                      
[26] "NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Clostridiales_Family_XIII._Incertae_Sedis_NA"
[27] "Parvimonas"                                                                                     
[28] "Lachnoanaerobaculum"                                                                            
[29] "NA_p__Candidatus_Saccharibacteria_NA_NA_NA_NA"                                                  
[30] "Corynebacterium"                                                                                
[31] "NA_p__Firmicutes_c__Clostridia_o__Clostridiales_f__Lachnospiraceae_NA"                          
[32] "Clostridium"                                                                                    
[33] "Eikenella"                                                                                      
[34] "Ottowia"                                                                                        
[35] "Bacillus"                                                                                       
[36] "Cutibacterium"                                                                                  
[37] "Pseudomonas"                                                                                    
[38] "Lactobacillus"                                                                                  
[39] "Staphylococcus"                                                                                 
[40] "Shigella"                                                                                       
[41] "Mycoplasma"                                                                                     
[42] "Chryseobacterium"                                                                               
[43] "Olsenella"                                                                                      
[44] "Enterococcus"                                                                                   
[45] "Alistipes"                                                                                      
[46] "Streptomyces"                                                                                   
[47] "Pseudopropionibacterium"                                                                        
[48] "Lachnoclostridium"                                                                              
[49] "Acinetobacter"                                                                                  
[50] "Clostridioides"                                                                                 
[51] "Cardiobacterium"                                                                                
[52] "Blautia"                                                                                        
[53] "-other"                                                                                         </code></pre>
</div>
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select genera above 1% </span></span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a>importance_df<span class="sc">$</span>genera_above_1pc <span class="ot">=</span> <span class="fu">ifelse</span>(importance_df<span class="sc">$</span>genus <span class="sc">%in%</span> top_Genus_above_1_nog, importance_df<span class="sc">$</span>genus, <span class="st">"-other"</span>)</span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(importance_df<span class="sc">$</span>genera_above_1pc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Treponema"     "-other"        "Streptococcus" "Bacteroides"  
[5] "Porphyromonas" "Rothia"        "Filifactor"   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a>importance_df</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                   genus
1                                                                              Treponema
2                                                                          Mesorhizobium
3                                                                         Nitratifractor
4                                                                             Legionella
5                                                                         Bradyrhizobium
6                                                                            Aromatoleum
7                                                                              Wolinella
8                                                                           Methylomonas
9                                                                           Methylophaga
10                                                                          Helicobacter
11                                                                         Streptococcus
12                                                                         Gluconobacter
13             NA_p__Chloroflexi_c__Anaerolineae_o__Anaerolineales_f__Anaerolineaceae_NA
14                                                                           Bacteroides
15                                                                         Sphaerochaeta
16                                                                      Halothiobacillus
17                                                                          Sphingomonas
18                                                                        Flavonifractor
19                                                                Candidatus_Fonsibacter
20                                                                             Apibacter
21                                                                   Sediminispirochaeta
22                                                                          Bdellovibrio
23                                                                           Agarilytica
24                                                                             Aeromonas
25                                                                         Porphyromonas
26                                                                       Christensenella
27                                                                           Acetobacter
28                                                                      Polynucleobacter
29                                                                 Candidatus_Sneabacter
30                                                                           Polaromonas
31                                                                           Cronobacter
32                                                                                Rothia
33                                                                      Sulfurospirillum
34                                                                            Filifactor
35 NA_p__Proteobacteria_c__Alphaproteobacteria_o__Caulobacterales_f__Caulobacteraceae_NA
36                                                                          Caldisericum
37                                                                         Granulibacter
38                                                                         Nitrosococcus
39                                                                           Spiribacter
40                                                                              Niabella
41                                                                        Thiomicrospira
42                                                                            Microvirga
43                                                                               Aquifex
44                             NA_p__Bacteroidetes_c__Bacteroidia_o__Bacteroidales_NA_NA
45                                                                          Mucinivorans
46                                                                           Nibricoccus
47                                                                             Niastella
48                                                                             Aminipila
49                                                                       Hydrogenovibrio
50                                                               Candidatus_Amoebophilus
   diseased  healthy MeanDecreaseAccuracy MeanDecreaseGini genera_above_1pc
1  3.813811 3.819483             3.848799       0.60475676        Treponema
2  3.570529 3.645768             3.773284       0.48313916           -other
3  3.138665 3.133025             3.274970       0.36511057           -other
4  3.062566 3.243658             3.206423       0.43013065           -other
5  2.694328 2.686472             2.932664       0.30171679           -other
6  2.650086 2.366701             2.879448       0.24106379           -other
7  2.607550 2.521147             2.655158       0.22976600           -other
8  2.586981 2.494072             2.623596       0.24079986           -other
9  2.570876 2.561898             2.596500       0.22322705           -other
10 2.432749 2.348981             2.561684       0.19925256           -other
11 2.319519 2.492408             2.511051       0.22478624    Streptococcus
12 2.392176 2.491314             2.509198       0.21521953           -other
13 2.380656 2.200615             2.508079       0.20537851           -other
14 2.411051 2.425043             2.429416       0.25324492      Bacteroides
15 2.051026 2.384500             2.339719       0.21503016           -other
16 2.236657 2.321358             2.299651       0.22578579           -other
17 2.131900 2.288343             2.294035       0.20024590           -other
18 2.106615 2.069594             2.200827       0.20025867           -other
19 1.997069 2.114682             2.190388       0.12792185           -other
20 2.008355 2.246114             2.163743       0.17016244           -other
21 2.036366 2.124261             2.131244       0.14557493           -other
22 1.844693 1.805033             2.095141       0.10072209           -other
23 1.869259 1.417051             2.094648       0.12665271           -other
24 1.970407 1.944258             2.094234       0.13217465           -other
25 1.993040 2.017367             2.071909       0.16514619    Porphyromonas
26 2.024045 2.062304             2.069871       0.16235124           -other
27 1.698914 1.848734             1.976354       0.08043117           -other
28 1.945733 1.953848             1.968231       0.11276652           -other
29 1.948810 1.861360             1.956894       0.13603510           -other
30 1.981629 1.882805             1.952359       0.10155997           -other
31 1.001002 1.945821             1.944458       0.06886140           -other
32 1.848734 1.904810             1.944324       0.09979541           Rothia
33 1.309934 1.710589             1.921914       0.11598459           -other
34 1.699753 1.905190             1.892781       0.07596673       Filifactor
35 1.831032 1.827335             1.887358       0.13728722           -other
36 1.663615 1.959120             1.874178       0.09961450           -other
37 1.973227 1.754286             1.869145       0.13454294           -other
38 1.973227 1.790467             1.863095       0.16595553           -other
39 1.960967 1.667683             1.847650       0.11037310           -other
40 1.698914 1.842235             1.834682       0.11174129           -other
41 1.919564 1.597561             1.828645       0.13044920           -other
42 1.812456 1.781629             1.803640       0.14192021           -other
43 1.001002 1.417051             1.737270       0.01053333           -other
44 1.669110 1.702140             1.736480       0.14007668           -other
45 1.705119 1.675258             1.734990       0.09765827           -other
46 1.714697 1.723004             1.734291       0.06718977           -other
47 1.607707 1.728068             1.727477       0.09126100           -other
48 1.714697 1.730034             1.721261       0.08931950           -other
49 1.702664 1.661502             1.716955       0.09289381           -other
50 1.653148 1.730034             1.710589       0.08705834           -other</code></pre>
</div>
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a>importance_df_to_plot <span class="ot">=</span> importance_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genera_above_1pc <span class="sc">!=</span> <span class="st">"-other"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA"</span>, genus))</span>
<span id="cb191-2"><a href="#cb191-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-3"><a href="#cb191-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the top 10 important predictors above 1%</span></span>
<span id="cb191-4"><a href="#cb191-4" aria-hidden="true" tabindex="-1"></a>gp2 <span class="ot">=</span> <span class="fu">ggplot</span>(importance_df_to_plot, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(genera_above_1pc, <span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span>), <span class="at">y =</span> <span class="st">`</span><span class="at">MeanDecreaseAccuracy</span><span class="st">`</span>)) <span class="sc">+</span></span>
<span id="cb191-5"><a href="#cb191-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb191-6"><a href="#cb191-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip to make the plot horizontal</span></span>
<span id="cb191-7"><a href="#cb191-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb191-8"><a href="#cb191-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Top Most Important Genera Predictors"</span>,</span>
<span id="cb191-9"><a href="#cb191-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Genera above 1% overall abundance"</span>,</span>
<span id="cb191-10"><a href="#cb191-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Increase in Mean Squared Error (%IncMSE)"</span></span>
<span id="cb191-11"><a href="#cb191-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb191-12"><a href="#cb191-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>)</span>
<span id="cb191-13"><a href="#cb191-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb191-14"><a href="#cb191-14" aria-hidden="true" tabindex="-1"></a>gp2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-40-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_3A.png"</span>, <span class="at">plot =</span> gp2, <span class="at">width =</span> <span class="dv">14</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">dpi =</span> <span class="dv">400</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cross-validation" class="level2">
<h2 class="anchored" data-anchor-id="cross-validation">6.4 Cross validation</h2>
<section id="cross-validation-1" class="level3">
<h3 class="anchored" data-anchor-id="cross-validation-1">6.4.1 Cross validation</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb193"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb193-1"><a href="#cb193-1" aria-hidden="true" tabindex="-1"></a>  otu_table_g <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">otu_table</span>(phyloseq_obj))</span>
<span id="cb193-2"><a href="#cb193-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb193-3"><a href="#cb193-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(otu_table_g) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">colnames</span>(otu_table_g))</span>
<span id="cb193-4"><a href="#cb193-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(otu_table_g)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "A10"        "A11"        "A13"        "A14"        "A16"       
 [6] "A17"        "A18"        "A19"        "A1"         "A2"        
[11] "A3"         "A4"         "A5"         "A6"         "A7"        
[16] "A8"         "A9"         "DRR214959"  "DRR214960"  "DRR214961" 
[21] "DRR214962"  "DRR241310"  "SRR5892208" "SRR5892209" "SRR5892210"
[26] "SRR5892211" "SRR5892212" "SRR5892213" "SRR5892214" "SRR5892215"
[31] "SRR5892216" "SRR5892217" "SRS013942"  "SRS014468"  "SRS014692" 
[36] "SRS015055"  "SRS019120" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb195"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb195-1"><a href="#cb195-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove columns containing NA</span></span>
<span id="cb195-2"><a href="#cb195-2" aria-hidden="true" tabindex="-1"></a>  otu_table_g <span class="ot">=</span> otu_table_g[, <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colnames</span>(otu_table_g))]</span>
<span id="cb195-3"><a href="#cb195-3" aria-hidden="true" tabindex="-1"></a>  otu_table_g <span class="ot">=</span> otu_table_g[, <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA"</span>, <span class="fu">colnames</span>(otu_table_g))]</span>
<span id="cb195-4"><a href="#cb195-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb195-5"><a href="#cb195-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colSums</span>(<span class="fu">is.na</span>(otu_table_g)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       A10        A11        A13        A14        A16        A17        A18 
         0          0          0          0          0          0          0 
       A19         A1         A2         A3         A4         A5         A6 
         0          0          0          0          0          0          0 
        A7         A8         A9  DRR214959  DRR214960  DRR214961  DRR214962 
         0          0          0          0          0          0          0 
 DRR241310 SRR5892208 SRR5892209 SRR5892210 SRR5892211 SRR5892212 SRR5892213 
         0          0          0          0          0          0          0 
SRR5892214 SRR5892215 SRR5892216 SRR5892217  SRS013942  SRS014468  SRS014692 
         0          0          0          0          0          0          0 
 SRS015055  SRS019120 
         0          0 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a>  <span class="co">#otu_table_g = t(otu_table_g)</span></span>
<span id="cb197-2"><a href="#cb197-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb197-3"><a href="#cb197-3" aria-hidden="true" tabindex="-1"></a>  condition <span class="ot">=</span> <span class="fu">as.data.frame</span>(metadata<span class="sc">$</span>condition)</span>
<span id="cb197-4"><a href="#cb197-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(condition) <span class="ot">=</span> <span class="fu">rownames</span>(metadata)</span>
<span id="cb197-5"><a href="#cb197-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb197-6"><a href="#cb197-6" aria-hidden="true" tabindex="-1"></a>  otu_table_g <span class="ot">=</span> <span class="fu">t</span>(otu_table_g)</span>
<span id="cb197-7"><a href="#cb197-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb197-8"><a href="#cb197-8" aria-hidden="true" tabindex="-1"></a>  otu_table_g <span class="ot">=</span> <span class="fu">cbind</span>(condition, otu_table_g)</span>
<span id="cb197-9"><a href="#cb197-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb197-10"><a href="#cb197-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Split data into training and test sets (80/20 split)</span></span>
<span id="cb197-11"><a href="#cb197-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb197-12"><a href="#cb197-12" aria-hidden="true" tabindex="-1"></a>  train_index <span class="ot">=</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(otu_table_g), <span class="fl">0.8</span> <span class="sc">*</span> <span class="fu">nrow</span>(otu_table_g))</span>
<span id="cb197-13"><a href="#cb197-13" aria-hidden="true" tabindex="-1"></a>  train_data <span class="ot">=</span> otu_table_g[train_index, ]</span>
<span id="cb197-14"><a href="#cb197-14" aria-hidden="true" tabindex="-1"></a>  test_data <span class="ot">=</span> otu_table_g[<span class="sc">-</span>train_index, ]</span>
<span id="cb197-15"><a href="#cb197-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb197-16"><a href="#cb197-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span>(train_data)  <span class="co"># Check dimensions of your dataset</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]   29 1758</code></pre>
</div>
<div class="sourceCode cell-code" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span>(test_data) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]    8 1758</code></pre>
</div>
<div class="sourceCode cell-code" id="cb201"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb201-1"><a href="#cb201-1" aria-hidden="true" tabindex="-1"></a>  train_data<span class="sc">$</span><span class="st">`</span><span class="at">metadata$condition</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">as.factor</span>(train_data<span class="sc">$</span><span class="st">`</span><span class="at">metadata$condition</span><span class="st">`</span>)</span>
<span id="cb201-2"><a href="#cb201-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb201-3"><a href="#cb201-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Train Random Forest</span></span>
<span id="cb201-4"><a href="#cb201-4" aria-hidden="true" tabindex="-1"></a>  rf_model_2 <span class="ot">=</span> <span class="fu">randomForest</span>(<span class="st">`</span><span class="at">metadata$condition</span><span class="st">`</span> <span class="sc">~</span> ., <span class="at">data =</span> train_data, <span class="at">importance =</span> <span class="cn">TRUE</span>)</span>
<span id="cb201-5"><a href="#cb201-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb201-6"><a href="#cb201-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Predict on test set</span></span>
<span id="cb201-7"><a href="#cb201-7" aria-hidden="true" tabindex="-1"></a>  pred <span class="ot">=</span> <span class="fu">predict</span>(rf_model_2, test_data, <span class="at">type =</span> <span class="st">"prob"</span>)[, <span class="dv">2</span>]</span>
<span id="cb201-8"><a href="#cb201-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb201-9"><a href="#cb201-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Evaluate model performance</span></span>
<span id="cb201-10"><a href="#cb201-10" aria-hidden="true" tabindex="-1"></a>  roc_obj <span class="ot">=</span> <span class="fu">roc</span>(test_data<span class="sc">$</span><span class="st">`</span><span class="at">metadata$condition</span><span class="st">`</span>, pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting levels: control = diseased, case = healthy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting direction: controls &lt; cases</code></pre>
</div>
<div class="sourceCode cell-code" id="cb204"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb204-1"><a href="#cb204-1" aria-hidden="true" tabindex="-1"></a>  roc_plot <span class="ot">=</span> <span class="fu">plot</span>(roc_obj, <span class="at">main =</span> <span class="st">"ROC Curve for Random Forest"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>  roc_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
roc.default(response = test_data$`metadata$condition`, predictor = pred)

Data: pred in 4 controls (test_data$`metadata$condition` diseased) &lt; 4 cases (test_data$`metadata$condition` healthy).
Area under the curve: 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">auc</span>(roc_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Area under the curve: 1</code></pre>
</div>
</div>
</section>
<section id="train-random-forests-model-with-80-of-dataset" class="level3">
<h3 class="anchored" data-anchor-id="train-random-forests-model-with-80-of-dataset">6.4.1 Train random forests model with 80% of dataset</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load required libraries</span></span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-3"><a href="#cb209-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-4"><a href="#cb209-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Split data into training and testing sets</span></span>
<span id="cb209-5"><a href="#cb209-5" aria-hidden="true" tabindex="-1"></a><span class="co"># This is a crucial step you were missing</span></span>
<span id="cb209-6"><a href="#cb209-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>) <span class="co"># For reproducibility</span></span>
<span id="cb209-7"><a href="#cb209-7" aria-hidden="true" tabindex="-1"></a>train_index <span class="ot">=</span> <span class="fu">createDataPartition</span>(ml_data_clean<span class="sc">$</span>condition, <span class="at">p =</span> <span class="fl">0.8</span>, <span class="at">list =</span> <span class="cn">FALSE</span>)</span>
<span id="cb209-8"><a href="#cb209-8" aria-hidden="true" tabindex="-1"></a>train_data <span class="ot">=</span> ml_data_clean[train_index, ]</span>
<span id="cb209-9"><a href="#cb209-9" aria-hidden="true" tabindex="-1"></a>test_data <span class="ot">=</span> ml_data_clean[<span class="sc">-</span>train_index, ]</span>
<span id="cb209-10"><a href="#cb209-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-11"><a href="#cb209-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Train Random Forest model on training data</span></span>
<span id="cb209-12"><a href="#cb209-12" aria-hidden="true" tabindex="-1"></a>rf_model_cv <span class="ot">=</span> <span class="fu">randomForest</span>(condition <span class="sc">~</span> ., </span>
<span id="cb209-13"><a href="#cb209-13" aria-hidden="true" tabindex="-1"></a>                         <span class="at">data =</span> train_data, </span>
<span id="cb209-14"><a href="#cb209-14" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ntree =</span> <span class="dv">500</span>, </span>
<span id="cb209-15"><a href="#cb209-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">importance =</span> <span class="cn">TRUE</span>)</span>
<span id="cb209-16"><a href="#cb209-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-17"><a href="#cb209-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-18"><a href="#cb209-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the rf model object from paper</span></span>
<span id="cb209-19"><a href="#cb209-19" aria-hidden="true" tabindex="-1"></a>rf_model_cv <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"../data/rf_model_cv.rds"</span>)</span>
<span id="cb209-20"><a href="#cb209-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb209-21"><a href="#cb209-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Print model summary</span></span>
<span id="cb209-22"><a href="#cb209-22" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(rf_model_cv)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = `metadata$condition` ~ ., data = train_data,      importance = TRUE) 
               Type of random forest: classification
                     Number of trees: 500
No. of variables tried at each split: 41

        OOB estimate of  error rate: 6.9%
Confusion matrix:
         diseased healthy class.error
diseased       12       1  0.07692308
healthy         1      15  0.06250000</code></pre>
</div>
</div>
</section>
<section id="variable-importance" class="level3">
<h3 class="anchored" data-anchor-id="variable-importance">6.4.1 Variable importance</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variable importance</span></span>
<span id="cb211-2"><a href="#cb211-2" aria-hidden="true" tabindex="-1"></a>importance_rf_model_cv <span class="ot">=</span> <span class="fu">importance</span>(rf_model_cv)</span>
<span id="cb211-3"><a href="#cb211-3" aria-hidden="true" tabindex="-1"></a><span class="fu">varImpPlot</span>(rf_model_cv, <span class="at">main =</span> <span class="st">"Variable Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="predicting-probabilities-of-test-data-based-on-trained-model" class="level3">
<h3 class="anchored" data-anchor-id="predicting-probabilities-of-test-data-based-on-trained-model">6.4.1 Predicting probabilities of test data based on trained model</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Evaluate on testing data - never evaluate on the same data you trained on</span></span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>predictions_prob <span class="ot">=</span> <span class="fu">predict</span>(rf_model_cv, test_data, <span class="at">type =</span> <span class="st">"prob"</span>)</span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check structure of predictions</span></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(predictions_prob)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> 'matrix' num [1:7, 1:2] 0.83 0.842 0.91 0.4 0.308 0.302 0.362 0.17 0.158 0.09 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:7] "A10" "A13" "A5" "DRR214959" ...
  ..$ : chr [1:2] "diseased" "healthy"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract probabilities for the positive class (assuming binary classification)</span></span>
<span id="cb214-2"><a href="#cb214-2" aria-hidden="true" tabindex="-1"></a><span class="co"># If condition has levels "No" and "Yes" or "0" and "1"</span></span>
<span id="cb214-3"><a href="#cb214-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the column name of the positive class (typically the second column)</span></span>
<span id="cb214-4"><a href="#cb214-4" aria-hidden="true" tabindex="-1"></a>positive_class <span class="ot">=</span> <span class="fu">colnames</span>(predictions_prob)[<span class="dv">2</span>]</span>
<span id="cb214-5"><a href="#cb214-5" aria-hidden="true" tabindex="-1"></a>prob_positive <span class="ot">=</span> predictions_prob[, positive_class]</span>
<span id="cb214-6"><a href="#cb214-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-7"><a href="#cb214-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create class predictions based on probability threshold</span></span>
<span id="cb214-8"><a href="#cb214-8" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">=</span> <span class="fl">0.5</span></span>
<span id="cb214-9"><a href="#cb214-9" aria-hidden="true" tabindex="-1"></a>predictions_class <span class="ot">=</span> <span class="fu">factor</span>(<span class="fu">ifelse</span>(prob_positive <span class="sc">&gt;</span> threshold, </span>
<span id="cb214-10"><a href="#cb214-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">levels</span>(test_data<span class="sc">$</span>condition)[<span class="dv">2</span>], </span>
<span id="cb214-11"><a href="#cb214-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">levels</span>(test_data<span class="sc">$</span>condition)[<span class="dv">1</span>]),</span>
<span id="cb214-12"><a href="#cb214-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">levels =</span> <span class="fu">levels</span>(test_data<span class="sc">$</span>condition))</span>
<span id="cb214-13"><a href="#cb214-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb214-14"><a href="#cb214-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Create confusion matrix</span></span>
<span id="cb214-15"><a href="#cb214-15" aria-hidden="true" tabindex="-1"></a>conf_matrix <span class="ot">=</span> <span class="fu">confusionMatrix</span>(predictions_class, test_data<span class="sc">$</span>condition)</span>
<span id="cb214-16"><a href="#cb214-16" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(conf_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction diseased healthy
  diseased        3       0
  healthy         0       4
                                     
               Accuracy : 1          
                 95% CI : (0.5904, 1)
    No Information Rate : 0.5714     
    P-Value [Acc &gt; NIR] : 0.01989    
                                     
                  Kappa : 1          
                                     
 Mcnemar's Test P-Value : NA         
                                     
            Sensitivity : 1.0000     
            Specificity : 1.0000     
         Pos Pred Value : 1.0000     
         Neg Pred Value : 1.0000     
             Prevalence : 0.4286     
         Detection Rate : 0.4286     
   Detection Prevalence : 0.4286     
      Balanced Accuracy : 1.0000     
                                     
       'Positive' Class : diseased   
                                     </code></pre>
</div>
</div>
</section>
<section id="roc-curve-and-auc-with-rocr" class="level3">
<h3 class="anchored" data-anchor-id="roc-curve-and-auc-with-rocr">6.4.1 ROC curve and AUC with ROCR</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ROC curve using ROCR</span></span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a><span class="co"># For ROCR, we need numeric labels (convert factor to numeric)</span></span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a>numeric_labels <span class="ot">=</span> <span class="fu">as.numeric</span>(test_data<span class="sc">$</span>condition) <span class="sc">-</span> <span class="dv">1</span>  <span class="co"># Convert to 0/1</span></span>
<span id="cb216-4"><a href="#cb216-4" aria-hidden="true" tabindex="-1"></a>roc_pred <span class="ot">=</span> <span class="fu">prediction</span>(<span class="at">predictions =</span> prob_positive, <span class="at">labels =</span> numeric_labels)</span>
<span id="cb216-5"><a href="#cb216-5" aria-hidden="true" tabindex="-1"></a>roc_perf <span class="ot">=</span> <span class="fu">performance</span>(roc_pred, <span class="st">"tpr"</span>, <span class="st">"fpr"</span>)</span>
<span id="cb216-6"><a href="#cb216-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb216-7"><a href="#cb216-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot ROC curve</span></span>
<span id="cb216-8"><a href="#cb216-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(roc_perf,</span>
<span id="cb216-9"><a href="#cb216-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">colorize =</span> <span class="cn">TRUE</span>,</span>
<span id="cb216-10"><a href="#cb216-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">print.cutoffs.at =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.1</span>),</span>
<span id="cb216-11"><a href="#cb216-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">text.adj =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.2</span>, <span class="fl">1.7</span>),</span>
<span id="cb216-12"><a href="#cb216-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"ROC Curve"</span>)</span>
<span id="cb216-13"><a href="#cb216-13" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a =</span> <span class="dv">0</span>, <span class="at">b =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">"gray"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb217"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate AUC</span></span>
<span id="cb217-2"><a href="#cb217-2" aria-hidden="true" tabindex="-1"></a>auc_ROCR <span class="ot">=</span> <span class="fu">performance</span>(roc_pred, <span class="at">measure =</span> <span class="st">"auc"</span>)</span>
<span id="cb217-3"><a href="#cb217-3" aria-hidden="true" tabindex="-1"></a>auc_value <span class="ot">=</span> auc_ROCR<span class="sc">@</span>y.values[[<span class="dv">1</span>]]</span>
<span id="cb217-4"><a href="#cb217-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"AUC:"</span>, <span class="fu">round</span>(auc_value, <span class="dv">4</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AUC: 1"</code></pre>
</div>
</div>
</section>
<section id="roc-curve-and-auc-with-proc" class="level3">
<h3 class="anchored" data-anchor-id="roc-curve-and-auc-with-proc">6.4.1 ROC curve and AUC with pROC</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Alternative ROC curve using pROC package</span></span>
<span id="cb219-2"><a href="#cb219-2" aria-hidden="true" tabindex="-1"></a>roc_curve <span class="ot">=</span> <span class="fu">roc</span>(<span class="at">response =</span> test_data<span class="sc">$</span>condition, </span>
<span id="cb219-3"><a href="#cb219-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">predictor =</span> prob_positive)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting levels: control = diseased, case = healthy</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Setting direction: controls &lt; cases</code></pre>
</div>
<div class="sourceCode cell-code" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(roc_curve, <span class="at">main =</span> <span class="st">"ROC Curve using pROC"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb223"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb223-1"><a href="#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"AUC (pROC):"</span>, <span class="fu">round</span>(<span class="fu">auc</span>(roc_curve), <span class="dv">4</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AUC (pROC): 1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb225"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb225-1"><a href="#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as pnf file</span></span>
<span id="cb225-2"><a href="#cb225-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Open png file</span></span>
<span id="cb225-3"><a href="#cb225-3" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"../imgs/Figure_3B.png"</span>, <span class="at">width =</span> <span class="dv">256</span>, <span class="at">height =</span> <span class="dv">256</span>, <span class="at">units =</span> <span class="st">"px"</span>)</span>
<span id="cb225-4"><a href="#cb225-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Create the plot</span></span>
<span id="cb225-5"><a href="#cb225-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(roc_curve, <span class="at">main =</span> <span class="st">"ROC Curve using pROC"</span>)</span>
<span id="cb225-6"><a href="#cb225-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"AUC (pROC):"</span>, <span class="fu">round</span>(<span class="fu">auc</span>(roc_curve), <span class="dv">4</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "AUC (pROC): 1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Close the file</span></span>
<span id="cb227-2"><a href="#cb227-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb229"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb229-1"><a href="#cb229-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Print performance metrics</span></span>
<span id="cb229-2"><a href="#cb229-2" aria-hidden="true" tabindex="-1"></a>accuracy <span class="ot">=</span> conf_matrix<span class="sc">$</span>overall[<span class="st">"Accuracy"</span>]</span>
<span id="cb229-3"><a href="#cb229-3" aria-hidden="true" tabindex="-1"></a>sensitivity <span class="ot">=</span> conf_matrix<span class="sc">$</span>byClass[<span class="st">"Sensitivity"</span>]</span>
<span id="cb229-4"><a href="#cb229-4" aria-hidden="true" tabindex="-1"></a>specificity <span class="ot">=</span> conf_matrix<span class="sc">$</span>byClass[<span class="st">"Specificity"</span>]</span>
<span id="cb229-5"><a href="#cb229-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb229-6"><a href="#cb229-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"Accuracy:"</span>, <span class="fu">round</span>(accuracy, <span class="dv">4</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Accuracy: 1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb231"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb231-1"><a href="#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"Sensitivity:"</span>, <span class="fu">round</span>(sensitivity, <span class="dv">4</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Sensitivity: 1"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"Specificity:"</span>, <span class="fu">round</span>(specificity, <span class="dv">4</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Specificity: 1"</code></pre>
</div>
</div>
</section>
</section>
<section id="permutation-testing-for-rf-accuracy" class="level2">
<h2 class="anchored" data-anchor-id="permutation-testing-for-rf-accuracy">6.5 Permutation testing for RF accuracy</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb235"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb235-1"><a href="#cb235-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure column names are correctly referenced</span></span>
<span id="cb235-2"><a href="#cb235-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb235-3"><a href="#cb235-3" aria-hidden="true" tabindex="-1"></a>  train_data<span class="sc">$</span><span class="st">`</span><span class="at">metadata$condition</span><span class="st">`</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<div class="sourceCode cell-code" id="cb237"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb237-1"><a href="#cb237-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(train_data)[<span class="fu">colnames</span>(train_data) <span class="sc">==</span> <span class="st">"metadata$condition"</span>] <span class="ot">=</span> <span class="st">"condition"</span></span>
<span id="cb237-2"><a href="#cb237-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(test_data)[<span class="fu">colnames</span>(test_data) <span class="sc">==</span> <span class="st">"metadata$condition"</span>] <span class="ot">=</span> <span class="st">"condition"</span></span>
<span id="cb237-3"><a href="#cb237-3" aria-hidden="true" tabindex="-1"></a>  train_data<span class="sc">$</span>condition</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] diseased diseased diseased diseased diseased diseased diseased diseased
 [9] diseased diseased diseased diseased diseased diseased healthy  healthy 
[17] healthy  healthy  healthy  healthy  healthy  healthy  healthy  healthy 
[25] healthy  healthy  healthy  healthy  healthy  healthy 
Levels: diseased healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a>  test_data<span class="sc">$</span>condition</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] diseased diseased diseased healthy  healthy  healthy  healthy 
Levels: diseased healthy</code></pre>
</div>
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Permutation testing</span></span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-3"><a href="#cb241-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># We will not run this as it takes alot of compute power</span></span>
<span id="cb241-4"><a href="#cb241-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Code is shown below for reference </span></span>
<span id="cb241-5"><a href="#cb241-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-6"><a href="#cb241-6" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb241-7"><a href="#cb241-7" aria-hidden="true" tabindex="-1"></a> <span class="co"># null_accuracies = replicate(1000, {</span></span>
<span id="cb241-8"><a href="#cb241-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Shuffle the labels</span></span>
<span id="cb241-9"><a href="#cb241-9" aria-hidden="true" tabindex="-1"></a> <span class="co">#   perm_labels = sample(train_data$condition)  # Use the actual column name for condition</span></span>
<span id="cb241-10"><a href="#cb241-10" aria-hidden="true" tabindex="-1"></a> <span class="co">#   train_data_perm = train_data</span></span>
<span id="cb241-11"><a href="#cb241-11" aria-hidden="true" tabindex="-1"></a> <span class="co">#   train_data_perm$condition = perm_labels  # Replace condition labels with permuted ones</span></span>
<span id="cb241-12"><a href="#cb241-12" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb241-13"><a href="#cb241-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Train random forest on permuted data</span></span>
<span id="cb241-14"><a href="#cb241-14" aria-hidden="true" tabindex="-1"></a> <span class="co">#   rf_perm = randomForest(condition ~ ., data = train_data_perm, importance = TRUE)</span></span>
<span id="cb241-15"><a href="#cb241-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb241-16"><a href="#cb241-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Test on test_data</span></span>
<span id="cb241-17"><a href="#cb241-17" aria-hidden="true" tabindex="-1"></a> <span class="co">#   mean(predict(rf_perm, test_data) == test_data$condition)  # Ensure test_data$condition matches</span></span>
<span id="cb241-18"><a href="#cb241-18" aria-hidden="true" tabindex="-1"></a> <span class="co"># }</span></span>
<span id="cb241-19"><a href="#cb241-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-20"><a href="#cb241-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the rf model object</span></span>
<span id="cb241-21"><a href="#cb241-21" aria-hidden="true" tabindex="-1"></a>null_accuracies <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"../data/null_accuracies.rds"</span>)</span>
<span id="cb241-22"><a href="#cb241-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-23"><a href="#cb241-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare observed accuracy to null distribution</span></span>
<span id="cb241-24"><a href="#cb241-24" aria-hidden="true" tabindex="-1"></a>observed_accuracy <span class="ot">=</span> <span class="fu">mean</span>(<span class="fu">predict</span>(rf_model_cv, test_data) <span class="sc">==</span> test_data<span class="sc">$</span>condition)</span>
<span id="cb241-25"><a href="#cb241-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-26"><a href="#cb241-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the p-value</span></span>
<span id="cb241-27"><a href="#cb241-27" aria-hidden="true" tabindex="-1"></a>p_value <span class="ot">=</span> <span class="fu">sum</span>(null_accuracies <span class="sc">&gt;=</span> observed_accuracy) <span class="sc">/</span> <span class="fu">length</span>(null_accuracies)</span>
<span id="cb241-28"><a href="#cb241-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb241-29"><a href="#cb241-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the p-value</span></span>
<span id="cb241-30"><a href="#cb241-30" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(p_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.022</code></pre>
</div>
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualize and interpret</span></span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb243-3"><a href="#cb243-3" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Create the plot</span></span>
<span id="cb243-4"><a href="#cb243-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(null_accuracies, <span class="at">main =</span><span class="fu">paste</span>(<span class="st">"Null Distribution of Accuracy</span><span class="sc">\n</span><span class="st">p-value ="</span>, <span class="fu">round</span>(p_value, <span class="dv">4</span>)), <span class="at">xlab =</span> <span class="st">"Accuracy"</span>)</span>
<span id="cb243-5"><a href="#cb243-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> observed_accuracy, <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lwd =</span> <span class="dv">2</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb244"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb244-1"><a href="#cb244-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as pnf file</span></span>
<span id="cb244-2"><a href="#cb244-2" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Open png file</span></span>
<span id="cb244-3"><a href="#cb244-3" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"../imgs/Figure_3C.png"</span>, <span class="at">width =</span> <span class="dv">256</span>, <span class="at">height =</span> <span class="dv">256</span>, <span class="at">units =</span> <span class="st">"px"</span>)</span>
<span id="cb244-4"><a href="#cb244-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Create the plot</span></span>
<span id="cb244-5"><a href="#cb244-5" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(null_accuracies, <span class="at">main =</span><span class="fu">paste</span>(<span class="st">"Null Distribution of Accuracy</span><span class="sc">\n</span><span class="st">p-value ="</span>, <span class="fu">round</span>(p_value, <span class="dv">4</span>)), <span class="at">xlab =</span> <span class="st">"Accuracy"</span>)</span>
<span id="cb244-6"><a href="#cb244-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> observed_accuracy, <span class="at">col =</span> <span class="st">"red"</span>, <span class="at">lwd =</span> <span class="dv">2</span>) </span>
<span id="cb244-7"><a href="#cb244-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Close the file</span></span>
<span id="cb244-8"><a href="#cb244-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
</div>
</section>
</section>
<section id="ordination-of-multivariate-data" class="level1">
<h1>7. Ordination of multivariate data</h1>
<p>Here we run ordination of multivariate data into two dimensions using the unsupervised learning techniques; Principal Coordinate Analysis (PCoA), Non-metric Multidimensional Scaling (NMDS), and Redundancy Analysis (RDA) all with Bray-Curtis dissimilarity.</p>
<section id="calculating-bray-curtis-dissimilarity" class="level2">
<h2 class="anchored" data-anchor-id="calculating-bray-curtis-dissimilarity">7.1 Calculating Bray-Curtis dissimilarity</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb246"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb246-1"><a href="#cb246-1" aria-hidden="true" tabindex="-1"></a>  phyloseq_obj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 1757 taxa and 37 samples ]
sample_data() Sample Data:       [ 37 samples by 1 sample variables ]
tax_table()   Taxonomy Table:    [ 1757 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb248"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb248-1"><a href="#cb248-1" aria-hidden="true" tabindex="-1"></a>  phyloseq_obj<span class="sc">@</span>sam_data[[<span class="st">"condition"</span>]] <span class="ot">=</span> <span class="fu">as.factor</span>(phyloseq_obj<span class="sc">@</span>sam_data[[<span class="st">"condition"</span>]])</span>
<span id="cb248-2"><a href="#cb248-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb248-3"><a href="#cb248-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate Bray-Curtis dissimilarity</span></span>
<span id="cb248-4"><a href="#cb248-4" aria-hidden="true" tabindex="-1"></a>  dist_bc <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_obj, <span class="at">method =</span> <span class="st">"bray"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="permanova" class="level2">
<h2 class="anchored" data-anchor-id="permanova">7.1 PERMANOVA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a>  dist_bc <span class="ot">=</span> phyloseq<span class="sc">::</span><span class="fu">distance</span>(phyloseq_obj, <span class="at">method =</span> <span class="st">"bray"</span>)</span>
<span id="cb249-2"><a href="#cb249-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">adonis2</span>(dist_bc <span class="sc">~</span> condition, <span class="at">data =</span> <span class="fu">as</span>(<span class="fu">sample_data</span>(phyloseq_obj), <span class="st">"data.frame"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Permutation test for adonis under reduced model
Permutation: free
Number of permutations: 999

adonis2(formula = dist_bc ~ condition, data = as(sample_data(phyloseq_obj), "data.frame"))
         Df SumOfSqs      R2      F Pr(&gt;F)    
Model     1   2.5163 0.40556 23.879  0.001 ***
Residual 35   3.6881 0.59444                  
Total    36   6.2044 1.00000                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
<section id="pcoa" class="level2">
<h2 class="anchored" data-anchor-id="pcoa">7.3 PCoA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Visualize PCoA</span></span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>  ordination_results <span class="ot">=</span> <span class="fu">ordinate</span>(phyloseq_obj, <span class="at">method =</span> <span class="st">"PCoA"</span>, <span class="at">distance =</span> <span class="st">"bray"</span>)</span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb251-4"><a href="#cb251-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert ordination to a data frame</span></span>
<span id="cb251-5"><a href="#cb251-5" aria-hidden="true" tabindex="-1"></a>  ordination_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(ordination_results<span class="sc">$</span>vectors)</span>
<span id="cb251-6"><a href="#cb251-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ordination_df) <span class="ot">=</span> <span class="fu">rownames</span>(ordination_results<span class="sc">$</span>vectors)</span>
<span id="cb251-7"><a href="#cb251-7" aria-hidden="true" tabindex="-1"></a>  ordination_df<span class="sc">$</span>condition <span class="ot">=</span> <span class="fu">sample_data</span>(phyloseq_obj)<span class="sc">$</span>condition</span>
<span id="cb251-8"><a href="#cb251-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(ordination_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        Axis.1     Axis.2      Axis.3      Axis.4       Axis.5        Axis.6
A10 -0.3990528 -0.2763281 -0.02763561  0.01750740 -0.138888422  0.0500316957
A11 -0.2163923  0.1093911  0.08424231 -0.01368495  0.077659141  0.1039769612
A13 -0.3788201 -0.0812252  0.02994839  0.04811056 -0.024370993  0.0032111610
A14 -0.2237416  0.1484954  0.07359393 -0.02273669  0.066676250  0.0005697727
A16 -0.1909393  0.3327742 -0.02746487  0.04747409  0.005856121  0.1327296103
A17 -0.2253981  0.1666075 -0.11150852  0.18120400  0.321480041 -0.1972564933
          Axis.7      Axis.8       Axis.9      Axis.10     Axis.11      Axis.12
A10 -0.014754793 -0.06335845  0.105871797  0.019766348  0.03963333 -0.045911689
A11  0.074412305  0.09558781 -0.043101902 -0.065767576  0.02368048  0.009730679
A13  0.062739465 -0.07794522  0.006988258  0.052129777  0.01741799 -0.026402558
A14  0.012886279  0.10306105  0.071850753  0.038935393 -0.03672565 -0.094311398
A16 -0.044221595  0.02505884 -0.053434452 -0.039630828  0.10089014  0.041121914
A17 -0.005963391 -0.07225894  0.074203753 -0.002761451  0.01035562  0.015650505
         Axis.13       Axis.14       Axis.15      Axis.16       Axis.17
A10 -0.007735055  0.0243225804 -0.0500978722  0.002754255  0.0141803813
A11  0.028719219  0.0468579695 -0.0008639627  0.013852955  0.0001041077
A13  0.090486148 -0.0029947033 -0.0005909820 -0.037640174  0.0089074175
A14 -0.021976023 -0.0189616196 -0.0040817690 -0.017059139  0.0036328221
A16 -0.036876441  0.0002420351  0.0005302102 -0.001988632  0.0121336219
A17 -0.003764348  0.0045354662 -0.0201016169  0.016929575 -0.0135230435
          Axis.18      Axis.19      Axis.20      Axis.21      Axis.22 condition
A10 -0.0204580789 -0.003451114  0.020110360  0.015963667 -0.008437933  diseased
A11 -0.0039070040 -0.022333723 -0.001090578 -0.001483207 -0.002401530  diseased
A13 -0.0053896699 -0.013526270 -0.003312557 -0.001061440 -0.002060376  diseased
A14  0.0251298650  0.012571149 -0.001016701  0.001364480  0.007975996  diseased
A16 -0.0009390817 -0.001842882  0.004423927  0.005939336 -0.004796995  diseased
A17  0.0006735671 -0.001796595  0.001398673 -0.001108894 -0.001930982  diseased</code></pre>
</div>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot using ggplot2</span></span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a>  PCoA_plot <span class="ot">=</span> <span class="fu">ggplot</span>(ordination_df, <span class="fu">aes</span>(<span class="at">x =</span> Axis<span class="fl">.1</span>, <span class="at">y =</span> Axis<span class="fl">.2</span>, <span class="at">color =</span> condition)) <span class="sc">+</span></span>
<span id="cb253-3"><a href="#cb253-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb253-4"><a href="#cb253-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb253-5"><a href="#cb253-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PCoA1"</span>, <span class="at">y =</span> <span class="st">"PCoA2"</span>)  <span class="sc">+</span></span>
<span id="cb253-6"><a href="#cb253-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"PCoA"</span>) <span class="sc">+</span></span>
<span id="cb253-7"><a href="#cb253-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>)</span>
<span id="cb253-8"><a href="#cb253-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb253-9"><a href="#cb253-9" aria-hidden="true" tabindex="-1"></a>  PCoA_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nmds" class="level2">
<h2 class="anchored" data-anchor-id="nmds">7.4 NMDS</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb254"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb254-1"><a href="#cb254-1" aria-hidden="true" tabindex="-1"></a>  ordination_results <span class="ot">=</span> <span class="fu">ordinate</span>(phyloseq_obj, <span class="at">method =</span> <span class="st">"NMDS"</span>, <span class="at">distance =</span> <span class="st">"bray"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Square root transformation
Wisconsin double standardization
Run 0 stress 0.102349 
Run 1 stress 0.102349 
... New best solution
... Procrustes: rmse 6.002987e-05  max resid 0.0003042707 
... Similar to previous best
Run 2 stress 0.131654 
Run 3 stress 0.1506378 
Run 4 stress 0.102349 
... Procrustes: rmse 6.517857e-05  max resid 0.0003164684 
... Similar to previous best
Run 5 stress 0.131758 
Run 6 stress 0.1501758 
Run 7 stress 0.1541716 
Run 8 stress 0.1532792 
Run 9 stress 0.102349 
... Procrustes: rmse 2.840524e-05  max resid 0.0001342199 
... Similar to previous best
Run 10 stress 0.1219459 
Run 11 stress 0.102349 
... Procrustes: rmse 1.307163e-06  max resid 3.359396e-06 
... Similar to previous best
Run 12 stress 0.1584376 
Run 13 stress 0.1481237 
Run 14 stress 0.102349 
... Procrustes: rmse 4.982236e-06  max resid 1.939235e-05 
... Similar to previous best
Run 15 stress 0.1197668 
Run 16 stress 0.1218091 
Run 17 stress 0.102349 
... New best solution
... Procrustes: rmse 1.056919e-05  max resid 4.351121e-05 
... Similar to previous best
Run 18 stress 0.1306804 
Run 19 stress 0.1455878 
Run 20 stress 0.102349 
... Procrustes: rmse 5.076611e-05  max resid 0.0002140536 
... Similar to previous best
*** Best solution repeated 2 times</code></pre>
</div>
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert ordination to a data frame</span></span>
<span id="cb256-2"><a href="#cb256-2" aria-hidden="true" tabindex="-1"></a>  ordination_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(ordination_results<span class="sc">$</span>points)</span>
<span id="cb256-3"><a href="#cb256-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ordination_df) <span class="ot">=</span> <span class="fu">rownames</span>(ordination_results<span class="sc">$</span>points)</span>
<span id="cb256-4"><a href="#cb256-4" aria-hidden="true" tabindex="-1"></a>  ordination_df<span class="sc">$</span>condition <span class="ot">=</span> <span class="fu">sample_data</span>(phyloseq_obj)<span class="sc">$</span>condition</span>
<span id="cb256-5"><a href="#cb256-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(ordination_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           MDS1         MDS2 condition
A10 -0.06262088  0.105930719  diseased
A11 -0.07493688  0.005846474  diseased
A13 -0.05959023  0.074260860  diseased
A14 -0.12131818  0.035252176  diseased
A16 -0.11959864 -0.006427995  diseased
A17 -0.25742374  0.112209839  diseased</code></pre>
</div>
<div class="sourceCode cell-code" id="cb258"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb258-1"><a href="#cb258-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot using ggplot2</span></span>
<span id="cb258-2"><a href="#cb258-2" aria-hidden="true" tabindex="-1"></a>  NMDS_plot <span class="ot">=</span> <span class="fu">ggplot</span>(ordination_df, <span class="fu">aes</span>(<span class="at">x =</span> MDS1, <span class="at">y =</span> MDS2, <span class="at">color =</span> condition)) <span class="sc">+</span></span>
<span id="cb258-3"><a href="#cb258-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb258-4"><a href="#cb258-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"MDS1"</span>, <span class="at">y =</span> <span class="st">"MDS2"</span>)  <span class="sc">+</span></span>
<span id="cb258-5"><a href="#cb258-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"NMDS"</span>) <span class="sc">+</span></span>
<span id="cb258-6"><a href="#cb258-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>)</span>
<span id="cb258-7"><a href="#cb258-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb258-8"><a href="#cb258-8" aria-hidden="true" tabindex="-1"></a>  NMDS_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-51-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="rda" class="level2">
<h2 class="anchored" data-anchor-id="rda">7.5 RDA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Visualize PCoA</span></span>
<span id="cb259-2"><a href="#cb259-2" aria-hidden="true" tabindex="-1"></a>  ordination_results <span class="ot">=</span> <span class="fu">ordinate</span>(phyloseq_obj, <span class="at">method =</span> <span class="st">"RDA"</span>, <span class="at">distance =</span> <span class="st">"bray"</span>)</span>
<span id="cb259-3"><a href="#cb259-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb259-4"><a href="#cb259-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert ordination to a data frame</span></span>
<span id="cb259-5"><a href="#cb259-5" aria-hidden="true" tabindex="-1"></a>  ordination_df <span class="ot">=</span> <span class="fu">as.data.frame</span>(ordination_results[[<span class="st">"CA"</span>]][[<span class="st">"u"</span>]])</span>
<span id="cb259-6"><a href="#cb259-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ordination_df) <span class="ot">=</span> <span class="fu">rownames</span>(ordination_results[[<span class="st">"CA"</span>]][[<span class="st">"u"</span>]])</span>
<span id="cb259-7"><a href="#cb259-7" aria-hidden="true" tabindex="-1"></a>  ordination_df<span class="sc">$</span>condition <span class="ot">=</span> <span class="fu">sample_data</span>(phyloseq_obj)<span class="sc">$</span>condition</span>
<span id="cb259-8"><a href="#cb259-8" aria-hidden="true" tabindex="-1"></a>  ordination_df<span class="sc">$</span>sample_names <span class="ot">=</span> <span class="fu">rownames</span>(ordination_df)</span>
<span id="cb259-9"><a href="#cb259-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(ordination_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            PC1         PC2          PC3         PC4         PC5         PC6
A10 -0.33204284  0.20912012 -0.144253041 -0.03059961  0.23588105 -0.03713143
A11 -0.08468054 -0.05827521  0.091839883 -0.03904873 -0.03025444  0.11074921
A13 -0.21986915  0.05864669 -0.005890016 -0.06419824  0.11837347  0.01683602
A14 -0.10014296 -0.07028627  0.129444325 -0.03692765 -0.06995082 -0.05230855
A16 -0.03148699 -0.19503334  0.263923709 -0.14196187  0.07884507  0.16475294
A17 -0.04975940 -0.15219201  0.110139728 -0.25735315 -0.72081181 -0.17642925
            PC7         PC8          PC9         PC10        PC11        PC12
A10 -0.02944898  0.25566690 -0.306760531  0.124883398 -0.12498715  0.16274571
A11  0.01094340 -0.15876290  0.115473979 -0.367725699 -0.16103047  0.05711542
A13  0.02455014  0.04861554 -0.109757220  0.007177984 -0.18117784 -0.42471427
A14 -0.10766862  0.28908187 -0.164363766  0.068477770  0.10221473  0.21414285
A16 -0.03183595 -0.11431324  0.176763963 -0.080347004 -0.16250451  0.27998110
A17 -0.11097290  0.11671967 -0.002509756  0.347931019 -0.08731619 -0.19967489
            PC13        PC14        PC15         PC16        PC17       PC18
A10 -0.095156533  0.09453161 -0.12202847  0.020785156  0.03601123 -0.1181814
A11 -0.007034822  0.15933808 -0.07668232  0.112010102 -0.12134268 -0.1053740
A13 -0.576576384  0.18839905  0.02470430  0.086624203  0.05976665 -0.1154655
A14  0.165565336 -0.03733827  0.18188986 -0.148754274  0.18201159 -0.1773570
A16 -0.059880870  0.26900679 -0.28375716 -0.031229954 -0.11600837 -0.2700773
A17 -0.045742598  0.03633465 -0.01428297  0.009103755 -0.13884746  0.1092161
           PC19        PC20        PC21         PC22        PC23        PC24
A10 -0.22109978  0.13269048  0.16100239  0.097298537  0.06213204  0.03879865
A11 -0.01730906  0.46492550 -0.16470574  0.271570983  0.33759611  0.12054865
A13  0.10220793 -0.14417527 -0.06732180 -0.122297369  0.16478915 -0.27003072
A14 -0.12276290 -0.13931812  0.08742432 -0.030674244 -0.01195781 -0.02473966
A16 -0.11544395 -0.03174451  0.02565170 -0.006926064 -0.51149227 -0.18306242
A17 -0.06962461  0.14244117 -0.10618803  0.080501463 -0.07313958 -0.07749316
           PC25        PC26          PC27        PC28        PC29        PC30
A10 -0.42812319 -0.21903386 -0.0007227003 -0.02340901 -0.02632542 -0.16877654
A11 -0.02948713 -0.02394383  0.0017440496  0.21644163 -0.18301587  0.31180164
A13  0.12137868  0.17231373  0.1538940397 -0.05300262  0.06280725  0.08988459
A14  0.10997873 -0.17703706  0.3085810106  0.31629916  0.11977402  0.43975276
A16  0.08708884 -0.08039283 -0.0044842244 -0.21725513  0.01552273 -0.10433124
A17 -0.12309220 -0.05488621 -0.0601495289  0.03133817 -0.02899176 -0.06770742
           PC31        PC32        PC33         PC34         PC35         PC36
A10 -0.03695492 -0.22395802 -0.04865635 -0.181440709 -0.037170808 -0.131063826
A11 -0.01621193 -0.05244742  0.05173978  0.173070007  0.064738719 -0.042649428
A13 -0.07342503  0.16770515  0.01757990  0.050672094  0.018517629  0.105086390
A14  0.02943466  0.31496951 -0.06218461  0.074158512  0.013174469 -0.024876276
A16  0.02869031  0.12802347 -0.09138193  0.042254488 -0.004793286 -0.020223887
A17 -0.01490583 -0.06973480 -0.03553475 -0.007599638  0.002019312  0.005315909
    condition sample_names
A10  diseased          A10
A11  diseased          A11
A13  diseased          A13
A14  diseased          A14
A16  diseased          A16
A17  diseased          A17</code></pre>
</div>
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot using ggplot2</span></span>
<span id="cb261-2"><a href="#cb261-2" aria-hidden="true" tabindex="-1"></a>  RDA_plot <span class="ot">=</span> <span class="fu">ggplot</span>(ordination_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> condition)) <span class="sc">+</span></span>
<span id="cb261-3"><a href="#cb261-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb261-4"><a href="#cb261-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"PC1"</span>, <span class="at">y =</span> <span class="st">"PC2"</span>)  <span class="sc">+</span></span>
<span id="cb261-5"><a href="#cb261-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">"RDA"</span>) <span class="sc">+</span></span>
<span id="cb261-6"><a href="#cb261-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">10</span>)</span>
<span id="cb261-7"><a href="#cb261-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-8"><a href="#cb261-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-9"><a href="#cb261-9" aria-hidden="true" tabindex="-1"></a>   RDA_plot </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Now we will save all the plots together</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a><span class="fu">grid.arrange</span>(PCoA_plot, NMDS_plot, RDA_plot, <span class="at">ncol=</span><span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_3D.png"</span>, </span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a>       PCoA_plot,</span>
<span id="cb263-3"><a href="#cb263-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>,</span>
<span id="cb263-4"><a href="#cb263-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="hierachical-clustering-of-centered-log-ratios" class="level1">
<h1>7. Hierachical Clustering of Centered log ratios</h1>
<p>Here we plotted a heatmap displaying the z-score standardized centered log-ratio (CLR) of the top 20 most abundant genera.</p>
<section id="preparing-the-data-3" class="level2">
<h2 class="anchored" data-anchor-id="preparing-the-data-3">7.1 Preparing the data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb264"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb264-1"><a href="#cb264-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure Genus </span></span>
<span id="cb264-2"><a href="#cb264-2" aria-hidden="true" tabindex="-1"></a>  tse_genus <span class="ot">=</span> <span class="fu">altExp</span>(tse_metaphlan, <span class="st">"Genus"</span>)</span>
<span id="cb264-3"><a href="#cb264-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb264-4"><a href="#cb264-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add clr-transformation (centred-lofg ratio)</span></span>
<span id="cb264-5"><a href="#cb264-5" aria-hidden="true" tabindex="-1"></a>  tse_genus <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_genus, <span class="at">assay.type =</span> <span class="st">"counts"</span>, <span class="at">method =</span> <span class="st">"relabundance"</span>)</span>
<span id="cb264-6"><a href="#cb264-6" aria-hidden="true" tabindex="-1"></a>  tse_genus <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_genus, <span class="at">assay.type =</span> <span class="st">"relabundance"</span>,</span>
<span id="cb264-7"><a href="#cb264-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">"clr"</span>, <span class="at">pseudocount =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>A pseudocount of 7.6187051401118e-09 was applied.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb266"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb266-1"><a href="#cb266-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply a z-transformation across rows (taxa)</span></span>
<span id="cb266-2"><a href="#cb266-2" aria-hidden="true" tabindex="-1"></a>  tse_genus <span class="ot">=</span> <span class="fu">transformAssay</span>(tse_genus, <span class="at">assay.type =</span> <span class="st">"clr"</span>, </span>
<span id="cb266-3"><a href="#cb266-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">MARGIN =</span> <span class="st">"features"</span>,</span>
<span id="cb266-4"><a href="#cb266-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">"standardize"</span>, <span class="at">name =</span> <span class="st">"clr_z"</span>)</span>
<span id="cb266-5"><a href="#cb266-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb266-6"><a href="#cb266-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get 20 most abundant genera, and subsets the data by them</span></span>
<span id="cb266-7"><a href="#cb266-7" aria-hidden="true" tabindex="-1"></a>  top_20_genus <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(total_relabundance_Genus, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>])</span>
<span id="cb266-8"><a href="#cb266-8" aria-hidden="true" tabindex="-1"></a>  tse_genus_subset <span class="ot">=</span> tse_genus[top_20_genus, ]</span>
<span id="cb266-9"><a href="#cb266-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb266-10"><a href="#cb266-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate the proportion of bacteria accounted for in the top 20 Genera</span></span>
<span id="cb266-11"><a href="#cb266-11" aria-hidden="true" tabindex="-1"></a>  (<span class="fu">sum</span>(<span class="fu">assay</span>(tse_genus_subset, <span class="st">"relabundance"</span>))) <span class="sc">/</span> <span class="dv">15</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.226121</code></pre>
</div>
<div class="sourceCode cell-code" id="cb268"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb268-1"><a href="#cb268-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># delete to _g__</span></span>
<span id="cb268-2"><a href="#cb268-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(tse_genus_subset) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">".*(g__*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">rownames</span>(tse_genus_subset))</span>
<span id="cb268-3"><a href="#cb268-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb268-4"><a href="#cb268-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gets the assay table</span></span>
<span id="cb268-5"><a href="#cb268-5" aria-hidden="true" tabindex="-1"></a>  mat <span class="ot">=</span> <span class="fu">assay</span>(tse_genus_subset, <span class="st">"clr_z"</span>)</span>
<span id="cb268-6"><a href="#cb268-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb268-7"><a href="#cb268-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a dataframe for sample information</span></span>
<span id="cb268-8"><a href="#cb268-8" aria-hidden="true" tabindex="-1"></a>  sample_data <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse_genus_subset))</span>
<span id="cb268-9"><a href="#cb268-9" aria-hidden="true" tabindex="-1"></a>  sample_data <span class="ot">=</span> sample_data[, <span class="fu">c</span>(<span class="st">"condition"</span>), drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb268-10"><a href="#cb268-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-11"><a href="#cb268-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a dataframe for taxonomic information</span></span>
<span id="cb268-12"><a href="#cb268-12" aria-hidden="true" tabindex="-1"></a>  taxonomic_data <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">rowData</span>(tse_genus_subset))</span>
<span id="cb268-13"><a href="#cb268-13" aria-hidden="true" tabindex="-1"></a>  taxonomic_data <span class="ot">=</span> taxonomic_data[, <span class="st">"Phylum"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-heatmap-of-centred-log-ratios" class="level2">
<h2 class="anchored" data-anchor-id="plotting-heatmap-of-centred-log-ratios">7.2 Plotting heatmap of centred log ratios</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb269"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb269-1"><a href="#cb269-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set heatmap scaling and colors</span></span>
<span id="cb269-2"><a href="#cb269-2" aria-hidden="true" tabindex="-1"></a>  breaks <span class="ot">=</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), <span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), </span>
<span id="cb269-3"><a href="#cb269-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">length.out =</span> <span class="fu">ifelse</span>( <span class="fu">max</span>(<span class="fu">abs</span>(mat))<span class="sc">&gt;</span><span class="dv">5</span>, <span class="dv">2</span><span class="sc">*</span><span class="fu">ceiling</span>(<span class="fu">max</span>(<span class="fu">abs</span>(mat))), <span class="dv">10</span> ) )</span>
<span id="cb269-4"><a href="#cb269-4" aria-hidden="true" tabindex="-1"></a>  colors <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"darkblue"</span>, <span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>, <span class="st">"darkred"</span>))(<span class="fu">length</span>(breaks)<span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb269-5"><a href="#cb269-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb269-6"><a href="#cb269-6" aria-hidden="true" tabindex="-1"></a>  colors <span class="ot">=</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"#1E3A5F"</span>, <span class="st">"#4682B4"</span>, <span class="st">"#FFFFFF"</span>, <span class="st">"#E41A1C"</span>, <span class="st">"#A11618"</span>))(<span class="fu">length</span>(breaks) <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb269-7"><a href="#cb269-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb269-8"><a href="#cb269-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set column annotation colors</span></span>
<span id="cb269-9"><a href="#cb269-9" aria-hidden="true" tabindex="-1"></a>  ann_colors <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb269-10"><a href="#cb269-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">condition =</span> <span class="fu">c</span>(<span class="st">"healthy"</span> <span class="ot">=</span> <span class="st">"#4682B4"</span>, <span class="st">"diseased"</span> <span class="ot">=</span> <span class="st">"#E41A1C"</span>),</span>
<span id="cb269-11"><a href="#cb269-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">taxonomic_data =</span> <span class="fu">c</span>(<span class="fu">brewer.pal</span>(<span class="dv">6</span>, <span class="st">"Dark2"</span>))</span>
<span id="cb269-12"><a href="#cb269-12" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb269-13"><a href="#cb269-13" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb269-14"><a href="#cb269-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create the heatmap</span></span>
<span id="cb269-15"><a href="#cb269-15" aria-hidden="true" tabindex="-1"></a>  genus_clusters <span class="ot">=</span> <span class="fu">pheatmap</span>(mat, </span>
<span id="cb269-16"><a href="#cb269-16" aria-hidden="true" tabindex="-1"></a>                             <span class="at">annotation_col =</span> sample_data, </span>
<span id="cb269-17"><a href="#cb269-17" aria-hidden="true" tabindex="-1"></a>                             <span class="at">annotation_row =</span> taxonomic_data, </span>
<span id="cb269-18"><a href="#cb269-18" aria-hidden="true" tabindex="-1"></a>                             <span class="at">annotation_colors =</span> ann_colors, </span>
<span id="cb269-19"><a href="#cb269-19" aria-hidden="true" tabindex="-1"></a>                             <span class="at">color =</span> colors, </span>
<span id="cb269-20"><a href="#cb269-20" aria-hidden="true" tabindex="-1"></a>                             <span class="at">breaks =</span> breaks,</span>
<span id="cb269-21"><a href="#cb269-21" aria-hidden="true" tabindex="-1"></a>                             <span class="at">clustering_distance_cols =</span> <span class="st">"euclidean"</span>, </span>
<span id="cb269-22"><a href="#cb269-22" aria-hidden="true" tabindex="-1"></a>                             <span class="at">clustering_method =</span> <span class="st">"complete"</span>)</span>
<span id="cb269-23"><a href="#cb269-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb269-24"><a href="#cb269-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># print plot</span></span>
<span id="cb269-25"><a href="#cb269-25" aria-hidden="true" tabindex="-1"></a>  genus_clusters</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb270"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb270-1"><a href="#cb270-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_3E.png"</span>, </span>
<span id="cb270-2"><a href="#cb270-2" aria-hidden="true" tabindex="-1"></a>       genus_clusters,</span>
<span id="cb270-3"><a href="#cb270-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">20</span>,</span>
<span id="cb270-4"><a href="#cb270-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">12</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="determining-mean-centred-log-ratos" class="level2">
<h2 class="anchored" data-anchor-id="determining-mean-centred-log-ratos">7.3 Determining mean Centred log ratos</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb271"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb271-1"><a href="#cb271-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine CLR z-scores with Sample Conditions</span></span>
<span id="cb271-2"><a href="#cb271-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine mat and sample_data</span></span>
<span id="cb271-3"><a href="#cb271-3" aria-hidden="true" tabindex="-1"></a>  clr_z_with_condition <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(mat))</span>
<span id="cb271-4"><a href="#cb271-4" aria-hidden="true" tabindex="-1"></a>  clr_z_with_condition<span class="sc">$</span>condition <span class="ot">=</span> sample_data<span class="sc">$</span>condition</span>
<span id="cb271-5"><a href="#cb271-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-6"><a href="#cb271-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Specify genera of interest</span></span>
<span id="cb271-7"><a href="#cb271-7" aria-hidden="true" tabindex="-1"></a>  genera_of_interest <span class="ot">=</span> top_Genus_above_1</span>
<span id="cb271-8"><a href="#cb271-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-9"><a href="#cb271-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-10"><a href="#cb271-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Identify genera present in the CLR z-score data</span></span>
<span id="cb271-11"><a href="#cb271-11" aria-hidden="true" tabindex="-1"></a>  valid_genera <span class="ot">=</span> <span class="fu">intersect</span>(genera_of_interest, <span class="fu">colnames</span>(clr_z_with_condition))</span>
<span id="cb271-12"><a href="#cb271-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-13"><a href="#cb271-13" aria-hidden="true" tabindex="-1"></a>  valid_genera <span class="ot">=</span> <span class="fu">c</span>(valid_genera, <span class="st">"g__Filifactor"</span>)</span>
<span id="cb271-14"><a href="#cb271-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-15"><a href="#cb271-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset for valid genera and condition</span></span>
<span id="cb271-16"><a href="#cb271-16" aria-hidden="true" tabindex="-1"></a>  clr_z_subset <span class="ot">=</span> clr_z_with_condition[, <span class="fu">c</span>(valid_genera, <span class="st">"condition"</span>)]</span>
<span id="cb271-17"><a href="#cb271-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-18"><a href="#cb271-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate mean z-scores</span></span>
<span id="cb271-19"><a href="#cb271-19" aria-hidden="true" tabindex="-1"></a>  mean_clr_z_scores <span class="ot">=</span> clr_z_subset <span class="sc">%&gt;%</span></span>
<span id="cb271-20"><a href="#cb271-20" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(condition) <span class="sc">%&gt;%</span></span>
<span id="cb271-21"><a href="#cb271-21" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span> <span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.names =</span> <span class="st">"mean_{.col}"</span>))</span>
<span id="cb271-22"><a href="#cb271-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-23"><a href="#cb271-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate means</span></span>
<span id="cb271-24"><a href="#cb271-24" aria-hidden="true" tabindex="-1"></a>  mean_clr_z_scores <span class="ot">=</span> clr_z_subset <span class="sc">%&gt;%</span></span>
<span id="cb271-25"><a href="#cb271-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(condition) <span class="sc">%&gt;%</span></span>
<span id="cb271-26"><a href="#cb271-26" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">all_of</span>(valid_genera), <span class="sc">~</span> <span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb271-27"><a href="#cb271-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb271-28"><a href="#cb271-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># View the result</span></span>
<span id="cb271-29"><a href="#cb271-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(mean_clr_z_scores)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 2 × 21
  condition g__Prevotella g__Streptococcus g__Neisseria g__Haemophilus
  &lt;chr&gt;             &lt;dbl&gt;            &lt;dbl&gt;        &lt;dbl&gt;          &lt;dbl&gt;
1 diseased          0.129           -0.940        0.196         -0.323
2 healthy          -0.110            0.799       -0.166          0.275
# ℹ 16 more variables: g__Veillonella &lt;dbl&gt;, g__Rothia &lt;dbl&gt;,
#   g__Fusobacterium &lt;dbl&gt;, g__Treponema &lt;dbl&gt;, g__Schaalia &lt;dbl&gt;,
#   g__Escherichia &lt;dbl&gt;, g__Aggregatibacter &lt;dbl&gt;, g__Selenomonas &lt;dbl&gt;,
#   g__Leptotrichia &lt;dbl&gt;, g__Actinomyces &lt;dbl&gt;, g__Campylobacter &lt;dbl&gt;,
#   g__Capnocytophaga &lt;dbl&gt;, g__Porphyromonas &lt;dbl&gt;, g__Bacteroides &lt;dbl&gt;,
#   g__Gemella &lt;dbl&gt;, g__Filifactor &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb273"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb273-1"><a href="#cb273-1" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert to long format for ggplot</span></span>
<span id="cb273-2"><a href="#cb273-2" aria-hidden="true" tabindex="-1"></a>  clr_z_long <span class="ot">=</span> mean_clr_z_scores <span class="sc">%&gt;%</span> </span>
<span id="cb273-3"><a href="#cb273-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>condition, <span class="at">names_to =</span> <span class="st">"Genus"</span>, <span class="at">values_to =</span> <span class="st">"Mean_Z_Score"</span>)</span>
<span id="cb273-4"><a href="#cb273-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb273-5"><a href="#cb273-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb273-6"><a href="#cb273-6" aria-hidden="true" tabindex="-1"></a>  Mean_CLR_Z_Score <span class="ot">=</span> <span class="fu">ggplot</span>(clr_z_long, <span class="fu">aes</span>(<span class="at">x =</span> Mean_Z_Score, <span class="at">y =</span> condition, <span class="at">fill =</span> condition)) <span class="sc">+</span></span>
<span id="cb273-7"><a href="#cb273-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span>  <span class="co"># Black border around bars</span></span>
<span id="cb273-8"><a href="#cb273-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span> Genus, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span>  <span class="co"># Separate panels for each genus</span></span>
<span id="cb273-9"><a href="#cb273-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Condition"</span>, <span class="at">y =</span> <span class="st">"Mean CLR Z-Score"</span>, <span class="at">fill =</span> <span class="st">"Condition"</span>, <span class="at">title =</span> <span class="st">"Mean CLR Z-Scores by Genus and Condition"</span>) <span class="sc">+</span></span>
<span id="cb273-10"><a href="#cb273-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb273-11"><a href="#cb273-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">size =</span> <span class="dv">1</span>))  <span class="co"># Add a border around each panel </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `size` argument of `element_rect()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb275"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb275-1"><a href="#cb275-1" aria-hidden="true" tabindex="-1"></a>  Mean_CLR_Z_Score</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Noma_vs_healthy_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb276"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb276-1"><a href="#cb276-1" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="st">"../imgs/Figure_3F.png"</span>, </span>
<span id="cb276-2"><a href="#cb276-2" aria-hidden="true" tabindex="-1"></a>       Mean_CLR_Z_Score,</span>
<span id="cb276-3"><a href="#cb276-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">10</span>,</span>
<span id="cb276-4"><a href="#cb276-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>